// @tablecheck/tablekit-panel/lib/esm/ResizablePanel.js
import { forwardRef, useMemo, useRef, useEffect } from 'react';
import { ResizablePanelContainer } from './styled/ResizablePanel.js';
import { defaultMaxHeight, defaultMaxWidth, baseDefaultHeight, baseDefaultWidth } from './themes.js';
import { PanelPosition } from './types.js';
import { useBodyScrollLock } from './useBodyScrollLock.js';
import { jsxs } from '@emotion/react/jsx-runtime';
var ResizablePanel = /*#__PURE__*/forwardRef(function (_ref, _ref2) {
  var className = _ref.className,
      isOpen = _ref.isOpen,
      children = _ref.children,
      togglePanel = _ref.togglePanel,
      onClickOutside = _ref.onClickOutside,
      _ref$shouldDisableBod = _ref.shouldDisableBodyScroll,
      shouldDisableBodyScroll = _ref$shouldDisableBod === void 0 ? false : _ref$shouldDisableBod,
      _ref$maxHeight = _ref.maxHeight,
      maxHeight = _ref$maxHeight === void 0 ? defaultMaxHeight : _ref$maxHeight,
      _ref$maxWidth = _ref.maxWidth,
      maxWidth = _ref$maxWidth === void 0 ? defaultMaxWidth : _ref$maxWidth,
      _ref$minHeight = _ref.minHeight,
      minHeight = _ref$minHeight === void 0 ? baseDefaultHeight : _ref$minHeight,
      _ref$minWidth = _ref.minWidth,
      minWidth = _ref$minWidth === void 0 ? baseDefaultWidth : _ref$minWidth,
      _ref$position = _ref.position,
      position = _ref$position === void 0 ? PanelPosition.Right : _ref$position,
      _ref$defaultHeight = _ref.defaultHeight,
      defaultHeight = _ref$defaultHeight === void 0 ? baseDefaultHeight : _ref$defaultHeight,
      _ref$defaultWidth = _ref.defaultWidth,
      defaultWidth = _ref$defaultWidth === void 0 ? baseDefaultWidth : _ref$defaultWidth;
  var resizeDirections = useMemo(function () {
    return {
      top: position === 'bottom',
      right: position === 'left',
      bottom: position === 'top',
      left: position === 'right',
      topRight: false,
      bottomRight: false,
      bottomLeft: false,
      topLeft: false
    };
  }, [position]);
  var containerRef = useRef(null);
  var observer = useBodyScrollLock({
    shouldDisableBodyScroll: shouldDisableBodyScroll,
    containerRef: containerRef,
    isOpen: isOpen
  });
  useEffect(function () {
    var listener = onClickOutside ? function (event) {
      var domContainer = containerRef.current;

      if (event.target instanceof Node && domContainer && !domContainer.contains(event.target)) {
        onClickOutside(event, togglePanel);
      }
    } : null;

    if (isOpen && listener) {
      document.addEventListener('mousedown', listener);
    }

    return function () {
      if (listener) {
        document.removeEventListener('mousedown', listener);
      }
    };
  }, [containerRef, onClickOutside, isOpen, togglePanel]);
  var defaultSize = position === PanelPosition.Bottom || position === PanelPosition.Top ? {
    height: defaultHeight,
    width: '100% !important'
  } : {
    width: defaultWidth,
    height: '100% important'
  };
  return jsxs(ResizablePanelContainer, {
    className: className,
    enable: resizeDirections,
    isOpen: isOpen,
    position: position,
    defaultSize: defaultSize,
    minWidth: minWidth,
    maxWidth: position === PanelPosition.Bottom || position === PanelPosition.Top ? 'auto' : maxWidth,
    minHeight: minHeight,
    maxHeight: maxHeight,
    onClick: function onClick(ev) {
      return ev.stopPropagation();
    },
    handleWrapperClass: "resizer",
    "data-ignore": "scroll-lock-ignore",
    ref: function ref(element) {
      containerRef.current = (element === null || element === void 0 ? void 0 : element.resizable) || null;

      if (_ref2) {
        if (typeof _ref2 === 'function') {
          _ref2(element);
        } else {
          _ref2.current = element;
        }
      }
    },
    children: [isOpen && children, observer]
  });
});
export { ResizablePanel };
