// @tablecheck/tablekit-panel/lib/esm/useBodyScrollLock.js
import { disableBodyScroll, enableBodyScroll, clearAllBodyScrollLocks } from 'body-scroll-lock';
import { useEffect } from 'react';
import { jsx } from '@emotion/react/jsx-runtime';
var OBSERVER_CLASS_NAME = 'PanelScrollObserver';

var useBodyScrollLock = function useBodyScrollLock(_ref) {
  var shouldDisableBodyScroll = _ref.shouldDisableBodyScroll,
      containerRef = _ref.containerRef,
      isOpen = _ref.isOpen;
  useEffect(function () {
    var element = containerRef.current;

    if (shouldDisableBodyScroll && element) {
      if (isOpen) {
        disableBodyScroll(element, {
          allowTouchMove: function allowTouchMove(el) {
            while (el && el !== document.body) {
              if (el.getAttribute('data-ignore') === 'scroll-lock-ignore') {
                return true;
              } // eslint-disable-next-line no-param-reassign


              el = el.parentElement;
            }

            return false;
          }
        });
      } else {
        enableBodyScroll(element);
      }
    }
  }, [containerRef, isOpen, shouldDisableBodyScroll]);
  useEffect(function () {
    return function () {
      if (shouldDisableBodyScroll) {
        var hasOpenedPanel = document.getElementsByClassName(OBSERVER_CLASS_NAME).length > 0;

        if (!hasOpenedPanel) {
          clearAllBodyScrollLocks();
        }
      }
    };
  }, [shouldDisableBodyScroll]);
  var observer = isOpen && shouldDisableBodyScroll ? jsx("div", {
    style: {
      display: 'none'
    },
    className: OBSERVER_CLASS_NAME
  }) : null;
  return observer;
};

export { useBodyScrollLock };
