// @tablecheck/tablekit-spinner/lib/es5/components/styledContainer.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SpinnerOuter = exports.SpinnerContainer = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _types = require("../types.js");

var _variants, _variants2;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var SPINNER_SIZE = {
  REGULAR: _tablekitTheme.Spacing.L4,
  LARGE: _tablekitTheme.Spacing.L5
};
var SpinnerContainer = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "e1dai5il1"
} : {
  target: "e1dai5il1",
  label: "SpinnerContainer"
})("display:inline-flex;", (0, _tablekitUtils.variant)({
  prop: 'size',
  default: _types.SpinnerSize.Regular,
  variants: (_variants = {}, _defineProperty(_variants, _types.SpinnerSize.Large, /*#__PURE__*/(0, _react.css)("height:", SPINNER_SIZE.LARGE, ";width:", SPINNER_SIZE.LARGE, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:SpinnerContainer;"))), _defineProperty(_variants, _types.SpinnerSize.Regular, /*#__PURE__*/(0, _react.css)("height:", SPINNER_SIZE.REGULAR, ";width:", SPINNER_SIZE.REGULAR, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:SpinnerContainer;"))), _variants)
}), ";border-radius:10px;");
exports.SpinnerContainer = SpinnerContainer;
var SpinnerOuter = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "e1dai5il0"
} : {
  target: "e1dai5il0",
  label: "SpinnerOuter"
})("display:inline-flex;justify-content:center;align-items:center;", (0, _tablekitUtils.variant)({
  prop: 'size',
  default: _types.SpinnerSize.Regular,
  variants: (_variants2 = {}, _defineProperty(_variants2, _types.SpinnerSize.Large, /*#__PURE__*/(0, _react.css)("line-height:", SPINNER_SIZE.LARGE, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:SpinnerOuter;"))), _defineProperty(_variants2, _types.SpinnerSize.Regular, /*#__PURE__*/(0, _react.css)("line-height:", SPINNER_SIZE.REGULAR, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:SpinnerOuter;"))), _variants2)
}), ";");
exports.SpinnerOuter = SpinnerOuter;
