// @tablecheck/tablekit-select/lib/es5/createSelect.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.createSelect = createSelect;

var _react = require("@emotion/react");

var _tablekitInput = require("@tablecheck/tablekit-input");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _react2 = require("react");

var _reactSelect = require("react-select");

var animatedComponents = _interopRequireWildcard(require("react-select/animated"));

var _baseStyles = require("./baseStyles.js");

var index = _interopRequireWildcard(require("./components/index.js"));

var _styled = require("./styled.js");

var _themes = require("./themes.js");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

var _excluded = ["children"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var buildCachedComponents = function buildCachedComponents() {
  var components = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread(_objectSpread(_objectSpread({}, animatedComponents), index), components);
};

function createSelect(WrappedComponent) {
  var Select = function Select(props) {
    var options = props.options,
        placeholder = props.placeholder,
        iconBefore = props.iconBefore,
        isDisabled = props.isDisabled,
        components$1 = props.components,
        isInvalid = props.isInvalid,
        isWarning = props.isWarning,
        isSuccess = props.isSuccess,
        _props$size = props.size,
        size = _props$size === void 0 ? _tablekitInput.InputSize.Regular : _props$size,
        styles = props.styles,
        onChange = props.onChange,
        shouldNotChangeDropdownIcon = props.shouldNotChangeDropdownIcon,
        name = props.name;
    var tablekitTheme = (0, _react.useTheme)();
    var builtComponents = buildCachedComponents(components$1);

    if (!shouldNotChangeDropdownIcon && !isDisabled && isInvalid) {
      builtComponents.DropdownIndicator = builtComponents.ErrorIndicator;
    } else if (!shouldNotChangeDropdownIcon && !isDisabled && isWarning) {
      builtComponents.DropdownIndicator = builtComponents.WarningIndicator;
    } else if (!shouldNotChangeDropdownIcon && !isDisabled && isSuccess) {
      builtComponents.DropdownIndicator = builtComponents.SuccessIndicator;
    }

    if (iconBefore) {
      var Control = _reactSelect.components.Control;

      builtComponents.Control = function (_ref) {
        var children = _ref.children,
            innerProps = _objectWithoutProperties(_ref, _excluded);

        return (// cant use reactSelectComponents.Control directly here as it will refer to the local
          // components instead of reactSelectComponents in build
          (0, _jsxRuntime.jsxs)(Control, _objectSpread(_objectSpread({}, innerProps), {}, {
            children: [(0, _jsxRuntime.jsx)(_styled.IconContainer, {
              size: size,
              children: /*#__PURE__*/(0, _react2.cloneElement)(iconBefore, {
                size: size
              })
            }), children]
          }))
        );
      };
    }

    var handleChange = function handleChange(value, actionMeta) {
      if (onChange) {
        onChange(value, _objectSpread(_objectSpread({}, actionMeta), {}, {
          name: name
        }));
      }
    }; // Used to merge our theme with the ReactSelect theme


    var getTheme = function getTheme(reactSelectTheme) {
      var _selectClassicTheme$c, _selectClassicTheme$o, _selectClassicTheme$o2;

      var safeTheme = {
        theme: _objectSpread(_objectSpread({}, tablekitTheme), {}, {
          colors: tablekitTheme.colors || _tablekitTheme.CLASSIC_COLORS
        })
      };
      return _objectSpread(_objectSpread(_objectSpread({}, reactSelectTheme), tablekitTheme), {}, {
        borderRadius: _tablekitTheme.BORDER_RADIUS,
        spacing: _objectSpread(_objectSpread({}, reactSelectTheme.spacing || {}), {}, {
          controlHeight: _tablekitTheme.FieldHeight.Small
        }),
        colors: _objectSpread(_objectSpread(_objectSpread({}, safeTheme.theme.colors), tablekitTheme.colors), {}, {
          danger: tablekitTheme.colors.danger,
          dangerLight: tablekitTheme.colors.danger,
          neutral0: (0, _tablekitUtils.getThemeValue)("".concat(_themes.selectThemeNamespace, ".common.backgroundColor"), _themes.selectClassicTheme === null || _themes.selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$c = _themes.selectClassicTheme.common) === null || _selectClassicTheme$c === void 0 ? void 0 : _selectClassicTheme$c.backgroundColor)(safeTheme),
          primary: tablekitTheme.colors.primary,
          primary75: tablekitTheme.colors.primary,
          primary50: (0, _tablekitUtils.getThemeValue)("".concat(_themes.selectThemeNamespace, ".option.active.backgroundColor"), _themes.selectClassicTheme === null || _themes.selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$o = _themes.selectClassicTheme.option) === null || _selectClassicTheme$o === void 0 ? void 0 : (_selectClassicTheme$o2 = _selectClassicTheme$o.active) === null || _selectClassicTheme$o2 === void 0 ? void 0 : _selectClassicTheme$o2.backgroundColor)(safeTheme),
          primary25: tablekitTheme.colors.primary2
        })
      });
    };

    var tablekitStyles = (0, _baseStyles.baseStyles)(tablekitTheme, isInvalid, isWarning, isSuccess);
    return (// TODO weird mismatch going on here in builtComponents
      // eslint-disable-next-line @typescript-eslint/ban-ts-comment
      // @ts-ignore
      (0, _jsxRuntime.jsx)(WrappedComponent, _objectSpread(_objectSpread({}, props), {}, {
        options: options,
        placeholder: placeholder,
        isDisabled: isDisabled,
        theme: getTheme,
        components: builtComponents,
        onChange: handleChange,
        styles: styles ? (0, _reactSelect.mergeStyles)(tablekitStyles, styles) : tablekitStyles
      }))
    );
  };

  return Select;
}
