// @tablecheck/tablekit-select/lib/es5/SelectWrapper.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.SelectWrapper = void 0;

var _tablekitInput = require("@tablecheck/tablekit-input");

var _react = require("react");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

var SelectWrapper = function SelectWrapper(_ref) {
  var children = _ref.children,
      className = _ref.className,
      isRequired = _ref.isRequired,
      isInvalid = _ref.isInvalid,
      isMessageHidden = _ref.isMessageHidden,
      isWarning = _ref.isWarning,
      isSuccess = _ref.isSuccess,
      label = _ref.label,
      id = _ref.id,
      message = _ref.message;
  return (0, _jsxRuntime.jsxs)("div", {
    className: className,
    children: [label !== undefined ? (0, _jsxRuntime.jsx)(_tablekitInput.Label, {
      htmlFor: "react-select-".concat(id, "--input"),
      isRequired: isRequired,
      label: label || ''
    }) : null, /*#__PURE__*/(0, _react.cloneElement)(children, {
      instanceId: id,
      isInvalid: isInvalid,
      isWarning: isWarning,
      isSuccess: isSuccess
    }), !isMessageHidden && (0, _jsxRuntime.jsx)(_tablekitInput.InputMessage, {
      children: message
    })]
  });
};

exports.SelectWrapper = SelectWrapper;
