// @tablecheck/tablekit-select/lib/es5/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.IconContainer = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitInput = require("@tablecheck/tablekit-input");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _themes = require("./themes.js");

var _selectClassicTheme$i, _variants;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var IconContainer = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "e153nee80"
} : {
  target: "e153nee80",
  label: "IconContainer"
})("display:flex;align-items:center;color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.selectThemeNamespace, ".icon.color"), _themes.selectClassicTheme === null || _themes.selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$i = _themes.selectClassicTheme.icon) === null || _selectClassicTheme$i === void 0 ? void 0 : _selectClassicTheme$i.color), ";", (0, _tablekitUtils.variant)({
  prop: 'size',
  default: _tablekitInput.InputSize.Regular,
  variants: (_variants = {}, _defineProperty(_variants, _tablekitInput.InputSize.Large, process.env.NODE_ENV === "production" ? {
    name: "f9bxyi",
    styles: "padding-left:18px"
  } : {
    name: "tdfrwo-IconContainer",
    styles: "padding-left:18px;label:IconContainer;",
    toString: _EMOTION_STRINGIFIED_CSS_ERROR__
  }), _defineProperty(_variants, _tablekitInput.InputSize.Regular, /*#__PURE__*/(0, _react.css)("padding-left:", _tablekitTheme.Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:IconContainer;"))), _defineProperty(_variants, _tablekitInput.InputSize.Regular2, /*#__PURE__*/(0, _react.css)("padding-left:", _tablekitTheme.Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:IconContainer;"))), _defineProperty(_variants, _tablekitInput.InputSize.Small, process.env.NODE_ENV === "production" ? {
    name: "vnst0l",
    styles: "padding-left:10px"
  } : {
    name: "11cmjzs-IconContainer",
    styles: "padding-left:10px;label:IconContainer;",
    toString: _EMOTION_STRINGIFIED_CSS_ERROR__
  }), _variants)
}), ";");
exports.IconContainer = IconContainer;
