// @tablecheck/tablekit-input/lib/es5/Input.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Input = void 0;

var _tablekitIcon = require("@tablecheck/tablekit-icon");

var _tablekitSpinner = require("@tablecheck/tablekit-spinner");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _react = require("react");

var _Label = require("./Label.js");

var _index = require("./styled/index.js");

var _types = require("./types.js");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

var _excluded = ["id", "appearance", "className", "type", "shouldFitContainer", "isRequired", "label", "inputCss", "inputClassName", "message", "isMessageHidden", "isSpellCheckEnabled", "name", "placeholder", "onChange", "value", "size", "shape", "width", "alt"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

var Input = /*#__PURE__*/(0, _react.forwardRef)(function (props, ref) {
  var iconBefore = props.iconBefore,
      iconAfter = props.iconAfter;

  var id = props.id,
      appearance = props.appearance,
      className = props.className,
      type = props.type,
      shouldFitContainer = props.shouldFitContainer,
      isRequired = props.isRequired,
      label = props.label,
      inputCss = props.inputCss,
      inputClassName = props.inputClassName,
      message = props.message,
      isMessageHidden = props.isMessageHidden,
      isSpellCheckEnabled = props.isSpellCheckEnabled,
      name = props.name,
      placeholder = props.placeholder,
      onChange = props.onChange,
      value = props.value,
      _props$size = props.size,
      size = _props$size === void 0 ? _types.InputSize.Regular : _props$size,
      _props$shape = props.shape,
      shape = _props$shape === void 0 ? _types.InputShape.Rounded : _props$shape,
      _props$width = props.width,
      width = _props$width === void 0 ? _tablekitTheme.Size.Regular : _props$width,
      alt = props.alt,
      restProps = _objectWithoutProperties(props, _excluded);

  var isDisabled = appearance === _types.Appearance.Disabled;

  if (type === 'search') {
    iconBefore = (0, _jsxRuntime.jsx)(_tablekitIcon.Icon, {
      icon: (0, _tablekitIcon.getIcon)('search')
    });
  }

  var labelElement = label !== undefined ? (0, _jsxRuntime.jsx)(_Label.Label, {
    htmlFor: id,
    isRequired: isRequired,
    label: label
  }) : null;

  switch (appearance) {
    case _types.Appearance.Loading:
      iconAfter = (0, _jsxRuntime.jsx)(_tablekitSpinner.Spinner, {});
      break;

    case _types.Appearance.Invalid:
      iconAfter = (0, _jsxRuntime.jsx)(_tablekitIcon.DangerIcon, {});
      break;

    case _types.Appearance.Warning:
      iconAfter = (0, _jsxRuntime.jsx)(_tablekitIcon.WarningIcon, {});
      break;

    case _types.Appearance.Success:
      iconAfter = (0, _jsxRuntime.jsx)(_tablekitIcon.SuccessIcon, {});
      break;

    default:
  }

  return (0, _jsxRuntime.jsxs)(_index.InputContainer, {
    className: className,
    shouldFitContainer: shouldFitContainer,
    width: width,
    children: [labelElement, (0, _jsxRuntime.jsxs)(_index.InputWrapper, {
      fieldSize: size,
      children: [iconBefore && (0, _jsxRuntime.jsx)(_index.InputPrependElement, {
        isDisabled: isDisabled,
        elementSize: size,
        children: /*#__PURE__*/(0, _react.cloneElement)(iconBefore, {
          size: size
        })
      }), (0, _jsxRuntime.jsx)(_index.InputField, _objectSpread(_objectSpread({}, restProps), {}, {
        appearance: appearance,
        shape: shape,
        className: inputClassName,
        css: inputCss,
        disabled: isDisabled,
        hasIconAfter: !!iconAfter,
        hasIconBefore: !!iconBefore,
        id: id,
        ref: ref,
        name: name,
        onChange: onChange,
        placeholder: placeholder,
        alt: alt,
        readOnly: appearance === _types.Appearance.ReadOnly,
        required: isRequired,
        spellCheck: isSpellCheckEnabled,
        fieldSize: size,
        type: type,
        value: value
      })), iconAfter && (0, _jsxRuntime.jsx)(_index.InputAppendElement, {
        isDisabled: isDisabled,
        elementSize: size,
        children: /*#__PURE__*/(0, _react.cloneElement)(iconAfter, {
          size: size
        })
      })]
    }), !isMessageHidden && (0, _jsxRuntime.jsx)(_index.InputMessage, {
      children: message
    })]
  });
});
exports.Input = Input;
