// @tablecheck/tablekit-input/lib/es5/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputSize = exports.InputShape = exports.Appearance = void 0;
var InputSize;
exports.InputSize = InputSize;

(function (InputSize) {
  InputSize["Large"] = "large";
  InputSize["Regular"] = "regular";
  InputSize["Regular2"] = "regular2";
  InputSize["Small"] = "small";
})(InputSize || (exports.InputSize = InputSize = {}));

var InputShape;
exports.InputShape = InputShape;

(function (InputShape) {
  InputShape["Sharp"] = "sharp";
  InputShape["Rounded"] = "rounded";
  InputShape["Circular"] = "circular";
})(InputShape || (exports.InputShape = InputShape = {}));

var Appearance;
exports.Appearance = Appearance;

(function (Appearance) {
  Appearance["Default"] = "default";
  Appearance["Invalid"] = "invalid";
  Appearance["Disabled"] = "disabled";
  Appearance["Loading"] = "loading";
  Appearance["ReadOnly"] = "readonly";
  Appearance["Success"] = "success";
  Appearance["Warning"] = "warning";
})(Appearance || (exports.Appearance = Appearance = {}));
