// @tablecheck/tablekit-theme/lib/es5/fonts/plexSans.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.plexSansFont = plexSansFont;

var _react = require("@emotion/react");

var _buildFontFace = require("./buildFontFace.js");

var _weights = require("./weights.js");

var name = 'IBM Plex Sans';
var folder = 'IBMPlex-Sans';

function plexSansFont(fontWeights) {
  var sourceSansProExtraLightWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-ExtraLight.woff');
  var sourceSansProExtraLightWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-ExtraLight.woff2');
  var sourceSansProExtraLightItalicWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-ExtraLightItalic.woff');
  var sourceSansProExtraLightItalicWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-ExtraLightItalic.woff2');
  var sourceSansProLightWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Light.woff');
  var sourceSansProLightWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Light.woff2');
  var sourceSansProLightItalicWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-LightItalic.woff');
  var sourceSansProLightItalicWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-LightItalic.woff2');
  var sourceSansProRegularWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Regular.woff');
  var sourceSansProRegularWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Regular.woff2');
  var sourceSansProRegularItalicWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Italic.woff');
  var sourceSansProRegularItalicWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Italic.woff2');
  var sourceSansProMediumWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Medium.woff');
  var sourceSansProMediumWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Medium.woff2');
  var sourceSansProMediumItalicWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-MediumItalic.woff');
  var sourceSansProMediumItalicWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-MediumItalic.woff2');
  var sourceSansProSemiBoldWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-SemiBold.woff');
  var sourceSansProSemiBoldWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-SemiBold.woff2');
  var sourceSansProSemiBoldItalicWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-SemiBoldItalic.woff');
  var sourceSansProSemiBoldItalicWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-SemiBoldItalic.woff2');
  var sourceSansProBoldWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Bold.woff');
  var sourceSansProBoldWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-Bold.woff2');
  var sourceSansProBoldItalicWoff = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-BoldItalic.woff');
  var sourceSansProBoldItalicWoff2 = (0, _buildFontFace.getCdnPath)(folder, 'IBMPlexSans-BoldItalic.woff2');
  var preloadLinks = fontWeights.reduce(function (result, fontWeight) {
    if (fontWeight === _weights.FontWeight.ExtraLight) {
      return result.concat([(0, _buildFontFace.buildPreloadLink)(sourceSansProExtraLightWoff2), (0, _buildFontFace.buildPreloadLink)(sourceSansProExtraLightItalicWoff2)]);
    }

    if (fontWeight === _weights.FontWeight.Light) {
      return result.concat([(0, _buildFontFace.buildPreloadLink)(sourceSansProLightWoff2), (0, _buildFontFace.buildPreloadLink)(sourceSansProLightItalicWoff2)]);
    }

    if (fontWeight === _weights.FontWeight.Regular) {
      return result.concat([(0, _buildFontFace.buildPreloadLink)(sourceSansProRegularWoff2), (0, _buildFontFace.buildPreloadLink)(sourceSansProRegularItalicWoff2)]);
    }

    if (fontWeight === _weights.FontWeight.Medium) {
      return result.concat([(0, _buildFontFace.buildPreloadLink)(sourceSansProMediumWoff2), (0, _buildFontFace.buildPreloadLink)(sourceSansProMediumItalicWoff2)]);
    }

    if (fontWeight === _weights.FontWeight.SemiBold) {
      return result.concat([(0, _buildFontFace.buildPreloadLink)(sourceSansProSemiBoldWoff2), (0, _buildFontFace.buildPreloadLink)(sourceSansProSemiBoldItalicWoff2)]);
    }

    if (fontWeight === _weights.FontWeight.Bold) {
      return result.concat([(0, _buildFontFace.buildPreloadLink)(sourceSansProBoldWoff2), (0, _buildFontFace.buildPreloadLink)(sourceSansProBoldItalicWoff2)]);
    }

    return result;
  }, []);
  return [preloadLinks, /*#__PURE__*/(0, _react.css)((0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.ExtraLight,
    woff: sourceSansProExtraLightWoff,
    woff2: sourceSansProExtraLightWoff2
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.ExtraLight,
    woff: sourceSansProExtraLightItalicWoff,
    woff2: sourceSansProExtraLightItalicWoff2,
    isItalic: true
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.Light,
    woff: sourceSansProLightWoff,
    woff2: sourceSansProLightWoff2
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.Light,
    woff: sourceSansProLightItalicWoff,
    woff2: sourceSansProLightItalicWoff2,
    isItalic: true
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.Regular,
    woff: sourceSansProRegularWoff,
    woff2: sourceSansProRegularWoff2
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.Regular,
    woff: sourceSansProRegularItalicWoff,
    woff2: sourceSansProRegularItalicWoff2,
    isItalic: true
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.Medium,
    woff: sourceSansProMediumWoff,
    woff2: sourceSansProMediumWoff2
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.Medium,
    woff: sourceSansProMediumItalicWoff,
    woff2: sourceSansProMediumItalicWoff2,
    isItalic: true
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.SemiBold,
    woff: sourceSansProSemiBoldWoff,
    woff2: sourceSansProSemiBoldWoff2
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.SemiBold,
    woff: sourceSansProSemiBoldItalicWoff,
    woff2: sourceSansProSemiBoldItalicWoff2,
    isItalic: true
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.Bold,
    woff: sourceSansProBoldWoff,
    woff2: sourceSansProBoldWoff2
  }), " ", (0, _buildFontFace.buildFontFace)({
    name: name,
    weight: _weights.FontWeight.Bold,
    woff: sourceSansProBoldItalicWoff,
    woff2: sourceSansProBoldItalicWoff2,
    isItalic: true
  }), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:plexSansFont;"))];
}
