// @tablecheck/tablekit-input/lib/es5/styled/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InputWrapper = exports.InputPrependElement = exports.InputMessage = exports.InputField = exports.InputContainer = exports.InputAppendElement = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitTypography = require("@tablecheck/tablekit-typography");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _themes = require("../themes.js");

var _types = require("../types.js");

var _utils = require("./utils.js");

var _inputClassicTheme$in, _inputClassicTheme$in2, _variants, _inputClassicTheme$er;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var InputContainer = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "ee7mo056"
} : {
  target: "ee7mo056",
  label: "InputContainer"
})("display:flex;flex-direction:column;flex:1 0 auto;align-items:flex-start;justify-content:flex-start;max-width:", function (_ref) {
  var shouldFitContainer = _ref.shouldFitContainer,
      width = _ref.width;
  return shouldFitContainer ? '100%' : _themes.INPUT_WIDTHS[width];
}, ";");
exports.InputContainer = InputContainer;
var InputWrapper = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "ee7mo055"
} : {
  target: "ee7mo055",
  label: "InputWrapper"
})("position:relative;", _utils.getContainerShape, ";width:100%;");
exports.InputWrapper = InputWrapper;
var InputField = (0, _base.default)("input", process.env.NODE_ENV === "production" ? {
  target: "ee7mo054"
} : {
  target: "ee7mo054",
  label: "InputField"
})("position:absolute;left:0;top:0;right:0;bottom:0;width:100%;outline:none;transition:box-shadow 200ms ease-in-out;&,&::placeholder{", _tablekitTypography.Typography.Body1, ";line-height:initial;}&::placeholder{color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".input.default.placeholderColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in = _themes.inputClassicTheme.input) === null || _inputClassicTheme$in === void 0 ? void 0 : (_inputClassicTheme$in2 = _inputClassicTheme$in.default) === null || _inputClassicTheme$in2 === void 0 ? void 0 : _inputClassicTheme$in2.placeholderColor), ";}", _utils.getPadding, ";", _utils.getContainerShape, ";", _utils.getContext, ";&[type='search']{-webkit-appearance:textfield;}&[type='search']::-webkit-search-decoration{-webkit-appearance:none;}@-moz-document url-prefix(){&::placeholder{line-height:revert;}}");
exports.InputField = InputField;
var IconElement = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "ee7mo053"
} : {
  target: "ee7mo053",
  label: "IconElement"
})("position:absolute;display:flex;align-items:center;top:0;pointer-events:none;color:", function (props) {
  var _inputClassicTheme$in3, _inputClassicTheme$in4, _inputClassicTheme$in5, _inputClassicTheme$in6;

  return props.isDisabled ? (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".input.disabled.iconColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in3 = _themes.inputClassicTheme.input) === null || _inputClassicTheme$in3 === void 0 ? void 0 : (_inputClassicTheme$in4 = _inputClassicTheme$in3.disabled) === null || _inputClassicTheme$in4 === void 0 ? void 0 : _inputClassicTheme$in4.iconColor)(props) : (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".input.default.iconColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in5 = _themes.inputClassicTheme.input) === null || _inputClassicTheme$in5 === void 0 ? void 0 : (_inputClassicTheme$in6 = _inputClassicTheme$in5.default) === null || _inputClassicTheme$in6 === void 0 ? void 0 : _inputClassicTheme$in6.iconColor)(props);
}, ";", (0, _tablekitUtils.variant)({
  prop: 'elementSize',
  default: _types.InputSize.Regular,
  variants: (_variants = {}, _defineProperty(_variants, _types.InputSize.Large, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Large, ";width:", _tablekitTheme.FieldHeight.Large, ";padding:0 18px;" + (process.env.NODE_ENV === "production" ? "" : ";label:IconElement;"))), _defineProperty(_variants, _types.InputSize.Regular, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Regular, ";width:", _tablekitTheme.FieldHeight.Regular, ";padding:0 ", _tablekitTheme.Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:IconElement;"))), _defineProperty(_variants, _types.InputSize.Regular2, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Regular2, ";width:", _tablekitTheme.FieldHeight.Regular2, ";padding:0 ", _tablekitTheme.Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:IconElement;"))), _defineProperty(_variants, _types.InputSize.Small, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Small, ";width:", _tablekitTheme.FieldHeight.Small, ";padding:0 10px;" + (process.env.NODE_ENV === "production" ? "" : ";label:IconElement;"))), _variants)
}), ";");
var InputAppendElement = /*#__PURE__*/(0, _base.default)(IconElement, process.env.NODE_ENV === "production" ? {
  target: "ee7mo052"
} : {
  target: "ee7mo052",
  label: "InputAppendElement"
})((0, _tablekitUtils.ifRtl)('left', 'right'), ":0;");
exports.InputAppendElement = InputAppendElement;
var InputPrependElement = /*#__PURE__*/(0, _base.default)(IconElement, process.env.NODE_ENV === "production" ? {
  target: "ee7mo051"
} : {
  target: "ee7mo051",
  label: "InputPrependElement"
})((0, _tablekitUtils.ifRtl)('right', 'left'), ":0;");
exports.InputPrependElement = InputPrependElement;
var InputMessage = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "ee7mo050"
} : {
  target: "ee7mo050",
  label: "InputMessage"
})(_tablekitTypography.Typography.Body2, ";min-height:", _tablekitTypography.Typography.Body2.lineHeight, ";color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".error.textColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$er = _themes.inputClassicTheme.error) === null || _inputClassicTheme$er === void 0 ? void 0 : _inputClassicTheme$er.textColor), ";margin-top:", _tablekitTheme.Spacing.L2, ";width:100%;");
exports.InputMessage = InputMessage;
