// @tablecheck/tablekit-input/lib/es5/styled/label.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.RequiredIndicator = exports.InputLabelWrapper = exports.InputLabelInner = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitTypography = require("@tablecheck/tablekit-typography");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _themes = require("../themes.js");

var _inputClassicTheme$la;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var InputLabelWrapper = (0, _base.default)("label", process.env.NODE_ENV === "production" ? {
  target: "ehdunnl2"
} : {
  target: "ehdunnl2",
  label: "InputLabelWrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "1082qq3",
  styles: "display:block;width:100%"
} : {
  name: "1082qq3",
  styles: "display:block;width:100%",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});
exports.InputLabelWrapper = InputLabelWrapper;
var InputLabelInner = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "ehdunnl1"
} : {
  target: "ehdunnl1",
  label: "InputLabelInner"
})(_tablekitTypography.Typography.Label2, ";font-style:normal;letter-spacing:0em;text-align:", (0, _tablekitUtils.ifRtl)('right', 'left'), ";color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".label.textColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$la = _themes.inputClassicTheme.label) === null || _inputClassicTheme$la === void 0 ? void 0 : _inputClassicTheme$la.textColor), ";margin-bottom:", _tablekitTheme.Spacing.L2, ";position:relative;");
exports.InputLabelInner = InputLabelInner;
var RequiredIndicator = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "ehdunnl0"
} : {
  target: "ehdunnl0",
  label: "RequiredIndicator"
})("color:", function (_ref) {
  var theme = _ref.theme;
  return theme.colors.danger;
}, ";", (0, _tablekitUtils.padding)({
  left: '2px'
}), ";");
exports.RequiredIndicator = RequiredIndicator;
