// @tablecheck/tablekit-input/lib/es5/styled/utils.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.getContainerShape = exports.getBorderRadius = void 0;
exports.getContext = getContext;
exports.getPadding = void 0;

var _react = require("@emotion/react");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _themes = require("../themes.js");

var _types = require("../types.js");

var _excluded = ["hasIconAfter", "hasIconBefore"],
    _excluded2 = ["hasIconAfter", "hasIconBefore"],
    _excluded3 = ["hasIconAfter", "hasIconBefore"],
    _excluded4 = ["hasIconAfter", "hasIconBefore"];

var _variants, _variants2;

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var getBorderRadius = function getBorderRadius(fieldHeight) {
  var shape = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : _types.InputShape.Rounded;

  switch (shape) {
    case _types.InputShape.Circular:
      return fieldHeight;

    case _types.InputShape.Sharp:
      return 0;

    default:
      return "".concat(_tablekitTheme.BORDER_RADIUS, "px");
  }
};

exports.getBorderRadius = getBorderRadius;

function getContext(props) {
  var _inputClassicTheme$in5, _inputClassicTheme$in6, _inputClassicTheme$in7, _inputClassicTheme$in8, _inputClassicTheme$in9, _inputClassicTheme$in10;

  var appearance = props.appearance;

  if (appearance === _types.Appearance.Disabled) {
    var _inputClassicTheme$in, _inputClassicTheme$in2, _inputClassicTheme$in3, _inputClassicTheme$in4;

    var _borderColor = (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".input.disabled.borderColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in = _themes.inputClassicTheme.input) === null || _inputClassicTheme$in === void 0 ? void 0 : (_inputClassicTheme$in2 = _inputClassicTheme$in.disabled) === null || _inputClassicTheme$in2 === void 0 ? void 0 : _inputClassicTheme$in2.borderColor)(props);

    var _textColor = (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".input.disabled.textColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in3 = _themes.inputClassicTheme.input) === null || _inputClassicTheme$in3 === void 0 ? void 0 : (_inputClassicTheme$in4 = _inputClassicTheme$in3.disabled) === null || _inputClassicTheme$in4 === void 0 ? void 0 : _inputClassicTheme$in4.textColor)(props);

    return /*#__PURE__*/(0, _react.css)("box-shadow:inset 0px 0px 0px 1px ", _borderColor, ";color:", _textColor, ";opacity:0.5;cursor:not-allowed;" + (process.env.NODE_ENV === "production" ? "" : ";label:getContext;"));
  }

  var borderColor;

  switch (appearance) {
    case _types.Appearance.Invalid:
      borderColor = props.theme.colors.danger;
      break;

    case _types.Appearance.Success:
      borderColor = props.theme.colors.success;
      break;

    case _types.Appearance.Warning:
      borderColor = props.theme.colors.warning;
      break;

    default:
      borderColor = (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".input.default.borderColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in5 = _themes.inputClassicTheme.input) === null || _inputClassicTheme$in5 === void 0 ? void 0 : (_inputClassicTheme$in6 = _inputClassicTheme$in5.default) === null || _inputClassicTheme$in6 === void 0 ? void 0 : _inputClassicTheme$in6.borderColor)(props);
  }

  var textColor = (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".input.default.textColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in7 = _themes.inputClassicTheme.input) === null || _inputClassicTheme$in7 === void 0 ? void 0 : (_inputClassicTheme$in8 = _inputClassicTheme$in7.default) === null || _inputClassicTheme$in8 === void 0 ? void 0 : _inputClassicTheme$in8.textColor)(props);
  var backgroundColor = (0, _tablekitUtils.getThemeValue)("".concat(_themes.inputThemeNamespace, ".input.default.backgroundColor"), _themes.inputClassicTheme === null || _themes.inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in9 = _themes.inputClassicTheme.input) === null || _inputClassicTheme$in9 === void 0 ? void 0 : (_inputClassicTheme$in10 = _inputClassicTheme$in9.default) === null || _inputClassicTheme$in10 === void 0 ? void 0 : _inputClassicTheme$in10.backgroundColor)(props);
  return /*#__PURE__*/(0, _react.css)("background-color:", backgroundColor, ";color:", textColor, ";box-shadow:inset 0px 0px 0px 1px ", borderColor, ";&:hover{box-shadow:inset 0px 0px 0px 2px ", function (_ref) {
    var theme = _ref.theme;
    return theme.colors.primary2;
  }(props), ";}&:focus{box-shadow:inset 0px 0px 0px 2px ", function (_ref2) {
    var theme = _ref2.theme;
    return theme.colors.secondary;
  }(props), ";outline:none;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getContext;"));
}

var getContainerShape = (0, _tablekitUtils.variant)({
  prop: 'fieldSize',
  default: _types.InputSize.Regular,
  variants: (_variants = {}, _defineProperty(_variants, _types.InputSize.Large, function (_ref3) {
    var shape = _ref3.shape;
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Large, ";border-radius:", getBorderRadius(_tablekitTheme.FieldHeight.Large, shape), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getContainerShape;"));
  }), _defineProperty(_variants, _types.InputSize.Regular, function (_ref4) {
    var shape = _ref4.shape;
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Regular, ";border-radius:", getBorderRadius(_tablekitTheme.FieldHeight.Regular, shape), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getContainerShape;"));
  }), _defineProperty(_variants, _types.InputSize.Regular2, function (_ref5) {
    var shape = _ref5.shape;
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Regular2, ";border-radius:", getBorderRadius(_tablekitTheme.FieldHeight.Regular2, shape), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getContainerShape;"));
  }), _defineProperty(_variants, _types.InputSize.Small, function (_ref6) {
    var shape = _ref6.shape;
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Small, ";border-radius:", getBorderRadius(_tablekitTheme.FieldHeight.Small, shape), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getContainerShape;"));
  }), _variants)
});
exports.getContainerShape = getContainerShape;
var getPadding = (0, _tablekitUtils.variant)({
  prop: 'fieldSize',
  default: _types.InputSize.Regular,
  variants: (_variants2 = {}, _defineProperty(_variants2, _types.InputSize.Large, function (_ref7) {
    var hasIconAfter = _ref7.hasIconAfter,
        hasIconBefore = _ref7.hasIconBefore,
        props = _objectWithoutProperties(_ref7, _excluded);

    return /*#__PURE__*/(0, _react.css)("padding-", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Large : '18px', ";padding-", (0, _tablekitUtils.ifRtl)('right', 'left')(props), ":", hasIconBefore ? _tablekitTheme.FieldHeight.Large : '18px', ";background-position:right ", _tablekitTheme.FieldHeight.Large, " top 50%!important;&:hover,&:focus{padding-", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Large : '18px', ";padding-", (0, _tablekitUtils.ifRtl)('right', 'left')(props), ":", hasIconBefore ? _tablekitTheme.FieldHeight.Large : '18px', ";background-position:right ", _tablekitTheme.FieldHeight.Large, " top 50%!important;}&~span{left:auto!important;right:auto!important;", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Large : '18px', "!important;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getPadding;"));
  }), _defineProperty(_variants2, _types.InputSize.Regular, function (_ref8) {
    var hasIconAfter = _ref8.hasIconAfter,
        hasIconBefore = _ref8.hasIconBefore,
        props = _objectWithoutProperties(_ref8, _excluded2);

    return /*#__PURE__*/(0, _react.css)("padding-", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Regular : _tablekitTheme.Spacing.L3, ";padding-", (0, _tablekitUtils.ifRtl)('right', 'left')(props), ":", hasIconBefore ? _tablekitTheme.FieldHeight.Regular : _tablekitTheme.Spacing.L3, ";background-position:right ", _tablekitTheme.FieldHeight.Regular, " top 50%!important;&:hover,&:focus{padding-", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Regular : _tablekitTheme.Spacing.L3, ";padding-", (0, _tablekitUtils.ifRtl)('right', 'left')(props), ":", hasIconBefore ? _tablekitTheme.FieldHeight.Regular : _tablekitTheme.Spacing.L3, ";background-position:right ", _tablekitTheme.FieldHeight.Regular, " top 50%!important;}&~span{left:auto!important;right:auto!important;", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Regular : _tablekitTheme.Spacing.L3, "!important;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getPadding;"));
  }), _defineProperty(_variants2, _types.InputSize.Regular2, function (_ref9) {
    var hasIconAfter = _ref9.hasIconAfter,
        hasIconBefore = _ref9.hasIconBefore,
        props = _objectWithoutProperties(_ref9, _excluded3);

    return /*#__PURE__*/(0, _react.css)("padding-", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Regular2 : _tablekitTheme.Spacing.L3, ";padding-", (0, _tablekitUtils.ifRtl)('right', 'left')(props), ":", hasIconBefore ? _tablekitTheme.FieldHeight.Regular2 : _tablekitTheme.Spacing.L3, ";background-position:right ", _tablekitTheme.FieldHeight.Regular2, " top 50%!important;&:hover,&:focus{padding-", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Regular2 : _tablekitTheme.Spacing.L3, ";padding-", (0, _tablekitUtils.ifRtl)('right', 'left')(props), ":", hasIconBefore ? _tablekitTheme.FieldHeight.Regular2 : _tablekitTheme.Spacing.L3, ";background-position:right ", _tablekitTheme.FieldHeight.Regular2, " top 50%!important;}&~span{left:auto!important;right:auto!important;", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Regular2 : _tablekitTheme.Spacing.L3, "!important;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getPadding;"));
  }), _defineProperty(_variants2, _types.InputSize.Small, function (_ref10) {
    var hasIconAfter = _ref10.hasIconAfter,
        hasIconBefore = _ref10.hasIconBefore,
        props = _objectWithoutProperties(_ref10, _excluded4);

    return /*#__PURE__*/(0, _react.css)("padding-", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Small : '10px', ";padding-", (0, _tablekitUtils.ifRtl)('right', 'left')(props), ":", hasIconBefore ? _tablekitTheme.FieldHeight.Small : '10px', ";background-position:right ", _tablekitTheme.FieldHeight.Small, " top 50%!important;&:hover,&:focus{padding-", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Small : '10px', ";padding-", (0, _tablekitUtils.ifRtl)('right', 'left')(props), ":", hasIconBefore ? _tablekitTheme.FieldHeight.Small : '10px', ";background-position:right ", _tablekitTheme.FieldHeight.Small, " top 50%!important;}&~span{left:auto!important;right:auto!important;", (0, _tablekitUtils.ifRtl)('left', 'right')(props), ":", hasIconAfter ? _tablekitTheme.FieldHeight.Small : '10px', "!important;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getPadding;"));
  }), _variants2)
});
exports.getPadding = getPadding;
