// @tablecheck/tablekit-tag/lib/es5/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.transition = exports.borderWidth = exports.TagWrapper = exports.TagContent = exports.StyledTag = exports.RemoveIconWrapper = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitTypography = require("@tablecheck/tablekit-typography");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _themes = require("./themes.js");

var _types = require("./types.js");

var _tagClassicTheme$comm, _variants, _variants2;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var borderWidth = 1;
exports.borderWidth = borderWidth;
var transition = process.env.NODE_ENV === "production" ? {
  name: "1uaihgp",
  styles: "transition:120ms ease-in-out"
} : {
  name: "1w3axge-transition",
  styles: "transition:120ms ease-in-out;label:transition;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};
exports.transition = transition;
var TagContent = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "e1ed3sk3"
} : {
  target: "e1ed3sk3",
  label: "TagContent"
})("display:flex;align-items:center;align-self:center;margin:", function (_ref) {
  var isSelected = _ref.isSelected;
  return isSelected ? '0 6px 0 2px' : "0 ".concat(_tablekitTheme.Spacing.L3);
}, ";max-width:100%;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;font-weight:bold;");
exports.TagContent = TagContent;
var RemoveIconWrapper = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "e1ed3sk2"
} : {
  target: "e1ed3sk2",
  label: "RemoveIconWrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "xatqlf",
  styles: "align-items:center;justify-content:center;width:auto;height:auto;align-self:center;display:flex;flex-shrink:0;margin:0;user-select:none"
} : {
  name: "xatqlf",
  styles: "align-items:center;justify-content:center;width:auto;height:auto;align-self:center;display:flex;flex-shrink:0;margin:0;user-select:none",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});
exports.RemoveIconWrapper = RemoveIconWrapper;
var TagWrapper = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "e1ed3sk1"
} : {
  target: "e1ed3sk1",
  label: "TagWrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "1j7wp6a",
  styles: "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative"
} : {
  name: "1j7wp6a",
  styles: "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});
exports.TagWrapper = TagWrapper;

function renderAppearance(appearance, props) {
  var isSelected = props.isSelected;

  if (isSelected) {
    return /*#__PURE__*/(0, _react.css)("color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.tagThemeNamespace, ".").concat(appearance, ".selectedTextColor"), _themes.tagClassicTheme[appearance].selectedTextColor)(props), ";background:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.tagThemeNamespace, ".").concat(appearance, ".selectedBackgroundColor"), _themes.tagClassicTheme[appearance].selectedBackgroundColor)(props), ";&:hover{background:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.tagThemeNamespace, ".").concat(appearance, ".hoverSelectedBackgroundColor"), _themes.tagClassicTheme[appearance].hoverSelectedBackgroundColor)(props), ";}", RemoveIconWrapper, "{color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.tagThemeNamespace, ".").concat(appearance, ".selectedTextColor"), _themes.tagClassicTheme[appearance].selectedTextColor)(props), ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:renderAppearance;"));
  }

  return /*#__PURE__*/(0, _react.css)("color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.tagThemeNamespace, ".").concat(appearance, ".textColor"), _themes.tagClassicTheme[appearance].textColor)(props), ";background:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.tagThemeNamespace, ".").concat(appearance, ".backgroundColor"), _themes.tagClassicTheme[appearance].backgroundColor)(props), ";&:hover{background:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.tagThemeNamespace, ".").concat(appearance, ".hoverBackgroundColor"), _themes.tagClassicTheme[appearance].hoverBackgroundColor)(props), ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:renderAppearance;"));
}

var StyledTag = (0, _base.default)("button", process.env.NODE_ENV === "production" ? {
  target: "e1ed3sk0"
} : {
  target: "e1ed3sk0",
  label: "StyledTag"
})("text-decoration:none;border-color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.tagThemeNamespace, ".common.borderColor"), _themes.tagClassicTheme === null || _themes.tagClassicTheme === void 0 ? void 0 : (_tagClassicTheme$comm = _themes.tagClassicTheme.common) === null || _tagClassicTheme$comm === void 0 ? void 0 : _tagClassicTheme$comm.borderColor), ";", (0, _tablekitUtils.variant)({
  prop: 'appearance',
  default: _types.TagAppearance.Outline,
  variants: (_variants = {}, _defineProperty(_variants, _types.TagAppearance.Outline, function (props) {
    return renderAppearance(_types.TagAppearance.Outline, props);
  }), _defineProperty(_variants, _types.TagAppearance.Solid, function (props) {
    return renderAppearance(_types.TagAppearance.Solid, props);
  }), _variants)
}), " align-items:baseline;border-width:", borderWidth, "px;border-style:solid;box-sizing:border-box;cursor:pointer;display:inline-flex;margin:0;max-width:100%;outline:none;pointer-events:auto;text-align:center;white-space:nowrap;width:auto;vertical-align:middle;border-radius:", _tablekitTheme.BORDER_RADIUS, "px;", _tablekitTypography.Typography.Body2, ";padding:0 ", _tablekitTheme.Spacing.L1, ";", (0, _tablekitUtils.variant)({
  prop: 'size',
  default: _types.TagSize.Regular,
  variants: (_variants2 = {}, _defineProperty(_variants2, _types.TagSize.Regular, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.Spacing.L6, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:StyledTag;"))), _defineProperty(_variants2, _types.TagSize.Small, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.Spacing.L5, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:StyledTag;"))), _variants2)
}), " & .svg-inline--fa{width:", _tablekitTheme.Spacing.L4, ";height:", _tablekitTheme.Spacing.L4, ";}&::-moz-focus-inner{border:0;margin:0;padding:0;}font-weight:bold;", transition, ";&::after{border-radius:", _tablekitTheme.BORDER_RADIUS, "px;", transition, ";}");
exports.StyledTag = StyledTag;
