// @tablecheck/tablekit-input/lib/esm/styled/index.js
var _inputClassicTheme$in, _inputClassicTheme$in2, _variants, _inputClassicTheme$er;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import _styled from '@emotion/styled/base';
import { css } from '@emotion/react';
import { FieldHeight, Spacing } from '@tablecheck/tablekit-theme';
import { Typography } from '@tablecheck/tablekit-typography';
import { getThemeValue, variant, ifRtl } from '@tablecheck/tablekit-utils';
import { INPUT_WIDTHS, inputThemeNamespace, inputClassicTheme } from '../themes.js';
import { InputSize } from '../types.js';
import { getContainerShape, getPadding, getContext } from './utils.js';

var InputContainer = _styled("div", process.env.NODE_ENV === "production" ? {
  target: "ee7mo056"
} : {
  target: "ee7mo056",
  label: "InputContainer"
})("display:flex;flex-direction:column;flex:1 0 auto;align-items:flex-start;justify-content:flex-start;max-width:", function (_ref) {
  var shouldFitContainer = _ref.shouldFitContainer,
      width = _ref.width;
  return shouldFitContainer ? '100%' : INPUT_WIDTHS[width];
}, ";");

var InputWrapper = _styled("div", process.env.NODE_ENV === "production" ? {
  target: "ee7mo055"
} : {
  target: "ee7mo055",
  label: "InputWrapper"
})("position:relative;", getContainerShape, ";width:100%;");

var InputField = _styled("input", process.env.NODE_ENV === "production" ? {
  target: "ee7mo054"
} : {
  target: "ee7mo054",
  label: "InputField"
})("position:absolute;left:0;top:0;right:0;bottom:0;width:100%;outline:none;transition:box-shadow 200ms ease-in-out;&,&::placeholder{", Typography.Body1, ";line-height:initial;}&::placeholder{color:", getThemeValue("".concat(inputThemeNamespace, ".input.default.placeholderColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in = inputClassicTheme.input) === null || _inputClassicTheme$in === void 0 ? void 0 : (_inputClassicTheme$in2 = _inputClassicTheme$in.default) === null || _inputClassicTheme$in2 === void 0 ? void 0 : _inputClassicTheme$in2.placeholderColor), ";}", getPadding, ";", getContainerShape, ";", getContext, ";&[type='search']{-webkit-appearance:textfield;}&[type='search']::-webkit-search-decoration{-webkit-appearance:none;}@-moz-document url-prefix(){&::placeholder{line-height:revert;}}");

var IconElement = _styled("div", process.env.NODE_ENV === "production" ? {
  target: "ee7mo053"
} : {
  target: "ee7mo053",
  label: "IconElement"
})("position:absolute;display:flex;align-items:center;top:0;pointer-events:none;color:", function (props) {
  var _inputClassicTheme$in3, _inputClassicTheme$in4, _inputClassicTheme$in5, _inputClassicTheme$in6;

  return props.isDisabled ? getThemeValue("".concat(inputThemeNamespace, ".input.disabled.iconColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in3 = inputClassicTheme.input) === null || _inputClassicTheme$in3 === void 0 ? void 0 : (_inputClassicTheme$in4 = _inputClassicTheme$in3.disabled) === null || _inputClassicTheme$in4 === void 0 ? void 0 : _inputClassicTheme$in4.iconColor)(props) : getThemeValue("".concat(inputThemeNamespace, ".input.default.iconColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in5 = inputClassicTheme.input) === null || _inputClassicTheme$in5 === void 0 ? void 0 : (_inputClassicTheme$in6 = _inputClassicTheme$in5.default) === null || _inputClassicTheme$in6 === void 0 ? void 0 : _inputClassicTheme$in6.iconColor)(props);
}, ";", variant({
  prop: 'elementSize',
  default: InputSize.Regular,
  variants: (_variants = {}, _defineProperty(_variants, InputSize.Large, /*#__PURE__*/css("height:", FieldHeight.Large, ";width:", FieldHeight.Large, ";padding:0 18px;" + (process.env.NODE_ENV === "production" ? "" : ";label:IconElement;"))), _defineProperty(_variants, InputSize.Regular, /*#__PURE__*/css("height:", FieldHeight.Regular, ";width:", FieldHeight.Regular, ";padding:0 ", Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:IconElement;"))), _defineProperty(_variants, InputSize.Regular2, /*#__PURE__*/css("height:", FieldHeight.Regular2, ";width:", FieldHeight.Regular2, ";padding:0 ", Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:IconElement;"))), _defineProperty(_variants, InputSize.Small, /*#__PURE__*/css("height:", FieldHeight.Small, ";width:", FieldHeight.Small, ";padding:0 10px;" + (process.env.NODE_ENV === "production" ? "" : ";label:IconElement;"))), _variants)
}), ";");

var InputAppendElement = /*#__PURE__*/_styled(IconElement, process.env.NODE_ENV === "production" ? {
  target: "ee7mo052"
} : {
  target: "ee7mo052",
  label: "InputAppendElement"
})(ifRtl('left', 'right'), ":0;");

var InputPrependElement = /*#__PURE__*/_styled(IconElement, process.env.NODE_ENV === "production" ? {
  target: "ee7mo051"
} : {
  target: "ee7mo051",
  label: "InputPrependElement"
})(ifRtl('right', 'left'), ":0;");

var InputMessage = _styled("div", process.env.NODE_ENV === "production" ? {
  target: "ee7mo050"
} : {
  target: "ee7mo050",
  label: "InputMessage"
})(Typography.Body2, ";min-height:", Typography.Body2.lineHeight, ";color:", getThemeValue("".concat(inputThemeNamespace, ".error.textColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$er = inputClassicTheme.error) === null || _inputClassicTheme$er === void 0 ? void 0 : _inputClassicTheme$er.textColor), ";margin-top:", Spacing.L2, ";width:100%;");

export { InputAppendElement, InputContainer, InputField, InputMessage, InputPrependElement, InputWrapper };
