// @tablecheck/tablekit-input/lib/esm/styled/utils.js
var _excluded = ["hasIconAfter", "hasIconBefore"],
    _excluded2 = ["hasIconAfter", "hasIconBefore"],
    _excluded3 = ["hasIconAfter", "hasIconBefore"],
    _excluded4 = ["hasIconAfter", "hasIconBefore"];

var _variants, _variants2;

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { css } from '@emotion/react';
import { BORDER_RADIUS, FieldHeight, Spacing } from '@tablecheck/tablekit-theme';
import { getThemeValue, variant, ifRtl } from '@tablecheck/tablekit-utils';
import { inputThemeNamespace, inputClassicTheme } from '../themes.js';
import { InputShape, Appearance, InputSize } from '../types.js';

var getBorderRadius = function getBorderRadius(fieldHeight) {
  var shape = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : InputShape.Rounded;

  switch (shape) {
    case InputShape.Circular:
      return fieldHeight;

    case InputShape.Sharp:
      return 0;

    default:
      return "".concat(BORDER_RADIUS, "px");
  }
};

function getContext(props) {
  var _inputClassicTheme$in5, _inputClassicTheme$in6, _inputClassicTheme$in7, _inputClassicTheme$in8, _inputClassicTheme$in9, _inputClassicTheme$in10;

  var appearance = props.appearance;

  if (appearance === Appearance.Disabled) {
    var _inputClassicTheme$in, _inputClassicTheme$in2, _inputClassicTheme$in3, _inputClassicTheme$in4;

    var _borderColor = getThemeValue("".concat(inputThemeNamespace, ".input.disabled.borderColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in = inputClassicTheme.input) === null || _inputClassicTheme$in === void 0 ? void 0 : (_inputClassicTheme$in2 = _inputClassicTheme$in.disabled) === null || _inputClassicTheme$in2 === void 0 ? void 0 : _inputClassicTheme$in2.borderColor)(props);

    var _textColor = getThemeValue("".concat(inputThemeNamespace, ".input.disabled.textColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in3 = inputClassicTheme.input) === null || _inputClassicTheme$in3 === void 0 ? void 0 : (_inputClassicTheme$in4 = _inputClassicTheme$in3.disabled) === null || _inputClassicTheme$in4 === void 0 ? void 0 : _inputClassicTheme$in4.textColor)(props);

    return /*#__PURE__*/css("box-shadow:inset 0px 0px 0px 1px ", _borderColor, ";color:", _textColor, ";opacity:0.5;cursor:not-allowed;" + (process.env.NODE_ENV === "production" ? "" : ";label:getContext;"));
  }

  var borderColor;

  switch (appearance) {
    case Appearance.Invalid:
      borderColor = props.theme.colors.danger;
      break;

    case Appearance.Success:
      borderColor = props.theme.colors.success;
      break;

    case Appearance.Warning:
      borderColor = props.theme.colors.warning;
      break;

    default:
      borderColor = getThemeValue("".concat(inputThemeNamespace, ".input.default.borderColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in5 = inputClassicTheme.input) === null || _inputClassicTheme$in5 === void 0 ? void 0 : (_inputClassicTheme$in6 = _inputClassicTheme$in5.default) === null || _inputClassicTheme$in6 === void 0 ? void 0 : _inputClassicTheme$in6.borderColor)(props);
  }

  var textColor = getThemeValue("".concat(inputThemeNamespace, ".input.default.textColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in7 = inputClassicTheme.input) === null || _inputClassicTheme$in7 === void 0 ? void 0 : (_inputClassicTheme$in8 = _inputClassicTheme$in7.default) === null || _inputClassicTheme$in8 === void 0 ? void 0 : _inputClassicTheme$in8.textColor)(props);
  var backgroundColor = getThemeValue("".concat(inputThemeNamespace, ".input.default.backgroundColor"), inputClassicTheme === null || inputClassicTheme === void 0 ? void 0 : (_inputClassicTheme$in9 = inputClassicTheme.input) === null || _inputClassicTheme$in9 === void 0 ? void 0 : (_inputClassicTheme$in10 = _inputClassicTheme$in9.default) === null || _inputClassicTheme$in10 === void 0 ? void 0 : _inputClassicTheme$in10.backgroundColor)(props);
  return /*#__PURE__*/css("background-color:", backgroundColor, ";color:", textColor, ";box-shadow:inset 0px 0px 0px 1px ", borderColor, ";&:hover{box-shadow:inset 0px 0px 0px 2px ", function (_ref) {
    var theme = _ref.theme;
    return theme.colors.primary2;
  }(props), ";}&:focus{box-shadow:inset 0px 0px 0px 2px ", function (_ref2) {
    var theme = _ref2.theme;
    return theme.colors.secondary;
  }(props), ";outline:none;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getContext;"));
}

var getContainerShape = variant({
  prop: 'fieldSize',
  default: InputSize.Regular,
  variants: (_variants = {}, _defineProperty(_variants, InputSize.Large, function (_ref3) {
    var shape = _ref3.shape;
    return /*#__PURE__*/css("height:", FieldHeight.Large, ";border-radius:", getBorderRadius(FieldHeight.Large, shape), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getContainerShape;"));
  }), _defineProperty(_variants, InputSize.Regular, function (_ref4) {
    var shape = _ref4.shape;
    return /*#__PURE__*/css("height:", FieldHeight.Regular, ";border-radius:", getBorderRadius(FieldHeight.Regular, shape), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getContainerShape;"));
  }), _defineProperty(_variants, InputSize.Regular2, function (_ref5) {
    var shape = _ref5.shape;
    return /*#__PURE__*/css("height:", FieldHeight.Regular2, ";border-radius:", getBorderRadius(FieldHeight.Regular2, shape), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getContainerShape;"));
  }), _defineProperty(_variants, InputSize.Small, function (_ref6) {
    var shape = _ref6.shape;
    return /*#__PURE__*/css("height:", FieldHeight.Small, ";border-radius:", getBorderRadius(FieldHeight.Small, shape), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getContainerShape;"));
  }), _variants)
});
var getPadding = variant({
  prop: 'fieldSize',
  default: InputSize.Regular,
  variants: (_variants2 = {}, _defineProperty(_variants2, InputSize.Large, function (_ref7) {
    var hasIconAfter = _ref7.hasIconAfter,
        hasIconBefore = _ref7.hasIconBefore,
        props = _objectWithoutProperties(_ref7, _excluded);

    return /*#__PURE__*/css("padding-", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Large : '18px', ";padding-", ifRtl('right', 'left')(props), ":", hasIconBefore ? FieldHeight.Large : '18px', ";background-position:right ", FieldHeight.Large, " top 50%!important;&:hover,&:focus{padding-", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Large : '18px', ";padding-", ifRtl('right', 'left')(props), ":", hasIconBefore ? FieldHeight.Large : '18px', ";background-position:right ", FieldHeight.Large, " top 50%!important;}&~span{left:auto!important;right:auto!important;", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Large : '18px', "!important;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getPadding;"));
  }), _defineProperty(_variants2, InputSize.Regular, function (_ref8) {
    var hasIconAfter = _ref8.hasIconAfter,
        hasIconBefore = _ref8.hasIconBefore,
        props = _objectWithoutProperties(_ref8, _excluded2);

    return /*#__PURE__*/css("padding-", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Regular : Spacing.L3, ";padding-", ifRtl('right', 'left')(props), ":", hasIconBefore ? FieldHeight.Regular : Spacing.L3, ";background-position:right ", FieldHeight.Regular, " top 50%!important;&:hover,&:focus{padding-", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Regular : Spacing.L3, ";padding-", ifRtl('right', 'left')(props), ":", hasIconBefore ? FieldHeight.Regular : Spacing.L3, ";background-position:right ", FieldHeight.Regular, " top 50%!important;}&~span{left:auto!important;right:auto!important;", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Regular : Spacing.L3, "!important;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getPadding;"));
  }), _defineProperty(_variants2, InputSize.Regular2, function (_ref9) {
    var hasIconAfter = _ref9.hasIconAfter,
        hasIconBefore = _ref9.hasIconBefore,
        props = _objectWithoutProperties(_ref9, _excluded3);

    return /*#__PURE__*/css("padding-", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Regular2 : Spacing.L3, ";padding-", ifRtl('right', 'left')(props), ":", hasIconBefore ? FieldHeight.Regular2 : Spacing.L3, ";background-position:right ", FieldHeight.Regular2, " top 50%!important;&:hover,&:focus{padding-", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Regular2 : Spacing.L3, ";padding-", ifRtl('right', 'left')(props), ":", hasIconBefore ? FieldHeight.Regular2 : Spacing.L3, ";background-position:right ", FieldHeight.Regular2, " top 50%!important;}&~span{left:auto!important;right:auto!important;", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Regular2 : Spacing.L3, "!important;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getPadding;"));
  }), _defineProperty(_variants2, InputSize.Small, function (_ref10) {
    var hasIconAfter = _ref10.hasIconAfter,
        hasIconBefore = _ref10.hasIconBefore,
        props = _objectWithoutProperties(_ref10, _excluded4);

    return /*#__PURE__*/css("padding-", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Small : '10px', ";padding-", ifRtl('right', 'left')(props), ":", hasIconBefore ? FieldHeight.Small : '10px', ";background-position:right ", FieldHeight.Small, " top 50%!important;&:hover,&:focus{padding-", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Small : '10px', ";padding-", ifRtl('right', 'left')(props), ":", hasIconBefore ? FieldHeight.Small : '10px', ";background-position:right ", FieldHeight.Small, " top 50%!important;}&~span{left:auto!important;right:auto!important;", ifRtl('left', 'right')(props), ":", hasIconAfter ? FieldHeight.Small : '10px', "!important;}" + (process.env.NODE_ENV === "production" ? "" : ";label:getPadding;"));
  }), _variants2)
});
export { getBorderRadius, getContainerShape, getContext, getPadding };
