// @tablecheck/tablekit-item/lib/es5/ItemGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ItemGroup = void 0;

var _react = require("react");

var _ItemGroup = require("./styled/ItemGroup.js");

var _types = require("./types.js");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

var ItemGroup = /*#__PURE__*/(0, _react.forwardRef)(function (_ref, ref) {
  var children = _ref.children,
      _ref$role = _ref.role,
      role = _ref$role === void 0 ? 'group' : _ref$role,
      _ref$orientation = _ref.orientation,
      orientation = _ref$orientation === void 0 ? _types.ItemGroupOrientation.Vertical : _ref$orientation,
      _ref$hasIndent = _ref.hasIndent,
      hasIndent = _ref$hasIndent === void 0 ? false : _ref$hasIndent,
      headingComponent = _ref.headingComponent,
      headingText = _ref.headingText,
      headingStyles = _ref.headingStyles,
      minWidth = _ref.minWidth,
      className = _ref.className,
      isUppercase = _ref.isUppercase,
      shouldWrap = _ref.shouldWrap;
  var heading = headingComponent || null;

  if (headingText) {
    heading = (0, _jsxRuntime.jsx)(_ItemGroup.TitleContainer, {
      children: (0, _jsxRuntime.jsx)(_ItemGroup.TitleText, {
        headingStyles: headingStyles,
        isUppercase: isUppercase,
        children: headingText
      })
    });
  }

  return (0, _jsxRuntime.jsxs)(_ItemGroup.ItemGroupContainer, {
    role: role,
    minWidth: minWidth,
    className: className,
    ref: ref,
    children: [heading, (0, _jsxRuntime.jsx)("div", {
      children: (0, _jsxRuntime.jsx)(_ItemGroup.ItemGroupContent, {
        orientation: orientation,
        hasIndent: hasIndent,
        shouldWrap: shouldWrap,
        children: children
      })
    })]
  });
});
exports.ItemGroup = ItemGroup;
