// @tablecheck/tablekit-item/lib/es5/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ItemInteractionState = exports.ItemGroupOrientation = exports.ItemBorderPosition = void 0;
var ItemBorderPosition;
exports.ItemBorderPosition = ItemBorderPosition;

(function (ItemBorderPosition) {
  ItemBorderPosition["Left"] = "left";
  ItemBorderPosition["Right"] = "right";
  ItemBorderPosition["Top"] = "top";
  ItemBorderPosition["Bottom"] = "bottom";
})(ItemBorderPosition || (exports.ItemBorderPosition = ItemBorderPosition = {}));

var ItemInteractionState;
exports.ItemInteractionState = ItemInteractionState;

(function (ItemInteractionState) {
  ItemInteractionState["Default"] = "default";
  ItemInteractionState["Hidden"] = "hidden";
  ItemInteractionState["Selected"] = "selected";
  ItemInteractionState["Disabled"] = "disabled";
})(ItemInteractionState || (exports.ItemInteractionState = ItemInteractionState = {}));

var ItemGroupOrientation;
exports.ItemGroupOrientation = ItemGroupOrientation;

(function (ItemGroupOrientation) {
  ItemGroupOrientation["Horizontal"] = "horizontal";
  ItemGroupOrientation["Vertical"] = "vertical";
})(ItemGroupOrientation || (exports.ItemGroupOrientation = ItemGroupOrientation = {}));
