// @tablecheck/tablekit-input-button/lib/esm/styled.js
var _variants2, _variants3, _variants4;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import _styled from '@emotion/styled/base';
import { css } from '@emotion/react';
import { ButtonShape } from '@tablecheck/tablekit-button';
import { Icon } from '@tablecheck/tablekit-icon';
import { InputSize } from '@tablecheck/tablekit-input';
import { BORDER_RADIUS, FieldHeight, FontWeight } from '@tablecheck/tablekit-theme';
import { Typography } from '@tablecheck/tablekit-typography';
import { ifRtl, variant } from '@tablecheck/tablekit-utils';
import { TRANSITION_SETTINGS, IS_CLICKED_SELECTOR } from './constants.js';
var transition = /*#__PURE__*/css("transition:border-color ", TRANSITION_SETTINGS, ",color ", TRANSITION_SETTINGS, ",background-color ", TRANSITION_SETTINGS, ",box-shadow ", TRANSITION_SETTINGS, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:transition;"));

var SelectedIcon = /*#__PURE__*/_styled(Icon, process.env.NODE_ENV === "production" ? {
  target: "e173lbmz4"
} : {
  target: "e173lbmz4",
  label: "SelectedIcon"
})("width:15px;height:15px;margin:2.5px;margin-", ifRtl('left', 'right'), ":10.5px;color:", function (_ref) {
  var theme = _ref.theme;
  return theme.colors.primary2;
}, ";");

var ButtonLabel = _styled("label", process.env.NODE_ENV === "production" ? {
  target: "e173lbmz3"
} : {
  target: "e173lbmz3",
  label: "ButtonLabel"
})("display:inline-flex;flex-direction:row;position:relative;cursor:pointer;", Typography.Body2, ";&,& *{user-select:none;}&[data-disabled='true']{cursor:not-allowed;}");

var ButtonDisplay = _styled("div", process.env.NODE_ENV === "production" ? {
  target: "e173lbmz2"
} : {
  target: "e173lbmz2",
  label: "ButtonDisplay"
})("border-radius:", variant({
  prop: 'shape',
  variants: (_variants2 = {}, _defineProperty(_variants2, ButtonShape.Sharp, 0), _defineProperty(_variants2, ButtonShape.Rounded, "".concat(BORDER_RADIUS, "px")), _defineProperty(_variants2, ButtonShape.Circular, function (props) {
    var _variants;

    return variant({
      prop: 'fieldSize',
      default: InputSize.Regular,
      variants: (_variants = {}, _defineProperty(_variants, InputSize.Large, FieldHeight.Large), _defineProperty(_variants, InputSize.Regular, FieldHeight.Regular), _defineProperty(_variants, InputSize.Regular2, FieldHeight.Regular2), _defineProperty(_variants, InputSize.Small, FieldHeight.Small), _variants)
    })(props);
  }), _variants2)
}), ";border:1px solid ", function (_ref2) {
  var theme = _ref2.theme;
  return theme.colors.border;
}, ";color:", function (_ref3) {
  var theme = _ref3.theme;
  return theme.colors.primary2;
}, ";max-width:300px;display:flex;justify-content:center;align-items:center;text-align:center;", transition, ";background-color:", function (_ref4) {
  var theme = _ref4.theme;
  return theme.colors.inputBackground;
}, ";&:not(", IS_CLICKED_SELECTOR, "):after{content:'';display:block;position:absolute;top:-5px;right:-5px;bottom:-5px;left:-5px;border:2px solid transparent;border-radius:", BORDER_RADIUS + 2, "px;", transition, ";}");

var TextWrapper = _styled("div", process.env.NODE_ENV === "production" ? {
  target: "e173lbmz1"
} : {
  target: "e173lbmz1",
  label: "TextWrapper"
})("overflow:hidden;text-overflow:ellipsis;white-space:nowrap;", Typography.Body1, ";font-weight:", FontWeight.SemiBold, ";&::after{content:attr(data-text);height:0;visibility:hidden;overflow:hidden;user-select:none;pointer-events:none;font-weight:600;display:block;@media speech{display:none;}}");

var HiddenInput = _styled("input", process.env.NODE_ENV === "production" ? {
  target: "e173lbmz0"
} : {
  target: "e173lbmz0",
  label: "HiddenInput"
})("position:absolute;user-select:none;opacity:0;&+div{", variant({
  prop: 'fieldSize',
  default: InputSize.Regular,
  variants: (_variants3 = {}, _defineProperty(_variants3, InputSize.Large, function () {
    return /*#__PURE__*/css("height:", FieldHeight.Large, ";padding:0 32px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _defineProperty(_variants3, InputSize.Regular, function () {
    return /*#__PURE__*/css("height:", FieldHeight.Regular, ";padding:0 22px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _defineProperty(_variants3, InputSize.Regular2, function () {
    return /*#__PURE__*/css("height:", FieldHeight.Regular2, ";padding:0 26px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _defineProperty(_variants3, InputSize.Small, function () {
    return /*#__PURE__*/css("height:", FieldHeight.Small, ";padding:0 20px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _variants3)
}), ";}&+div>svg{display:none;}&:hover+div{background:", function (_ref5) {
  var theme = _ref5.theme;
  return theme.colors.hoverPrimaryBackground;
}, ";}&:checked+div>svg{display:block;}&:checked+div{background:", function (_ref6) {
  var theme = _ref6.theme;
  return theme.colors.primarySelected;
}, ";border:1px solid ", function (_ref7) {
  var theme = _ref7.theme;
  return theme.colors.primary2;
}, ";", variant({
  prop: 'fieldSize',
  default: InputSize.Regular,
  variants: (_variants4 = {}, _defineProperty(_variants4, InputSize.Large, /*#__PURE__*/css("height:", FieldHeight.Large, ";padding:0 18px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"))), _defineProperty(_variants4, InputSize.Regular, /*#__PURE__*/css("height:", FieldHeight.Regular, ";padding:0 8px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"))), _defineProperty(_variants4, InputSize.Regular2, /*#__PURE__*/css("height:", FieldHeight.Regular2, ";padding:0 12px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"))), _defineProperty(_variants4, InputSize.Small, function (props) {
    return /*#__PURE__*/css("height:", FieldHeight.Small, ";padding:0 ", props.shape === ButtonShape.Circular ? '10px' : '6px', ";" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _variants4)
}), ";}&:checked:hover+div{box-shadow:0 0 0 1px ", function (_ref8) {
  var theme = _ref8.theme;
  return theme.colors.primary;
}, ";background:", function (_ref9) {
  var theme = _ref9.theme;
  return theme.colors.hoverPrimaryBackground;
}, ";}&:disabled+div,&:disabled:hover+div{border:2px solid transparent;background:", function (_ref10) {
  var theme = _ref10.theme;
  return theme.colors.disabledBackground;
}, ";color:", function (_ref11) {
  var theme = _ref11.theme;
  return theme.colors.textSubtle;
}, ";}&:focus+div:after{border-color:", function (_ref12) {
  var theme = _ref12.theme;
  return theme.colors.focusOutline;
}, ";}");

export { ButtonDisplay, ButtonLabel, HiddenInput, SelectedIcon, TextWrapper };
