// @tablecheck/tablekit-button/lib/es5/styled/getButtonStyles.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.StyledLink = exports.StyledButton = exports.IconWrapper = void 0;
exports.getButtonStyles = getButtonStyles;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitTypography = require("@tablecheck/tablekit-typography");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _types = require("../types.js");

var _commonStyles = require("./commonStyles.js");

var _index = require("./themeDefinitions/index.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _extends() {
  _extends = Object.assign || function (target) {
    for (var i = 1; i < arguments.length; i++) {
      var source = arguments[i];

      for (var key in source) {
        if (Object.prototype.hasOwnProperty.call(source, key)) {
          target[key] = source[key];
        }
      }
    }

    return target;
  };

  return _extends.apply(this, arguments);
}

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var _ref = process.env.NODE_ENV === "production" ? {
  name: "2kcxui",
  styles: "padding:0;justify-content:center"
} : {
  name: "1tskxu4-sizeOverride",
  styles: "padding:0;justify-content:center;label:sizeOverride;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};

function getButtonStyles(props) {
  var _variants;

  var cursor = 'pointer';
  var width = 'auto';
  var height = (0, _commonStyles.getButtonHeight)(props);
  var size = (0, _tablekitUtils.variant)({
    prop: 'size',
    default: _types.ButtonSize.Regular,
    variants: (_variants = {}, _defineProperty(_variants, _types.ButtonSize.Regular, /*#__PURE__*/(0, _react.css)(_tablekitTypography.Typography.Heading4, ";padding:0 ", _tablekitTheme.Spacing.L5, ";vertical-align:middle;", props.isOnlyChild ? '' : "min-width: ".concat(parseInt(_tablekitTheme.FieldHeight.Regular, 10) * 3, "px"), ";& .svg-inline--fa{width:", _tablekitTheme.Spacing.L4, ";height:", _tablekitTheme.Spacing.L4, ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:size;"))), _defineProperty(_variants, _types.ButtonSize.Regular2, /*#__PURE__*/(0, _react.css)(_tablekitTypography.Typography.Heading4, ";padding:0 ", _tablekitTheme.Spacing.L5, ";vertical-align:middle;", props.isOnlyChild ? '' : "min-width: ".concat(parseInt(_tablekitTheme.FieldHeight.Regular2, 10) * 3, "px"), ";& .svg-inline--fa{width:", _tablekitTheme.Spacing.L4, ";height:", _tablekitTheme.Spacing.L4, ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:size;"))), _defineProperty(_variants, _types.ButtonSize.Small, /*#__PURE__*/(0, _react.css)(_tablekitTypography.Typography.Heading4, ";padding:0 ", _tablekitTheme.Spacing.L4, ";vertical-align:middle;", props.isOnlyChild ? '' : "min-width: ".concat(parseInt(_tablekitTheme.FieldHeight.Small, 10) * 3, "px"), ";& .svg-inline--fa{width:", _tablekitTheme.Spacing.L4, ";height:", _tablekitTheme.Spacing.L4, ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:size;"))), _defineProperty(_variants, _types.ButtonSize.Large, /*#__PURE__*/(0, _react.css)(_tablekitTypography.Typography.Body1, ";padding:0 ", _tablekitTheme.Spacing.L5, ";vertical-align:middle;", props.isOnlyChild ? '' : "min-width: ".concat(parseInt(_tablekitTheme.FieldHeight.Large, 10) * 3, "px"), ";& .svg-inline--fa{width:", _tablekitTheme.Spacing.L5, ";height:", _tablekitTheme.Spacing.L5, ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:size;"))), _variants)
  })(props);
  var sizeOverride = '';

  if (props.disabled) {
    cursor = 'not-allowed';
  } // Fit to parent width


  if (props.fit) {
    width = '100%';
  } else if (props.isOnlyChild) {
    width = _tablekitTheme.FieldHeight.Regular;

    if (props.size === _types.ButtonSize.Large) {
      width = _tablekitTheme.FieldHeight.Large;
    } else if (props.size === _types.ButtonSize.Small) {
      width = _tablekitTheme.FieldHeight.Small;
    } else if (props.size === _types.ButtonSize.Regular2) {
      width = _tablekitTheme.FieldHeight.Regular2;
    }

    sizeOverride = _ref;
  }

  return /*#__PURE__*/(0, _react.css)("text-decoration:none;border-color:transparent;border-width:", _index.borderWidth, "px;border-style:solid;align-items:baseline;box-sizing:border-box;cursor:", cursor, ";display:inline-flex;justify-content:center;margin:0;max-width:100%;outline:none;pointer-events:auto;text-align:center;white-space:nowrap;width:", width, ";height:", height, ";", size, ";", sizeOverride, ";&::-moz-focus-inner{border:0;margin:0;padding:0;}font-weight:600;", _commonStyles.transition, ";position:relative;&:before{content:'';display:block;position:absolute;background:transparent;opacity:0;top:-", _index.borderWidth, "px;right:-", _index.borderWidth, "px;bottom:-", _index.borderWidth, "px;left:-", _index.borderWidth, "px;", _commonStyles.transition, ";}&:after{content:'';display:block;position:absolute;top:-", _index.borderWidth, "px;right:-", _index.borderWidth, "px;bottom:-", _index.borderWidth, "px;left:-", _index.borderWidth, "px;", _commonStyles.transition, ";}&,&:before{border-radius:", (0, _commonStyles.getBorderRadius)(height, props.shape), ";}&:after{border-radius:", (0, _commonStyles.getOutlineRadius)(height, props.shape), ";}", (0, _tablekitUtils.variant)({
    prop: 'appearance',
    default: _types.Appearance.Solid,
    variants: _index.defaultVariants
  })(props), " &:focus::after{border:2px solid ", props.theme.colors.focusOutline, ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:getButtonStyles;"));
}

var StyledButton = /*#__PURE__*/(0, _base.default)('button', process.env.NODE_ENV === "production" ? _extends({}, {
  target: "exzh0p02"
}, _commonStyles.styledComponentOptions) : _extends({}, {
  target: "exzh0p02",
  label: "StyledButton"
}, _commonStyles.styledComponentOptions))(getButtonStyles, ";");
exports.StyledButton = StyledButton;
var StyledLink = /*#__PURE__*/(0, _base.default)('a', process.env.NODE_ENV === "production" ? _extends({}, {
  target: "exzh0p01"
}, _commonStyles.styledComponentOptions) : _extends({}, {
  target: "exzh0p01",
  label: "StyledLink"
}, _commonStyles.styledComponentOptions))(getButtonStyles, ";&,&:hover,&:active,&:visited{text-decoration:none;}");
exports.StyledLink = StyledLink;
var IconWrapper = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "exzh0p00"
} : {
  target: "exzh0p00",
  label: "IconWrapper"
})("align-items:center;justify-content:center;width:auto;height:auto;align-self:center;display:flex;flex-shrink:0;margin:0;user-select:none;", _commonStyles.getLoadingStyle, ";");
exports.IconWrapper = IconWrapper;
