// @tablecheck/tablekit-select/lib/esm/baseStyles.js
var _inputSizes;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { InputSize, INPUT_WIDTHS } from '@tablecheck/tablekit-input';
import { FieldHeight, Spacing, Size, COLORS } from '@tablecheck/tablekit-theme';
import { Typography } from '@tablecheck/tablekit-typography';
import { getThemeValue } from '@tablecheck/tablekit-utils';
import chroma from 'chroma-js/chroma-tablekit';
import { selectThemeNamespace, selectClassicTheme } from './themes.js';
var inputSizes = (_inputSizes = {}, _defineProperty(_inputSizes, InputSize.Large, {
  height: FieldHeight.Large,
  font: Typography.Heading3,
  spacing: '18px'
}), _defineProperty(_inputSizes, InputSize.Regular, {
  height: FieldHeight.Regular,
  font: Typography.Heading4,
  spacing: Spacing.L3
}), _defineProperty(_inputSizes, InputSize.Regular2, {
  height: FieldHeight.Regular2,
  font: Typography.Heading4,
  spacing: Spacing.L3
}), _defineProperty(_inputSizes, InputSize.Small, {
  height: FieldHeight.Small,
  font: Typography.Heading4,
  spacing: '10px'
}), _inputSizes);

function getSizeStyles(size) {
  return inputSizes[size] || inputSizes[InputSize.Regular];
}

function baseStyles(theme, isInvalid, isWarning, isSuccess) {
  return {
    container: function container(css, _ref) {
      var _ref$selectProps = _ref.selectProps,
          shouldFitContainer = _ref$selectProps.shouldFitContainer,
          isButtonDropDown = _ref$selectProps.isButtonDropDown,
          width = _ref$selectProps.width,
          size = _ref$selectProps.size;

      var _getSizeStyles = getSizeStyles(size),
          height = _getSizeStyles.height;

      var styles = _objectSpread(_objectSpread({}, css), {}, {
        pointerEvents: 'auto',
        height: height
      }); // default styles inherited from react-select stretch the select out


      if (!shouldFitContainer && !isButtonDropDown) {
        styles.width = width ? INPUT_WIDTHS[width] : INPUT_WIDTHS[Size.Regular];
      }

      return styles;
    },
    valueContainer: function valueContainer(css, _ref2) {
      var size = _ref2.selectProps.size;

      var _getSizeStyles2 = getSizeStyles(size),
          spacing = _getSizeStyles2.spacing;

      return _objectSpread(_objectSpread({}, css), {}, {
        padding: "0 ".concat(spacing),
        whiteSpace: 'nowrap'
      });
    },
    control: function control(css, props) {
      var _selectClassicTheme$c;

      var isFocused = props.isFocused,
          isDisabled = props.isDisabled,
          size = props.selectProps.size;
      var borderColor = getThemeValue("".concat(selectThemeNamespace, ".common.borderColor"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$c = selectClassicTheme.common) === null || _selectClassicTheme$c === void 0 ? void 0 : _selectClassicTheme$c.borderColor)({
        theme: theme
      });

      if (!isDisabled) {
        if (isFocused) borderColor = theme.colors.secondary;

        if (isInvalid) {
          borderColor = theme.colors.danger;
        } else if (isWarning) {
          borderColor = theme.colors.warning;
        } else if (isSuccess) {
          borderColor = theme.colors.success;
        }
      }

      var borderColorHover = theme.colors.primary;
      var isLargeBorder = isFocused || isInvalid || isSuccess || isWarning;
      var transitionDuration = '200ms';

      var _getSizeStyles3 = getSizeStyles(size),
          height = _getSizeStyles3.height,
          font = _getSizeStyles3.font;

      return _objectSpread(_objectSpread({}, css), {}, {
        borderColor: borderColor,
        borderStyle: 'solid',
        borderWidth: isLargeBorder ? 2 : 1,
        boxShadow: 'none',
        padding: isLargeBorder ? 0 : 1,
        minHeight: height,
        fontSize: font.fontSize,
        lineHeight: font.lineHeight,
        opacity: isDisabled ? 0.5 : 1,
        pointerEvents: isDisabled ? 'none' : 'auto',
        transition: "background-color ".concat(transitionDuration, " ease-in-out,\n        border-color ").concat(transitionDuration, " ease-in-out"),
        color: theme.colors.text,
        cursor: isDisabled ? 'not-allowed' : 'pointer',
        ':hover': {
          borderColor: borderColorHover,
          borderWidth: 2,
          padding: 0
        }
      });
    },
    dropdownIndicator: function dropdownIndicator(css, _ref3) {
      var _selectClassicTheme$d, _selectClassicTheme$d2, _selectClassicTheme$d3, _selectClassicTheme$d4, _selectClassicTheme$d5, _selectClassicTheme$d6;

      var isFocused = _ref3.isFocused,
          size = _ref3.selectProps.size;
      var defaultColor = getThemeValue("".concat(selectThemeNamespace, ".dropdownIndicator.default.color"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$d = selectClassicTheme.dropdownIndicator) === null || _selectClassicTheme$d === void 0 ? void 0 : (_selectClassicTheme$d2 = _selectClassicTheme$d.default) === null || _selectClassicTheme$d2 === void 0 ? void 0 : _selectClassicTheme$d2.color)({
        theme: theme
      });
      var focusedColor = getThemeValue("".concat(selectThemeNamespace, ".dropdownIndicator.focused.color"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$d3 = selectClassicTheme.dropdownIndicator) === null || _selectClassicTheme$d3 === void 0 ? void 0 : (_selectClassicTheme$d4 = _selectClassicTheme$d3.focused) === null || _selectClassicTheme$d4 === void 0 ? void 0 : _selectClassicTheme$d4.color)({
        theme: theme
      });
      var hoveredColor = getThemeValue("".concat(selectThemeNamespace, ".dropdownIndicator.hovered.color"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$d5 = selectClassicTheme.dropdownIndicator) === null || _selectClassicTheme$d5 === void 0 ? void 0 : (_selectClassicTheme$d6 = _selectClassicTheme$d5.hovered) === null || _selectClassicTheme$d6 === void 0 ? void 0 : _selectClassicTheme$d6.color)({
        theme: theme
      });

      var _getSizeStyles4 = getSizeStyles(size),
          spacing = _getSizeStyles4.spacing;

      return _objectSpread(_objectSpread({}, css), {}, {
        padding: "0 ".concat(spacing),
        color: isFocused ? focusedColor : defaultColor,
        ':hover': {
          color: hoveredColor
        }
      });
    },
    clearIndicator: function clearIndicator(css, _ref4) {
      var _selectClassicTheme$c2, _selectClassicTheme$c3, _selectClassicTheme$c4, _selectClassicTheme$c5, _selectClassicTheme$c6, _selectClassicTheme$c7;

      var isFocused = _ref4.isFocused;
      var defaultColor = getThemeValue("".concat(selectThemeNamespace, ".clearIndicator.default.color"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$c2 = selectClassicTheme.clearIndicator) === null || _selectClassicTheme$c2 === void 0 ? void 0 : (_selectClassicTheme$c3 = _selectClassicTheme$c2.default) === null || _selectClassicTheme$c3 === void 0 ? void 0 : _selectClassicTheme$c3.color)({
        theme: theme
      });
      var focusedColor = getThemeValue("".concat(selectThemeNamespace, ".clearIndicator.focused.color"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$c4 = selectClassicTheme.clearIndicator) === null || _selectClassicTheme$c4 === void 0 ? void 0 : (_selectClassicTheme$c5 = _selectClassicTheme$c4.focused) === null || _selectClassicTheme$c5 === void 0 ? void 0 : _selectClassicTheme$c5.color)({
        theme: theme
      });
      var hoveredColor = getThemeValue("".concat(selectThemeNamespace, ".clearIndicator.hovered.color"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$c6 = selectClassicTheme.clearIndicator) === null || _selectClassicTheme$c6 === void 0 ? void 0 : (_selectClassicTheme$c7 = _selectClassicTheme$c6.hovered) === null || _selectClassicTheme$c7 === void 0 ? void 0 : _selectClassicTheme$c7.color)({
        theme: theme
      });
      return _objectSpread(_objectSpread({}, css), {}, {
        color: isFocused ? focusedColor : defaultColor,
        ':hover': {
          color: hoveredColor
        }
      });
    },
    option: function option(css, _ref5) {
      var data = _ref5.data,
          isDisabled = _ref5.isDisabled,
          isFocused = _ref5.isFocused,
          isSelected = _ref5.isSelected,
          size = _ref5.selectProps.size;

      var _getSizeStyles5 = getSizeStyles(size),
          font = _getSizeStyles5.font;

      var styles = _objectSpread(_objectSpread({}, css), {}, {
        fontSize: font.fontSize,
        lineHeight: font.lineHeight,
        cursor: isDisabled ? 'not-allowed' : 'pointer',
        opacity: isDisabled ? 0.5 : 1
      });

      if (data.color) {
        var tagColor = data.color;

        if (isSelected) {
          return _objectSpread(_objectSpread({}, styles), {}, {
            backgroundColor: chroma.mix(tagColor, COLORS.PURPLE.L1, 0.6).css(),
            color: theme.colors.text
          });
        }

        if (isFocused) {
          var activeBackgroundChroma = chroma(tagColor).brighten(0.4).alpha(0.4);
          var activeColor = chroma(tagColor);
          return _objectSpread(_objectSpread({}, styles), {}, {
            backgroundColor: activeBackgroundChroma.css(),
            color: activeColor
          });
        }

        return _objectSpread(_objectSpread({}, styles), {}, {
          color: tagColor
        });
      }

      var backgroundColor;
      var boxShadow;

      if (isSelected) {
        var _selectClassicTheme$o, _selectClassicTheme$o2, _selectClassicTheme$o3, _selectClassicTheme$o4;

        backgroundColor = getThemeValue("".concat(selectThemeNamespace, ".option.selected.backgroundColor"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$o = selectClassicTheme.option) === null || _selectClassicTheme$o === void 0 ? void 0 : (_selectClassicTheme$o2 = _selectClassicTheme$o.selected) === null || _selectClassicTheme$o2 === void 0 ? void 0 : _selectClassicTheme$o2.backgroundColor)({
          theme: theme
        });
        var borderColor = getThemeValue("".concat(selectThemeNamespace, ".option.selected.borderColor"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$o3 = selectClassicTheme.option) === null || _selectClassicTheme$o3 === void 0 ? void 0 : (_selectClassicTheme$o4 = _selectClassicTheme$o3.selected) === null || _selectClassicTheme$o4 === void 0 ? void 0 : _selectClassicTheme$o4.borderColor)({
          theme: theme
        });
        boxShadow = "inset 3px 0 0 -1px ".concat(borderColor);
      } else if (isFocused && !isDisabled) {
        var _selectClassicTheme$o5, _selectClassicTheme$o6;

        backgroundColor = getThemeValue("".concat(selectThemeNamespace, ".option.hovered.backgroundColor"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$o5 = selectClassicTheme.option) === null || _selectClassicTheme$o5 === void 0 ? void 0 : (_selectClassicTheme$o6 = _selectClassicTheme$o5.hovered) === null || _selectClassicTheme$o6 === void 0 ? void 0 : _selectClassicTheme$o6.backgroundColor)({
          theme: theme
        });
      }

      if (!isDisabled) {
        var _selectClassicTheme$o7, _selectClassicTheme$o8;

        // override the active state not to show primary background color on :selected:active
        var activeBackgroundColor = getThemeValue("".concat(selectThemeNamespace, ".option.active.backgroundColor"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$o7 = selectClassicTheme.option) === null || _selectClassicTheme$o7 === void 0 ? void 0 : (_selectClassicTheme$o8 = _selectClassicTheme$o7.active) === null || _selectClassicTheme$o8 === void 0 ? void 0 : _selectClassicTheme$o8.backgroundColor)({
          theme: theme
        });
        styles[':active'] = styles[':active'] || {};
        styles[':active'].backgroundColor = isSelected ? backgroundColor : activeBackgroundColor;
      }

      return _objectSpread(_objectSpread({}, styles), {}, {
        backgroundColor: backgroundColor,
        color: theme.colors.text,
        boxShadow: boxShadow
      });
    },
    placeholder: function placeholder(css, _ref6) {
      var size = _ref6.selectProps.size;

      var _getSizeStyles6 = getSizeStyles(size),
          spacing = _getSizeStyles6.spacing;

      return _objectSpread(_objectSpread({}, css), {}, {
        maxWidth: "calc(100% - ".concat(spacing, ")"),
        color: theme.colors.text
      });
    },
    singleValue: function singleValue(css, _ref7) {
      var size = _ref7.selectProps.size;

      var _getSizeStyles7 = getSizeStyles(size),
          spacing = _getSizeStyles7.spacing;

      return _objectSpread(_objectSpread({}, css), {}, {
        maxWidth: "calc(100% - ".concat(spacing, ")"),
        color: theme.colors.text
      });
    },
    menu: function menu(css, _ref8) {
      var _ref8$selectProps = _ref8.selectProps,
          isButtonDropDown = _ref8$selectProps.isButtonDropDown,
          width = _ref8$selectProps.width;

      var styles = _objectSpread(_objectSpread({}, css), {}, {
        marginTop: '1px',
        color: theme.colors.text
      });

      if (isButtonDropDown) {
        styles.width = width ? INPUT_WIDTHS[width] : INPUT_WIDTHS[Size.Regular];
      }

      return styles;
    },
    multiValueRemove: function multiValueRemove(css) {
      return _objectSpread(_objectSpread({}, css), {}, {
        ':hover': {
          backgroundColor: theme.colors.hoverNeutralBackground,
          color: theme.colors.text
        }
      });
    },
    multiValueLabel: function multiValueLabel(css) {
      return _objectSpread(_objectSpread({}, css), Typography.Heading4);
    }
  };
}

export { baseStyles };
