// @tablecheck/tablekit-select/lib/esm/SelectWrapper.js
import { Label, InputMessage } from '@tablecheck/tablekit-input';
import { cloneElement } from 'react';
import { jsxs, jsx } from '@emotion/react/jsx-runtime';

var SelectWrapper = function SelectWrapper(_ref) {
  var children = _ref.children,
      className = _ref.className,
      isRequired = _ref.isRequired,
      isInvalid = _ref.isInvalid,
      isMessageHidden = _ref.isMessageHidden,
      isWarning = _ref.isWarning,
      isSuccess = _ref.isSuccess,
      label = _ref.label,
      id = _ref.id,
      message = _ref.message;
  return jsxs("div", {
    className: className,
    children: [label !== undefined ? jsx(Label, {
      htmlFor: "react-select-".concat(id, "--input"),
      isRequired: isRequired,
      label: label || ''
    }) : null, /*#__PURE__*/cloneElement(children, {
      instanceId: id,
      isInvalid: isInvalid,
      isWarning: isWarning,
      isSuccess: isSuccess
    }), !isMessageHidden && jsx(InputMessage, {
      children: message
    })]
  });
};

export { SelectWrapper };
