// @tablecheck/tablekit-select/lib/esm/styled.js
var _selectClassicTheme$i, _variants;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import _styled from '@emotion/styled/base';
import { css } from '@emotion/react';
import { InputSize } from '@tablecheck/tablekit-input';
import { Spacing } from '@tablecheck/tablekit-theme';
import { getThemeValue, variant } from '@tablecheck/tablekit-utils';
import { selectThemeNamespace, selectClassicTheme } from './themes.js';

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var IconContainer = _styled("div", process.env.NODE_ENV === "production" ? {
  target: "e153nee80"
} : {
  target: "e153nee80",
  label: "IconContainer"
})("display:flex;align-items:center;color:", getThemeValue("".concat(selectThemeNamespace, ".icon.color"), selectClassicTheme === null || selectClassicTheme === void 0 ? void 0 : (_selectClassicTheme$i = selectClassicTheme.icon) === null || _selectClassicTheme$i === void 0 ? void 0 : _selectClassicTheme$i.color), ";", variant({
  prop: 'size',
  default: InputSize.Regular,
  variants: (_variants = {}, _defineProperty(_variants, InputSize.Large, process.env.NODE_ENV === "production" ? {
    name: "f9bxyi",
    styles: "padding-left:18px"
  } : {
    name: "tdfrwo-IconContainer",
    styles: "padding-left:18px;label:IconContainer;",
    toString: _EMOTION_STRINGIFIED_CSS_ERROR__
  }), _defineProperty(_variants, InputSize.Regular, /*#__PURE__*/css("padding-left:", Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:IconContainer;"))), _defineProperty(_variants, InputSize.Regular2, /*#__PURE__*/css("padding-left:", Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:IconContainer;"))), _defineProperty(_variants, InputSize.Small, process.env.NODE_ENV === "production" ? {
    name: "vnst0l",
    styles: "padding-left:10px"
  } : {
    name: "11cmjzs-IconContainer",
    styles: "padding-left:10px;label:IconContainer;",
    toString: _EMOTION_STRINGIFIED_CSS_ERROR__
  }), _variants)
}), ";");

export { IconContainer };
