// @tablecheck/tablekit-typography/lib/es5/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Link", {
  enumerable: true,
  get: function get() {
    return _Link.Link;
  }
});
Object.defineProperty(exports, "Typography", {
  enumerable: true,
  get: function get() {
    return _Typography.Typography;
  }
});
Object.defineProperty(exports, "TypographyHeadlineDesktop", {
  enumerable: true,
  get: function get() {
    return _Typography.TypographyHeadlineDesktop;
  }
});
Object.defineProperty(exports, "TypographyHeadlineMobile", {
  enumerable: true,
  get: function get() {
    return _Typography.TypographyHeadlineMobile;
  }
});
Object.defineProperty(exports, "TypographySize", {
  enumerable: true,
  get: function get() {
    return _Typography.TypographySize;
  }
});
Object.defineProperty(exports, "commonTypographyStyles", {
  enumerable: true,
  get: function get() {
    return _Typography.commonTypographyStyles;
  }
});
Object.defineProperty(exports, "desktopTypographyStyles", {
  enumerable: true,
  get: function get() {
    return _Typography.desktopTypographyStyles;
  }
});
Object.defineProperty(exports, "getLinkStyles", {
  enumerable: true,
  get: function get() {
    return _styled.getLinkStyles;
  }
});
Object.defineProperty(exports, "mobileTypographyStyles", {
  enumerable: true,
  get: function get() {
    return _Typography.mobileTypographyStyles;
  }
});

require("@tablecheck/tablekit-theme");

var _Typography = require("./Typography.js");

var _Link = require("./Link.js");

var _styled = require("./styled.js");
