// @tablecheck/tablekit-item/lib/es5/utils/theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.interactionStates = exports.getSizeStyle = exports.focusOutlineRadius = exports.borderWidth = exports.ElementBefore = exports.ElementAfter = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _themes = require("../themes.js");

var _types = require("../types.js");

var _variants;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var borderWidth = '2px';
exports.borderWidth = borderWidth;
var focusOutlineRadius = '6px'; // needs to be here to prevent circular dependencies

exports.focusOutlineRadius = focusOutlineRadius;
var BeforeAfterBase = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "eytf44w2"
} : {
  target: "eytf44w2",
  label: "BeforeAfterBase"
})(process.env.NODE_ENV === "production" ? {
  name: "12o7d5c",
  styles: "display:flex;justify-content:center;align-items:center;flex-shrink:0"
} : {
  name: "12o7d5c",
  styles: "display:flex;justify-content:center;align-items:center;flex-shrink:0",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});
var ElementBefore = /*#__PURE__*/(0, _base.default)(BeforeAfterBase, process.env.NODE_ENV === "production" ? {
  target: "eytf44w1"
} : {
  target: "eytf44w1",
  label: "ElementBefore"
})(function (_ref) {
  var hasOnlyIcon = _ref.hasOnlyIcon,
      theme = _ref.theme;
  return hasOnlyIcon ? '' : (0, _tablekitUtils.margin)({
    right: _tablekitTheme.Spacing.L2
  })({
    theme: theme
  });
}, ";");
exports.ElementBefore = ElementBefore;
var ElementAfter = /*#__PURE__*/(0, _base.default)(BeforeAfterBase, process.env.NODE_ENV === "production" ? {
  target: "eytf44w0"
} : {
  target: "eytf44w0",
  label: "ElementAfter"
})(function (_ref2) {
  var hasOnlyIcon = _ref2.hasOnlyIcon,
      theme = _ref2.theme;
  return hasOnlyIcon ? '' : (0, _tablekitUtils.margin)({
    left: _tablekitTheme.Spacing.L2
  })({
    theme: theme
  });
}, ";");
exports.ElementAfter = ElementAfter;
var interactionStates = (0, _tablekitUtils.variant)({
  prop: 'interactionState',
  key: "".concat(_themes.itemThemeNamespace, ".interactionStates"),
  default: _types.ItemInteractionState.Default,
  variants: {
    default: function _default(props) {
      return /*#__PURE__*/(0, _react.css)("display:flex;background-color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".default.background"), _themes.itemClassicTheme.default.background)(props), ";color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".default.textColor"), _themes.itemClassicTheme.default.textColor)(props), ";& ", ElementBefore, ",& ", ElementAfter, "{color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".default.beforeAfterColor"), _themes.itemClassicTheme.default.beforeAfterColor)(props), ";}&:focus{box-shadow:0 0 0 2px ", props.theme.colors.focusOutline, " inset;border-radius:", focusOutlineRadius, ";}&:hover{background-color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".default.hoverBackground"), _themes.itemClassicTheme.default.hoverBackground)(props), ";color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".default.hoverTextColor"), _themes.itemClassicTheme.default.hoverTextColor)(props), ";& ", ElementBefore, ",& ", ElementAfter, "{color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".default.beforeAfterColor"), _themes.itemClassicTheme.default.beforeAfterColor)(props), ";}}" + (process.env.NODE_ENV === "production" ? "" : ";label:default;"));
    },
    hidden: process.env.NODE_ENV === "production" ? {
      name: "eivff4",
      styles: "display:none"
    } : {
      name: "1mkqgn7-hidden",
      styles: "display:none;label:hidden;",
      toString: _EMOTION_STRINGIFIED_CSS_ERROR__
    },
    selected: function selected(upperProps) {
      return /*#__PURE__*/(0, _react.css)("display:flex;background-color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".selected.background"), _themes.itemClassicTheme.selected.background)(upperProps), ";color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".selected.textColor"), _themes.itemClassicTheme.selected.textColor)(upperProps), ";& ", ElementBefore, ",& ", ElementAfter, "{color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".selected.beforeAfterColor"), _themes.itemClassicTheme.selected.beforeAfterColor)(upperProps), ";}&:focus{box-shadow:0 0 0 2px ", upperProps.theme.colors.focusOutline, " inset;border-radius:", focusOutlineRadius, ";}", (0, _tablekitUtils.variant)({
        prop: 'borderPosition',
        default: _types.ItemBorderPosition.Left,
        variants: {
          left: function left(props) {
            return /*#__PURE__*/(0, _react.css)("box-shadow:inset ", borderWidth, " 0px ", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".selected.borderColor"), _themes.itemClassicTheme.selected.borderColor)(props), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:left;"));
          },
          right: function right(props) {
            return /*#__PURE__*/(0, _react.css)("box-shadow:inset -", borderWidth, " 0px ", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".selected.borderColor"), _themes.itemClassicTheme.selected.borderColor)(props), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:right;"));
          },
          top: function top(props) {
            return /*#__PURE__*/(0, _react.css)("box-shadow:inset 0px ", borderWidth, " ", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".selected.borderColor"), _themes.itemClassicTheme.selected.borderColor)(props), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:top;"));
          },
          bottom: function bottom(props) {
            return /*#__PURE__*/(0, _react.css)("box-shadow:inset 0px -", borderWidth, " ", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".selected.borderColor"), _themes.itemClassicTheme.selected.borderColor)(props), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:bottom;"));
          }
        }
      })(upperProps), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:selected;"));
    },
    disabled: function disabled(props) {
      return /*#__PURE__*/(0, _react.css)("display:flex;opacity:0.5;color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".disabled.textColor"), _themes.itemClassicTheme.disabled.textColor)(props), ";& ", ElementBefore, ",& ", ElementAfter, "{color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".disabled.beforeAfterColor"), _themes.itemClassicTheme.disabled.beforeAfterColor)(props), ";}cursor:not-allowed;&:hover{color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".disabled.textColor"), _themes.itemClassicTheme.disabled.textColor)(props), ";& ", ElementBefore, ",& ", ElementAfter, "{color:", (0, _tablekitUtils.getThemeValue)("".concat(_themes.itemThemeNamespace, ".disabled.beforeAfterColor"), _themes.itemClassicTheme.disabled.beforeAfterColor)(props), ";}}" + (process.env.NODE_ENV === "production" ? "" : ";label:disabled;"));
    }
  }
});
exports.interactionStates = interactionStates;
var getSizeStyle = (0, _tablekitUtils.variant)({
  prop: 'size',
  default: _tablekitTheme.Size.Regular,
  variants: (_variants = {}, _defineProperty(_variants, _tablekitTheme.Size.Regular, function (props) {
    return /*#__PURE__*/(0, _react.css)("height:", (0, _tablekitUtils.mediaQuery)('height', function (value) {
      return value || _tablekitTheme.FieldHeight.Regular;
    })(props), ";min-height:", _tablekitTheme.FieldHeight.Regular, ";padding:0 ", _tablekitTheme.Spacing.L3, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getSizeStyle;"));
  }), _defineProperty(_variants, _tablekitTheme.Size.Large, function (props) {
    return /*#__PURE__*/(0, _react.css)("height:", (0, _tablekitUtils.mediaQuery)('height', function (value) {
      return value || _tablekitTheme.FieldHeight.Large;
    })(props), ";min-height:", _tablekitTheme.FieldHeight.Large, ";padding:0 ", _tablekitTheme.Spacing.L4, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getSizeStyle;"));
  }), _variants)
});
exports.getSizeStyle = getSizeStyle;
