// @tablecheck/tablekit-package-namespace/lib/es5/PackageNamespace.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.PackageNamespace = void 0;
// This file is auto built via 'npm run update-package-namespace', do not edit manually
var PackageNamespace;
exports.PackageNamespace = PackageNamespace;

(function (PackageNamespace) {
  PackageNamespace["Alert"] = "alert";
  PackageNamespace["Avatar"] = "avatar";
  PackageNamespace["Badge"] = "badge";
  PackageNamespace["Banner"] = "banner";
  PackageNamespace["ButtonGroup"] = "button-group";
  PackageNamespace["Button"] = "button";
  PackageNamespace["Calendar"] = "calendar";
  PackageNamespace["Checkbox"] = "checkbox";
  PackageNamespace["Enzyme"] = "enzyme";
  PackageNamespace["FreeIconConfig"] = "free-icon-config";
  PackageNamespace["Icon"] = "icon";
  PackageNamespace["InlineDialog"] = "inline-dialog";
  PackageNamespace["InputButton"] = "input-button";
  PackageNamespace["Input"] = "input";
  PackageNamespace["Item"] = "item";
  PackageNamespace["LanguageSelector"] = "language-selector";
  PackageNamespace["Logo"] = "logo";
  PackageNamespace["ModalDialog"] = "modal-dialog";
  PackageNamespace["PackageNamespace"] = "package-namespace";
  PackageNamespace["Panel"] = "panel";
  PackageNamespace["PasswordField"] = "password-field";
  PackageNamespace["PhoneInput"] = "phone-input";
  PackageNamespace["ProIconConfig"] = "pro-icon-config";
  PackageNamespace["Radio"] = "radio";
  PackageNamespace["Select"] = "select";
  PackageNamespace["Sidenav"] = "sidenav";
  PackageNamespace["Spinner"] = "spinner";
  PackageNamespace["Table"] = "table";
  PackageNamespace["Tabs"] = "tabs";
  PackageNamespace["Tag"] = "tag";
  PackageNamespace["Textarea"] = "textarea";
  PackageNamespace["Toggle"] = "toggle";
  PackageNamespace["Tooltip"] = "tooltip";
  PackageNamespace["Topnav"] = "topnav";
  PackageNamespace["Typography"] = "typography";
})(PackageNamespace || (exports.PackageNamespace = PackageNamespace = {}));
