// @tablecheck/tablekit-item/lib/esm/ItemGroup.js
import { forwardRef } from 'react';
import { TitleContainer, TitleText, ItemGroupContainer, ItemGroupContent } from './styled/ItemGroup.js';
import { ItemGroupOrientation } from './types.js';
import { jsx, jsxs } from '@emotion/react/jsx-runtime';
var ItemGroup = /*#__PURE__*/forwardRef(function (_ref, ref) {
  var children = _ref.children,
      _ref$role = _ref.role,
      role = _ref$role === void 0 ? 'group' : _ref$role,
      _ref$orientation = _ref.orientation,
      orientation = _ref$orientation === void 0 ? ItemGroupOrientation.Vertical : _ref$orientation,
      _ref$hasIndent = _ref.hasIndent,
      hasIndent = _ref$hasIndent === void 0 ? false : _ref$hasIndent,
      headingComponent = _ref.headingComponent,
      headingText = _ref.headingText,
      headingStyles = _ref.headingStyles,
      minWidth = _ref.minWidth,
      className = _ref.className,
      isUppercase = _ref.isUppercase,
      shouldWrap = _ref.shouldWrap;
  var heading = headingComponent || null;

  if (headingText) {
    heading = jsx(TitleContainer, {
      children: jsx(TitleText, {
        headingStyles: headingStyles,
        isUppercase: isUppercase,
        children: headingText
      })
    });
  }

  return jsxs(ItemGroupContainer, {
    role: role,
    minWidth: minWidth,
    className: className,
    ref: ref,
    children: [heading, jsx("div", {
      children: jsx(ItemGroupContent, {
        orientation: orientation,
        hasIndent: hasIndent,
        shouldWrap: shouldWrap,
        children: children
      })
    })]
  });
});
export { ItemGroup };
