// @tablecheck/tablekit-button/lib/es5/Button.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ComponentButton = exports.Button = exports.AnchorButton = void 0;
exports.InternalButton = InternalButton;

var _react = require("@emotion/react");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var React = _interopRequireWildcard(require("react"));

var _ButtonContent = require("./styled/ButtonContent.js");

var _ButtonWrapper = require("./styled/ButtonWrapper.js");

var _CircleWrapper = require("./styled/CircleWrapper.js");

var _LoadingSpinner = require("./styled/LoadingSpinner.js");

var _getButtonStyles = require("./styled/getButtonStyles.js");

var _getButtonProps2 = require("./utils/getButtonProps.js");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

var _excluded = ["iconBefore", "iconAfter"],
    _excluded2 = ["component"];

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function renderIcon(icon, props) {
  if (typeof icon === 'function') {
    return icon(props);
  }

  var size = props.size || _tablekitTheme.Size.Regular;
  return /*#__PURE__*/React.cloneElement(icon, {
    size: size
  });
}

function InternalButton(props) {
  var children = props.children,
      isOnlyChild = props.isOnlyChild,
      _props$isLoading = props.isLoading,
      isLoading = _props$isLoading === void 0 ? false : _props$isLoading,
      shouldFitContainer = props.fit;
  var theme = (0, _react.useTheme)();

  var _getButtonProps = (0, _getButtonProps2.getButtonProps)(props, theme),
      iconBefore = _getButtonProps.iconBefore,
      iconAfter = _getButtonProps.iconAfter,
      buttonProps = _objectWithoutProperties(_getButtonProps, _excluded);

  if ((0, _getButtonProps2.isVerticalButton)(props)) {
    return (0, _jsxRuntime.jsxs)(_ButtonWrapper.ButtonWrapper, {
      fit: shouldFitContainer,
      isVertical: true,
      children: [isLoading ? (0, _jsxRuntime.jsx)(_LoadingSpinner.LoadingSpinner, {
        size: buttonProps.size
      }) : null, iconBefore && (isOnlyChild ? (0, _jsxRuntime.jsx)(_getButtonStyles.IconWrapper, {
        isLoading: isLoading,
        children: renderIcon(iconBefore, props)
      }) : (0, _jsxRuntime.jsx)(_CircleWrapper.CircleWrapper, _objectSpread(_objectSpread({
        isLoading: isLoading
      }, buttonProps), {}, {
        children: renderIcon(iconBefore, props)
      }))), children ? (0, _jsxRuntime.jsx)(_ButtonContent.ButtonContent, {
        hasIconBefore: !!iconBefore,
        isVertical: true,
        isLoading: !!isLoading,
        children: children
      }) : null]
    });
  }

  return (0, _jsxRuntime.jsxs)(_ButtonWrapper.ButtonWrapper, {
    fit: shouldFitContainer,
    children: [isLoading ? (0, _jsxRuntime.jsx)(_LoadingSpinner.LoadingSpinner, {
      size: buttonProps.size
    }) : null, iconBefore && (0, _jsxRuntime.jsx)(_getButtonStyles.IconWrapper, {
      isLoading: isLoading,
      children: renderIcon(iconBefore, props)
    }), children ? (0, _jsxRuntime.jsx)(_ButtonContent.ButtonContent, {
      hasIconAfter: !!iconAfter,
      hasIconBefore: !!iconBefore,
      isLoading: !!isLoading,
      followsIcon: !!iconBefore,
      children: children
    }) : null, iconAfter && (0, _jsxRuntime.jsx)(_getButtonStyles.IconWrapper, {
      isLoading: isLoading,
      children: renderIcon(iconAfter, props)
    })]
  });
}
/** @component */


var AnchorButton = /*#__PURE__*/React.forwardRef(function (props, innerRef) {
  var theme = (0, _react.useTheme)();
  var derivedProps = (0, _getButtonProps2.getButtonProps)(props, theme);
  return (0, _jsxRuntime.jsx)(_getButtonStyles.StyledLink, _objectSpread(_objectSpread({
    ref: innerRef
  }, derivedProps), {}, {
    onClick: function onClick(e) {
      if (derivedProps.isDisabled || derivedProps.isLoading) {
        e.stopPropagation();
      }

      if (derivedProps.onClick) derivedProps.onClick(e);
    },
    children: (0, _jsxRuntime.jsx)(InternalButton, _objectSpread({}, derivedProps))
  }));
});
exports.AnchorButton = AnchorButton;

AnchorButton.withProps = function withProps(staticProps) {
  // we have to disable typescript checking here or we risk JS Heap overflows on compliling
  // the function signature is correct so this should be fine as it's just a wrapper
  // eslint-disable-next-line @typescript-eslint/naming-convention
  var UntypedButton = AnchorButton;
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    return (0, _jsxRuntime.jsx)(UntypedButton, _objectSpread(_objectSpread(_objectSpread({}, props), staticProps), {}, {
      ref: ref
    }));
  });
};
/** @component */


var Button = /*#__PURE__*/React.forwardRef(function (props, innerRef) {
  var theme = (0, _react.useTheme)();
  var derivedProps = (0, _getButtonProps2.getButtonProps)(props, theme);
  return (0, _jsxRuntime.jsx)(_getButtonStyles.StyledButton, _objectSpread(_objectSpread({
    ref: innerRef
  }, derivedProps), {}, {
    onClick: function onClick(e) {
      if (derivedProps.isDisabled || derivedProps.isLoading) {
        e.stopPropagation();
      }

      if (derivedProps.onClick) derivedProps.onClick(e);
    },
    children: (0, _jsxRuntime.jsx)(InternalButton, _objectSpread({}, derivedProps))
  }));
});
exports.Button = Button;

Button.withProps = function withProps(staticProps) {
  // we have to disable typescript checking here or we risk JS Heap overflows on compliling
  // the function signature is correct so this should be fine as it's just a wrapper
  // eslint-disable-next-line @typescript-eslint/naming-convention
  var UntypedButton = Button;
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    return (0, _jsxRuntime.jsx)(UntypedButton, _objectSpread(_objectSpread(_objectSpread({}, props), staticProps), {}, {
      ref: ref
    }));
  });
};

var ComponentButtonInner = function ComponentButtonInner(_ref, innerRef) {
  var component = _ref.component,
      props = _objectWithoutProperties(_ref, _excluded2);

  var theme = (0, _react.useTheme)();
  var derivedProps = (0, _getButtonProps2.getButtonProps)(props, theme);
  var ResolvedComponent = React.useMemo(function () {
    return component ? _getButtonStyles.StyledButton.withComponent(component, process.env.NODE_ENV === "production" ? {
      target: "ex3kjbp0"
    } : {
      target: "ex3kjbp0",
      label: "ResolvedComponent"
    }) : _getButtonStyles.StyledButton;
  }, [component]);
  return (0, _jsxRuntime.jsx)(ResolvedComponent, _objectSpread(_objectSpread({
    ref: innerRef
  }, derivedProps), {}, {
    onClick: function onClick(e) {
      if (derivedProps.isDisabled || derivedProps.isLoading) {
        e.stopPropagation();
      }

      if (derivedProps.onClick) derivedProps.onClick(e);
    },
    children: (0, _jsxRuntime.jsx)(InternalButton, _objectSpread({}, derivedProps))
  }));
};
/** @component */


var ComponentButton = /*#__PURE__*/React.forwardRef(ComponentButtonInner);
exports.ComponentButton = ComponentButton;

ComponentButton.withComponent = function withComponent( // eslint-disable-next-line @typescript-eslint/naming-convention
NewComponent) {
  var options = arguments.length > 1 && arguments[1] !== undefined ? arguments[1] : {};
  var _options$staticProps = options.staticProps,
      staticProps = _options$staticProps === void 0 ? {} : _options$staticProps,
      propFilter = options.propFilter;
  var filteredComponent = NewComponent;

  if (propFilter) {
    filteredComponent = function filteredComponent(unfilteredProps) {
      return (0, _jsxRuntime.jsx)(NewComponent, _objectSpread({}, Object.keys(unfilteredProps).reduce(function (filteredProps, propKey) {
        if (propKey !== 'ref' && propKey !== 'component' && propFilter(propKey)) {
          return _objectSpread(_objectSpread({}, filteredProps), {}, _defineProperty({}, propKey, unfilteredProps[propKey]));
        }

        return filteredProps; // eslint-disable-next-line @typescript-eslint/no-explicit-any
      }, {})));
    };
  } // we have to disable typescript checking here or we risk JS Heap overflows on compliling
  // the function signature is correct so this should be fine as it's just a wrapper
  // eslint-disable-next-line @typescript-eslint/naming-convention


  var UntypedButton = ComponentButton;
  return /*#__PURE__*/React.forwardRef(function (props, ref) {
    return (0, _jsxRuntime.jsx)(UntypedButton, _objectSpread(_objectSpread(_objectSpread({}, props), staticProps), {}, {
      ref: ref,
      component: filteredComponent
    }));
  });
};
