// @tablecheck/tablekit-button/lib/es5/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AnchorButton", {
  enumerable: true,
  get: function get() {
    return _Button.AnchorButton;
  }
});
Object.defineProperty(exports, "Appearance", {
  enumerable: true,
  get: function get() {
    return _types.Appearance;
  }
});
Object.defineProperty(exports, "Button", {
  enumerable: true,
  get: function get() {
    return _Button.Button;
  }
});
Object.defineProperty(exports, "ButtonAppearance", {
  enumerable: true,
  get: function get() {
    return _types.Appearance;
  }
});
Object.defineProperty(exports, "ButtonColor", {
  enumerable: true,
  get: function get() {
    return _types.Color;
  }
});
Object.defineProperty(exports, "ButtonShape", {
  enumerable: true,
  get: function get() {
    return _types.ButtonShape;
  }
});
Object.defineProperty(exports, "ButtonSize", {
  enumerable: true,
  get: function get() {
    return _types.ButtonSize;
  }
});
Object.defineProperty(exports, "Color", {
  enumerable: true,
  get: function get() {
    return _types.Color;
  }
});
Object.defineProperty(exports, "ComponentButton", {
  enumerable: true,
  get: function get() {
    return _Button.ComponentButton;
  }
});
Object.defineProperty(exports, "InternalButton", {
  enumerable: true,
  get: function get() {
    return _Button.InternalButton;
  }
});
Object.defineProperty(exports, "buttonClassicTheme", {
  enumerable: true,
  get: function get() {
    return _themes.buttonClassicTheme;
  }
});
Object.defineProperty(exports, "buttonDarkTheme", {
  enumerable: true,
  get: function get() {
    return _themes.buttonDarkTheme;
  }
});
Object.defineProperty(exports, "buttonThemeNamespace", {
  enumerable: true,
  get: function get() {
    return _themes.buttonThemeNamespace;
  }
});

var _Button = require("./Button.js");

var _types = require("./types.js");

var _themes = require("./themes.js");
