// @tablecheck/tablekit-button/lib/es5/themes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.buttonThemeNamespace = exports.buttonDarkTheme = exports.buttonClassicTheme = void 0;

var _tablekitPackageNamespace = require("@tablecheck/tablekit-package-namespace");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _types = require("./types.js");

var _buttonClassicTheme, _objectSpread2;

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var buttonThemeNamespace = _tablekitPackageNamespace.PackageNamespace.Button;
exports.buttonThemeNamespace = buttonThemeNamespace;
var buttonClassicTheme = (_buttonClassicTheme = {}, _defineProperty(_buttonClassicTheme, _types.Color.Primary, {
  main: function main(_ref) {
    var theme = _ref.theme;
    return theme.colors.primary;
  },
  secondary: function secondary(_ref2) {
    var theme = _ref2.theme;
    return theme.colors.primary;
  },
  hover: function hover(_ref3) {
    var theme = _ref3.theme;
    return theme.colors.primaryHover;
  },
  text: function text(_ref4) {
    var theme = _ref4.theme;
    return theme.colors.invertedText;
  },
  active: function active(_ref5) {
    var theme = _ref5.theme;
    return theme.colors.primary;
  },
  hoverTransparent: function hoverTransparent(_ref6) {
    var theme = _ref6.theme;
    return theme.colors.hoverPrimaryBackground;
  },
  activeTransparent: function activeTransparent(_ref7) {
    var theme = _ref7.theme;
    return theme.colors.primarySelected;
  }
}), _defineProperty(_buttonClassicTheme, _types.Color.Success, {
  main: function main(_ref8) {
    var theme = _ref8.theme;
    return theme.colors.success;
  },
  text: 'white',
  hover: _tablekitTheme.COLORS.GREEN.L4,
  active: _tablekitTheme.COLORS.GREEN.L6,
  hoverTransparent: (0, _tablekitUtils.hexToRgba)(_tablekitTheme.COLORS.GREEN.L4, 0.1),
  activeTransparent: (0, _tablekitUtils.hexToRgba)(_tablekitTheme.COLORS.GREEN.L4, 0.2)
}), _defineProperty(_buttonClassicTheme, _types.Color.Warning, {
  main: function main(_ref9) {
    var theme = _ref9.theme;
    return theme.colors.warning;
  },
  text: 'white',
  hover: _tablekitTheme.COLORS.YELLOW.L5,
  active: _tablekitTheme.COLORS.YELLOW.L7,
  hoverTransparent: (0, _tablekitUtils.hexToRgba)(_tablekitTheme.COLORS.YELLOW.L5, 0.1),
  activeTransparent: (0, _tablekitUtils.hexToRgba)(_tablekitTheme.COLORS.YELLOW.L5, 0.2)
}), _defineProperty(_buttonClassicTheme, _types.Color.Danger, {
  main: function main(_ref10) {
    var theme = _ref10.theme;
    return theme.colors.danger;
  },
  text: 'white',
  hover: _tablekitTheme.COLORS.RED.L4,
  active: _tablekitTheme.COLORS.RED.L6,
  hoverTransparent: (0, _tablekitUtils.hexToRgba)(_tablekitTheme.COLORS.RED.L4, 0.1),
  activeTransparent: (0, _tablekitUtils.hexToRgba)(_tablekitTheme.COLORS.RED.L4, 0.2)
}), _defineProperty(_buttonClassicTheme, _types.Color.Ghost, {
  main: _tablekitTheme.COLORS.GRAY.L7,
  text: 'black',
  hover: _tablekitTheme.COLORS.GRAY.L5,
  active: _tablekitTheme.COLORS.GRAY.L6,
  hoverTransparent: _tablekitTheme.COLORS.GRAY_TRANSLUCENT.L3,
  activeTransparent: _tablekitTheme.COLORS.GRAY_TRANSLUCENT.L4
}), _buttonClassicTheme);
exports.buttonClassicTheme = buttonClassicTheme;

var buttonDarkTheme = _objectSpread(_objectSpread({}, buttonClassicTheme), {}, (_objectSpread2 = {}, _defineProperty(_objectSpread2, _types.Color.Primary, _objectSpread(_objectSpread({}, buttonClassicTheme[_types.Color.Primary]), {}, {
  secondary: function secondary(_ref11) {
    var theme = _ref11.theme;
    return theme.colors.primary2;
  },
  text: 'white'
})), _defineProperty(_objectSpread2, _types.Color.Ghost, {
  main: _tablekitTheme.COLORS.WHITE_TRANSLUCENT.L7,
  text: 'white',
  active: _tablekitTheme.COLORS.WHITE_TRANSLUCENT.L4,
  hover: _tablekitTheme.COLORS.WHITE_TRANSLUCENT.L4
}), _objectSpread2));

exports.buttonDarkTheme = buttonDarkTheme;
