// @tablecheck/tablekit-button/lib/es5/types.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Color = exports.ButtonSize = exports.ButtonShape = exports.Appearance = void 0;
var Appearance;
exports.Appearance = Appearance;

(function (Appearance) {
  Appearance["Disabled"] = "disabled";
  Appearance["Subtle"] = "subtle";
  Appearance["Solid"] = "solid";
  Appearance["Outline"] = "outline";
  Appearance["Inverse"] = "inverse";
  Appearance["SubtleOutline"] = "subtle-outline";
  Appearance["Vertical"] = "vertical";
  Appearance["DisabledVertical"] = "disabled-vertical";
})(Appearance || (exports.Appearance = Appearance = {}));

var Color;
exports.Color = Color;

(function (Color) {
  Color["Primary"] = "primary";
  Color["Ghost"] = "ghost";
  Color["Success"] = "success";
  Color["Warning"] = "warning";
  Color["Danger"] = "danger";
})(Color || (exports.Color = Color = {}));

var ButtonSize;
exports.ButtonSize = ButtonSize;

(function (ButtonSize) {
  ButtonSize["Small"] = "small";
  ButtonSize["Regular"] = "regular";
  ButtonSize["Regular2"] = "regular2";
  ButtonSize["Large"] = "large";
})(ButtonSize || (exports.ButtonSize = ButtonSize = {}));

var ButtonShape;
exports.ButtonShape = ButtonShape;

(function (ButtonShape) {
  ButtonShape["Sharp"] = "sharp";
  ButtonShape["Rounded"] = "rounded";
  ButtonShape["Circular"] = "circular";
})(ButtonShape || (exports.ButtonShape = ButtonShape = {}));
