// @tablecheck/tablekit-modal-dialog/lib/es5/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "AnimatedContent", {
  enumerable: true,
  get: function get() {
    return _AnimatedContent.AnimatedContent;
  }
});
Object.defineProperty(exports, "FooterWrapper", {
  enumerable: true,
  get: function get() {
    return _styled.FooterWrapper;
  }
});
Object.defineProperty(exports, "HeaderAppearance", {
  enumerable: true,
  get: function get() {
    return _types.HeaderAppearance;
  }
});
Object.defineProperty(exports, "HeaderWrapper", {
  enumerable: true,
  get: function get() {
    return _styled.HeaderWrapper;
  }
});
Object.defineProperty(exports, "ModalBody", {
  enumerable: true,
  get: function get() {
    return _styled.ModalBody;
  }
});
Object.defineProperty(exports, "ModalCloseButton", {
  enumerable: true,
  get: function get() {
    return _ModalCloseButton.ModalCloseButton;
  }
});
Object.defineProperty(exports, "ModalContent", {
  enumerable: true,
  get: function get() {
    return _styled.ModalContent;
  }
});
Object.defineProperty(exports, "ModalDialog", {
  enumerable: true,
  get: function get() {
    return _ModalDialog.ModalDialog;
  }
});
Object.defineProperty(exports, "ModalHeader", {
  enumerable: true,
  get: function get() {
    return _ModalHeader.ModalHeader;
  }
});
Object.defineProperty(exports, "ModalOverlay", {
  enumerable: true,
  get: function get() {
    return _styled.ModalOverlay;
  }
});
Object.defineProperty(exports, "ModalRoot", {
  enumerable: true,
  get: function get() {
    return _styled.ModalRoot;
  }
});
Object.defineProperty(exports, "ModalTitle", {
  enumerable: true,
  get: function get() {
    return _styled.ModalTitle;
  }
});
Object.defineProperty(exports, "modalClassicTheme", {
  enumerable: true,
  get: function get() {
    return _themes.modalClassicTheme;
  }
});
Object.defineProperty(exports, "modalDarkTheme", {
  enumerable: true,
  get: function get() {
    return _themes.modalDarkTheme;
  }
});
Object.defineProperty(exports, "modalThemeNamespace", {
  enumerable: true,
  get: function get() {
    return _themes.modalThemeNamespace;
  }
});

var _ModalDialog = require("./ModalDialog.js");

var _styled = require("./styled.js");

var _ModalCloseButton = require("./components/ModalCloseButton.js");

var _ModalHeader = require("./components/ModalHeader.js");

var _AnimatedContent = require("./components/AnimatedContent.js");

var _themes = require("./themes.js");

var _types = require("./types.js");
