// @tablecheck/tablekit-modal-dialog/lib/es5/themes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.modalThemeNamespace = exports.modalDarkTheme = exports.modalClassicTheme = void 0;

var _tablekitPackageNamespace = require("@tablecheck/tablekit-package-namespace");

var modalThemeNamespace = _tablekitPackageNamespace.PackageNamespace.ModalDialog;
exports.modalThemeNamespace = modalThemeNamespace;
var modalClassicTheme = {
  body: {
    textColor: function textColor(_ref) {
      var theme = _ref.theme;
      return theme.colors.text;
    },
    backgroundColor: function backgroundColor(_ref2) {
      var theme = _ref2.theme;
      return theme.colors.surface;
    }
  },
  keylineColor: function keylineColor(_ref3) {
    var theme = _ref3.theme;
    return theme.colors.border;
  },
  keylineHeight: '2px'
};
exports.modalClassicTheme = modalClassicTheme;
var modalDarkTheme = {
  body: {
    textColor: function textColor(_ref4) {
      var theme = _ref4.theme;
      return theme.colors.text;
    },
    backgroundColor: function backgroundColor(_ref5) {
      var theme = _ref5.theme;
      return theme.colors.surface;
    }
  },
  keylineColor: function keylineColor(_ref6) {
    var theme = _ref6.theme;
    return theme.colors.border;
  },
  keylineHeight: '2px'
};
exports.modalDarkTheme = modalDarkTheme;
