// @tablecheck/tablekit-language-selector/lib/es5/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ScrollWrapper = exports.LangItem = exports.ItemSubtitle = exports.InlineDialogWrapper = exports.Container = exports.ButtonWrapper = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _tablekitInlineDialog = require("@tablecheck/tablekit-inline-dialog");

var _tablekitItem = require("@tablecheck/tablekit-item");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitTypography = require("@tablecheck/tablekit-typography");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
} // the props passed as generic to styled component definition are correct
// the two generic args tend to conflict with each other


var InlineDialogWrapper = /*#__PURE__*/(0, _base.default)(_tablekitInlineDialog.InlineDialog, process.env.NODE_ENV === "production" ? {
  target: "e1382i4x5"
} : {
  target: "e1382i4x5",
  label: "InlineDialogWrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "jpnmw",
  styles: "height:100%;width:72px;>span{width:100%;}"
} : {
  name: "jpnmw",
  styles: "height:100%;width:72px;>span{width:100%;}",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});
exports.InlineDialogWrapper = InlineDialogWrapper;
var Container = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "e1382i4x4"
} : {
  target: "e1382i4x4",
  label: "Container"
})("display:flex;width:500px;flex-wrap:wrap;", _tablekitTypography.Typography.Body2, ";");
exports.Container = Container;
var ItemSubtitle = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "e1382i4x3"
} : {
  target: "e1382i4x3",
  label: "ItemSubtitle"
})("margin-left:", _tablekitTheme.Spacing.L1, ";font-size:14px;color:", function (_ref) {
  var theme = _ref.theme;
  return theme.colors.textSubtle;
}, ";");
exports.ItemSubtitle = ItemSubtitle;
var LangItem = /*#__PURE__*/(0, _base.default)(_tablekitItem.Item, process.env.NODE_ENV === "production" ? {
  target: "e1382i4x2"
} : {
  target: "e1382i4x2",
  label: "LangItem"
})("width:", function (_ref2) {
  var itemWidth = _ref2.itemWidth;
  return itemWidth;
}, ";height:", function (_ref3) {
  var itemHeight = _ref3.itemHeight;
  return itemHeight;
}, ";");
exports.LangItem = LangItem;
var ButtonWrapper = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "e1382i4x1"
} : {
  target: "e1382i4x1",
  label: "ButtonWrapper"
})("text-align:", function (_ref4) {
  var isRtl = _ref4.theme.isRtl;
  return isRtl ? 'left' : 'right';
}, ";padding:", _tablekitTheme.Spacing.L3, ";");
exports.ButtonWrapper = ButtonWrapper;
var ScrollWrapper = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "e1382i4x0"
} : {
  target: "e1382i4x0",
  label: "ScrollWrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "1im0r96",
  styles: "overflow-y:scroll"
} : {
  name: "1im0r96",
  styles: "overflow-y:scroll",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});
exports.ScrollWrapper = ScrollWrapper;
