// @tablecheck/tablekit-item/lib/es5/styled/ItemGroup.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TitleText = exports.TitleContainer = exports.ItemGroupContent = exports.ItemGroupContainer = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitTypography = require("@tablecheck/tablekit-typography");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ItemGroupContainer = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "ew9a6643"
} : {
  target: "ew9a6643",
  label: "ItemGroupContainer"
})("display:flex;flex-direction:column;min-width:", function (_ref) {
  var minWidth = _ref.minWidth;
  return minWidth || '1px';
}, ";");
exports.ItemGroupContainer = ItemGroupContainer;
var TitleContainer = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "ew9a6642"
} : {
  target: "ew9a6642",
  label: "TitleContainer"
})("margin:", _tablekitTheme.Spacing.L2, " ", _tablekitTheme.Spacing.L2, ";");
exports.TitleContainer = TitleContainer;
var TitleText = (0, _base.default)("p", process.env.NODE_ENV === "production" ? {
  target: "ew9a6641"
} : {
  target: "ew9a6641",
  label: "TitleText"
})(function (_ref2) {
  var isUppercase = _ref2.isUppercase;
  return isUppercase ? 'text-transform:  uppercase' : '';
}, ";font-weight:600;", _tablekitTypography.Typography.Body2, ";min-width:0;", function (_ref3) {
  var headingStyles = _ref3.headingStyles;
  return headingStyles;
}, ";");
exports.TitleText = TitleText;
var ItemGroupContent = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "ew9a6640"
} : {
  target: "ew9a6640",
  label: "ItemGroupContent"
})("display:flex;", function (props) {
  var orientation = props.orientation,
      hasIndent = props.hasIndent,
      shouldWrap = props.shouldWrap;
  var isHorizontal = orientation === 'horizontal';
  var flexDirection = isHorizontal ? 'row' : 'column';
  var wrapIndent = parseInt(_tablekitTheme.Spacing.L2, 10) / 2;

  if (shouldWrap) {
    return /*#__PURE__*/(0, _react.css)("flex-direction:", flexDirection, ";flex-wrap:wrap;margin:", hasIndent ? "-".concat(wrapIndent, "px") : '', ";&>*{margin:", hasIndent ? "".concat(wrapIndent, "px") : '', ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:ItemGroupContent;"));
  }

  if (hasIndent) {
    var lastChildMargin = isHorizontal ? {
      right: _tablekitTheme.Spacing.L2
    } : {
      bottom: _tablekitTheme.Spacing.L2
    };
    return /*#__PURE__*/(0, _react.css)("flex-direction:", flexDirection, ";&>*:not(:last-child){", (0, _tablekitUtils.margin)(lastChildMargin)(props), ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:ItemGroupContent;"));
  }

  return /*#__PURE__*/(0, _react.css)("flex-direction:", flexDirection, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:ItemGroupContent;"));
}, ";");
exports.ItemGroupContent = ItemGroupContent;
