// @tablecheck/tablekit-item/lib/es5/styled/Item.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ElementAfter", {
  enumerable: true,
  get: function get() {
    return _theme.ElementAfter;
  }
});
Object.defineProperty(exports, "ElementBefore", {
  enumerable: true,
  get: function get() {
    return _theme.ElementBefore;
  }
});
exports.ItemContent = exports.ItemBaseHtml = exports.ItemBaseComponent = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _isPropValid = _interopRequireDefault(require("@emotion/is-prop-valid"));

var _tablekitTypography = require("@tablecheck/tablekit-typography");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _theme = require("../utils/theme.js");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

var ItemBaseContainer = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "ejbi4qu3"
} : {
  target: "ejbi4qu3",
  label: "ItemBaseContainer"
})("box-sizing:border-box;", _theme.getSizeStyle, ";", _tablekitTypography.Typography.Body2, ";width:", (0, _tablekitUtils.mediaQuery)('width', function (value) {
  return value || '100%';
}), ";", function (_ref) {
  var shouldCenterAlignment = _ref.shouldCenterAlignment,
      hasOnlyIcon = _ref.hasOnlyIcon;
  return shouldCenterAlignment || hasOnlyIcon ? 'justify-content: center' : '';
}, ";align-items:center;text-decoration:none;cursor:pointer;&:focus{outline:none;text-decoration:none;}", _theme.interactionStates, " ", function (_ref2) {
  var customStyles = _ref2.customStyles;
  return customStyles;
}, ";");
var ItemBaseHtml = /*#__PURE__*/(0, _base.default)(ItemBaseContainer, process.env.NODE_ENV === "production" ? {
  shouldForwardProp: function shouldForwardProp(prop) {
    return (0, _isPropValid.default)(prop) && ['shouldCenterAlignment', 'interactionState', 'useVariantAsClass', 'size', 'borderPosition', 'isDisabled', 'isSelected', 'isHidden', 'hasOnlyIcon'].indexOf(prop.toString()) === -1;
  },
  target: "ejbi4qu2"
} : {
  shouldForwardProp: function shouldForwardProp(prop) {
    return (0, _isPropValid.default)(prop) && ['shouldCenterAlignment', 'interactionState', 'useVariantAsClass', 'size', 'borderPosition', 'isDisabled', 'isSelected', 'isHidden', 'hasOnlyIcon'].indexOf(prop.toString()) === -1;
  },
  target: "ejbi4qu2",
  label: "ItemBaseHtml"
})();
exports.ItemBaseHtml = ItemBaseHtml;
var ItemBaseComponent = /*#__PURE__*/(0, _base.default)(ItemBaseContainer, process.env.NODE_ENV === "production" ? {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['as', 'shouldCenterAlignment', 'interactionState', 'useVariantAsClass', 'size', 'borderPosition', 'isDisabled', 'isSelected', 'isHidden', 'hasOnlyIcon'].indexOf(prop.toString()) === -1;
  },
  target: "ejbi4qu1"
} : {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['as', 'shouldCenterAlignment', 'interactionState', 'useVariantAsClass', 'size', 'borderPosition', 'isDisabled', 'isSelected', 'isHidden', 'hasOnlyIcon'].indexOf(prop.toString()) === -1;
  },
  target: "ejbi4qu1",
  label: "ItemBaseComponent"
})();
exports.ItemBaseComponent = ItemBaseComponent;
var ItemContent = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "ejbi4qu0"
} : {
  target: "ejbi4qu0",
  label: "ItemContent"
})("width:", function (_ref3) {
  var shouldCenterAlignment = _ref3.shouldCenterAlignment;
  return shouldCenterAlignment ? 'auto' : '100%';
}, ";white-space:nowrap;overflow:hidden;text-overflow:ellipsis;");
exports.ItemContent = ItemContent;
