// @tablecheck/tablekit-modal-dialog/lib/es5/components/ModalHeader.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ModalHeader = exports.HeaderIconWrapper = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitIcon = require("@tablecheck/tablekit-icon");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _styled2 = require("../styled.js");

var _types = require("../types.js");

var _ModalCloseButton = require("./ModalCloseButton.js");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

var _variants;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var HeaderIconWrapper = (0, _base.default)("span", process.env.NODE_ENV === "production" ? {
  target: "euu9wjt0"
} : {
  target: "euu9wjt0",
  label: "HeaderIconWrapper"
})((0, _tablekitUtils.variant)({
  prop: 'appearance',
  default: 'default',
  variants: (_variants = {
    default: process.env.NODE_ENV === "production" ? {
      name: "19y1cr8",
      styles: "color:currentColor"
    } : {
      name: "kiw0cb-default",
      styles: "color:currentColor;label:default;",
      toString: _EMOTION_STRINGIFIED_CSS_ERROR__
    }
  }, _defineProperty(_variants, _types.HeaderAppearance.Danger, function (_ref) {
    var theme = _ref.theme;
    return /*#__PURE__*/(0, _react.css)("color:", theme.colors.danger, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:HeaderIconWrapper;"));
  }), _defineProperty(_variants, _types.HeaderAppearance.Warning, function (_ref2) {
    var theme = _ref2.theme;
    return /*#__PURE__*/(0, _react.css)("color:", theme.colors.warning, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:HeaderIconWrapper;"));
  }), _variants)
}), ";margin-right:", _tablekitTheme.Spacing.L2, ";flex:0 0 auto;padding:calc(", _tablekitTheme.Spacing.L3, " / 2) 0;");
exports.HeaderIconWrapper = HeaderIconWrapper;

var ModalHeader = function ModalHeader(_ref3) {
  var appearance = _ref3.appearance,
      hasCloseIcon = _ref3.hasCloseIcon,
      children = _ref3.children,
      hasKeyline = _ref3.hasKeyline;
  return (0, _jsxRuntime.jsxs)(_styled2.HeaderWrapper, {
    hasKeyline: hasKeyline,
    hasCloseIcon: hasCloseIcon,
    children: [appearance ? (0, _jsxRuntime.jsx)(HeaderIconWrapper, {
      appearance: appearance,
      children: (0, _jsxRuntime.jsx)(_tablekitIcon.Icon, {
        icon: (0, _tablekitIcon.getIcon)(appearance),
        size: _tablekitTheme.Size.Regular2
      })
    }) : null, children, hasCloseIcon ? (0, _jsxRuntime.jsx)(_ModalCloseButton.ModalCloseButton, {
      iconBefore: function iconBefore() {
        return (0, _jsxRuntime.jsx)(_tablekitIcon.Icon, {
          icon: (0, _tablekitIcon.getIcon)('close')
        });
      },
      tabIndex: -1
    }) : null]
  });
};

exports.ModalHeader = ModalHeader;
