// @tablecheck/tablekit-tag/lib/esm/styled.js
var _tagClassicTheme$comm, _variants, _variants2;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import _styled from '@emotion/styled/base';
import { css } from '@emotion/react';
import { Spacing, BORDER_RADIUS } from '@tablecheck/tablekit-theme';
import { Typography } from '@tablecheck/tablekit-typography';
import { getThemeValue, variant } from '@tablecheck/tablekit-utils';
import { tagThemeNamespace, tagClassicTheme } from './themes.js';
import { TagAppearance, TagSize } from './types.js';

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var borderWidth = 1;
var transition = process.env.NODE_ENV === "production" ? {
  name: "1uaihgp",
  styles: "transition:120ms ease-in-out"
} : {
  name: "1w3axge-transition",
  styles: "transition:120ms ease-in-out;label:transition;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};

var TagContent = _styled("span", process.env.NODE_ENV === "production" ? {
  target: "e1ed3sk3"
} : {
  target: "e1ed3sk3",
  label: "TagContent"
})("display:flex;align-items:center;align-self:center;margin:", function (_ref) {
  var isSelected = _ref.isSelected;
  return isSelected ? '0 6px 0 2px' : "0 ".concat(Spacing.L3);
}, ";max-width:100%;overflow:hidden;text-overflow:ellipsis;white-space:nowrap;font-weight:bold;");

var RemoveIconWrapper = _styled("span", process.env.NODE_ENV === "production" ? {
  target: "e1ed3sk2"
} : {
  target: "e1ed3sk2",
  label: "RemoveIconWrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "xatqlf",
  styles: "align-items:center;justify-content:center;width:auto;height:auto;align-self:center;display:flex;flex-shrink:0;margin:0;user-select:none"
} : {
  name: "xatqlf",
  styles: "align-items:center;justify-content:center;width:auto;height:auto;align-self:center;display:flex;flex-shrink:0;margin:0;user-select:none",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});

var TagWrapper = _styled("span", process.env.NODE_ENV === "production" ? {
  target: "e1ed3sk1"
} : {
  target: "e1ed3sk1",
  label: "TagWrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "1j7wp6a",
  styles: "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative"
} : {
  name: "1j7wp6a",
  styles: "align-self:center;display:inline-flex;flex-wrap:nowrap;max-width:100%;position:relative",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});

function renderAppearance(appearance, props) {
  var isSelected = props.isSelected;

  if (isSelected) {
    return /*#__PURE__*/css("color:", getThemeValue("".concat(tagThemeNamespace, ".").concat(appearance, ".selectedTextColor"), tagClassicTheme[appearance].selectedTextColor)(props), ";background:", getThemeValue("".concat(tagThemeNamespace, ".").concat(appearance, ".selectedBackgroundColor"), tagClassicTheme[appearance].selectedBackgroundColor)(props), ";&:hover{background:", getThemeValue("".concat(tagThemeNamespace, ".").concat(appearance, ".hoverSelectedBackgroundColor"), tagClassicTheme[appearance].hoverSelectedBackgroundColor)(props), ";}", RemoveIconWrapper, "{color:", getThemeValue("".concat(tagThemeNamespace, ".").concat(appearance, ".selectedTextColor"), tagClassicTheme[appearance].selectedTextColor)(props), ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:renderAppearance;"));
  }

  return /*#__PURE__*/css("color:", getThemeValue("".concat(tagThemeNamespace, ".").concat(appearance, ".textColor"), tagClassicTheme[appearance].textColor)(props), ";background:", getThemeValue("".concat(tagThemeNamespace, ".").concat(appearance, ".backgroundColor"), tagClassicTheme[appearance].backgroundColor)(props), ";&:hover{background:", getThemeValue("".concat(tagThemeNamespace, ".").concat(appearance, ".hoverBackgroundColor"), tagClassicTheme[appearance].hoverBackgroundColor)(props), ";}" + (process.env.NODE_ENV === "production" ? "" : ";label:renderAppearance;"));
}

var StyledTag = _styled("button", process.env.NODE_ENV === "production" ? {
  target: "e1ed3sk0"
} : {
  target: "e1ed3sk0",
  label: "StyledTag"
})("text-decoration:none;border-color:", getThemeValue("".concat(tagThemeNamespace, ".common.borderColor"), tagClassicTheme === null || tagClassicTheme === void 0 ? void 0 : (_tagClassicTheme$comm = tagClassicTheme.common) === null || _tagClassicTheme$comm === void 0 ? void 0 : _tagClassicTheme$comm.borderColor), ";", variant({
  prop: 'appearance',
  default: TagAppearance.Outline,
  variants: (_variants = {}, _defineProperty(_variants, TagAppearance.Outline, function (props) {
    return renderAppearance(TagAppearance.Outline, props);
  }), _defineProperty(_variants, TagAppearance.Solid, function (props) {
    return renderAppearance(TagAppearance.Solid, props);
  }), _variants)
}), " align-items:baseline;border-width:", borderWidth, "px;border-style:solid;box-sizing:border-box;cursor:pointer;display:inline-flex;margin:0;max-width:100%;outline:none;pointer-events:auto;text-align:center;white-space:nowrap;width:auto;vertical-align:middle;border-radius:", BORDER_RADIUS, "px;", Typography.Body2, ";padding:0 ", Spacing.L1, ";", variant({
  prop: 'size',
  default: TagSize.Regular,
  variants: (_variants2 = {}, _defineProperty(_variants2, TagSize.Regular, /*#__PURE__*/css("height:", Spacing.L6, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:StyledTag;"))), _defineProperty(_variants2, TagSize.Small, /*#__PURE__*/css("height:", Spacing.L5, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:StyledTag;"))), _variants2)
}), " & .svg-inline--fa{width:", Spacing.L4, ";height:", Spacing.L4, ";}&::-moz-focus-inner{border:0;margin:0;padding:0;}font-weight:bold;", transition, ";&::after{border-radius:", BORDER_RADIUS, "px;", transition, ";}");

export { RemoveIconWrapper, StyledTag, TagContent, TagWrapper, borderWidth, transition };
