// @tablecheck/tablekit-tag/lib/esm/Tag.js
var _excluded = ["color", "children", "isSelected", "className", "appearance", "size"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import { useTheme, ThemeProvider } from '@emotion/react';
import { Icon, getIcon } from '@tablecheck/tablekit-icon';
import { DARK_COLORS, CLASSIC_COLORS } from '@tablecheck/tablekit-theme';
import chroma from 'chroma-js/chroma-tablekit';
import { StyledTag, TagWrapper, TagContent, RemoveIconWrapper } from './styled.js';
import { tagThemeNamespace } from './themes.js';
import { TagAppearance, TagSize } from './types.js';
import { isDarkColor } from './utils.js';
import { jsx, jsxs } from '@emotion/react/jsx-runtime';

var Tag = function Tag(_ref) {
  var tagColor = _ref.color,
      children = _ref.children,
      _ref$isSelected = _ref.isSelected,
      isSelected = _ref$isSelected === void 0 ? false : _ref$isSelected,
      className = _ref.className,
      _ref$appearance = _ref.appearance,
      appearance = _ref$appearance === void 0 ? TagAppearance.Outline : _ref$appearance,
      _ref$size = _ref.size,
      size = _ref$size === void 0 ? TagSize.Regular : _ref$size,
      props = _objectWithoutProperties(_ref, _excluded);

  var parentTheme = useTheme();
  var tag = jsx(StyledTag, _objectSpread(_objectSpread({}, props), {}, {
    appearance: appearance,
    size: size,
    className: className,
    isSelected: isSelected,
    children: jsxs(TagWrapper, {
      children: [children ? jsx(TagContent, {
        isSelected: isSelected,
        children: children
      }) : null, isSelected ? jsx(RemoveIconWrapper, {
        children: jsx(Icon, {
          icon: getIcon('close')
        })
      }) : null]
    })
  }));

  if (!tagColor) {
    return tag;
  }

  var textColor = isDarkColor(tagColor) ? DARK_COLORS.text : CLASSIC_COLORS.text;
  var selectedSolidBgColor = chroma(tagColor).darken(2).hex();
  var selectedSolidTextColor = isDarkColor(selectedSolidBgColor) ? DARK_COLORS.text : CLASSIC_COLORS.text;
  return jsx(ThemeProvider, {
    theme: _defineProperty({}, tagThemeNamespace, {
      common: {
        borderColor: tagColor
      },
      outline: {
        textColor: parentTheme.colors.text,
        backgroundColor: 'transparent',
        hoverBackgroundColor: chroma(tagColor).alpha(0.2).css(),
        selectedBackgroundColor: tagColor,
        hoverSelectedBackgroundColor: chroma(tagColor).darken(0.7).css(),
        selectedTextColor: textColor
      },
      solid: {
        textColor: textColor,
        backgroundColor: tagColor,
        hoverBackgroundColor: chroma(tagColor).darken(0.7).css(),
        selectedBackgroundColor: selectedSolidBgColor,
        selectedTextColor: selectedSolidTextColor,
        hoverSelectedBackgroundColor: chroma(selectedSolidBgColor).brighten(0.7).css()
      }
    }),
    children: tag
  });
};

export { Tag };
