// @tablecheck/tablekit-input-button/lib/es5/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextWrapper = exports.SelectedIcon = exports.HiddenInput = exports.ButtonLabel = exports.ButtonDisplay = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _react = require("@emotion/react");

var _tablekitButton = require("@tablecheck/tablekit-button");

var _tablekitIcon = require("@tablecheck/tablekit-icon");

var _tablekitInput = require("@tablecheck/tablekit-input");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitTypography = require("@tablecheck/tablekit-typography");

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _constants = require("./constants.js");

var _variants2, _variants3, _variants4;

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

var transition = /*#__PURE__*/(0, _react.css)("transition:border-color ", _constants.TRANSITION_SETTINGS, ",color ", _constants.TRANSITION_SETTINGS, ",background-color ", _constants.TRANSITION_SETTINGS, ",box-shadow ", _constants.TRANSITION_SETTINGS, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:transition;"));
var SelectedIcon = /*#__PURE__*/(0, _base.default)(_tablekitIcon.Icon, process.env.NODE_ENV === "production" ? {
  target: "e173lbmz4"
} : {
  target: "e173lbmz4",
  label: "SelectedIcon"
})("width:15px;height:15px;margin:2.5px;margin-", (0, _tablekitUtils.ifRtl)('left', 'right'), ":10.5px;color:", function (_ref) {
  var theme = _ref.theme;
  return theme.colors.primary2;
}, ";");
exports.SelectedIcon = SelectedIcon;
var ButtonLabel = (0, _base.default)("label", process.env.NODE_ENV === "production" ? {
  target: "e173lbmz3"
} : {
  target: "e173lbmz3",
  label: "ButtonLabel"
})("display:inline-flex;flex-direction:row;position:relative;cursor:pointer;", _tablekitTypography.Typography.Body2, ";&,& *{user-select:none;}&[data-disabled='true']{cursor:not-allowed;}");
exports.ButtonLabel = ButtonLabel;
var ButtonDisplay = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "e173lbmz2"
} : {
  target: "e173lbmz2",
  label: "ButtonDisplay"
})("border-radius:", (0, _tablekitUtils.variant)({
  prop: 'shape',
  variants: (_variants2 = {}, _defineProperty(_variants2, _tablekitButton.ButtonShape.Sharp, 0), _defineProperty(_variants2, _tablekitButton.ButtonShape.Rounded, "".concat(_tablekitTheme.BORDER_RADIUS, "px")), _defineProperty(_variants2, _tablekitButton.ButtonShape.Circular, function (props) {
    var _variants;

    return (0, _tablekitUtils.variant)({
      prop: 'fieldSize',
      default: _tablekitInput.InputSize.Regular,
      variants: (_variants = {}, _defineProperty(_variants, _tablekitInput.InputSize.Large, _tablekitTheme.FieldHeight.Large), _defineProperty(_variants, _tablekitInput.InputSize.Regular, _tablekitTheme.FieldHeight.Regular), _defineProperty(_variants, _tablekitInput.InputSize.Regular2, _tablekitTheme.FieldHeight.Regular2), _defineProperty(_variants, _tablekitInput.InputSize.Small, _tablekitTheme.FieldHeight.Small), _variants)
    })(props);
  }), _variants2)
}), ";border:1px solid ", function (_ref2) {
  var theme = _ref2.theme;
  return theme.colors.border;
}, ";color:", function (_ref3) {
  var theme = _ref3.theme;
  return theme.colors.primary2;
}, ";max-width:300px;display:flex;justify-content:center;align-items:center;text-align:center;", transition, ";background-color:", function (_ref4) {
  var theme = _ref4.theme;
  return theme.colors.inputBackground;
}, ";&:not(", _constants.IS_CLICKED_SELECTOR, "):after{content:'';display:block;position:absolute;top:-5px;right:-5px;bottom:-5px;left:-5px;border:2px solid transparent;border-radius:", _tablekitTheme.BORDER_RADIUS + 2, "px;", transition, ";}");
exports.ButtonDisplay = ButtonDisplay;
var TextWrapper = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "e173lbmz1"
} : {
  target: "e173lbmz1",
  label: "TextWrapper"
})("overflow:hidden;text-overflow:ellipsis;white-space:nowrap;", _tablekitTypography.Typography.Body1, ";font-weight:", _tablekitTheme.FontWeight.SemiBold, ";&::after{content:attr(data-text);height:0;visibility:hidden;overflow:hidden;user-select:none;pointer-events:none;font-weight:600;display:block;@media speech{display:none;}}");
exports.TextWrapper = TextWrapper;
var HiddenInput = (0, _base.default)("input", process.env.NODE_ENV === "production" ? {
  target: "e173lbmz0"
} : {
  target: "e173lbmz0",
  label: "HiddenInput"
})("position:absolute;user-select:none;opacity:0;&+div{", (0, _tablekitUtils.variant)({
  prop: 'fieldSize',
  default: _tablekitInput.InputSize.Regular,
  variants: (_variants3 = {}, _defineProperty(_variants3, _tablekitInput.InputSize.Large, function () {
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Large, ";padding:0 32px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _defineProperty(_variants3, _tablekitInput.InputSize.Regular, function () {
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Regular, ";padding:0 22px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _defineProperty(_variants3, _tablekitInput.InputSize.Regular2, function () {
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Regular2, ";padding:0 26px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _defineProperty(_variants3, _tablekitInput.InputSize.Small, function () {
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Small, ";padding:0 20px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _variants3)
}), ";}&+div>svg{display:none;}&:hover+div{background:", function (_ref5) {
  var theme = _ref5.theme;
  return theme.colors.hoverPrimaryBackground;
}, ";}&:checked+div>svg{display:block;}&:checked+div{background:", function (_ref6) {
  var theme = _ref6.theme;
  return theme.colors.primarySelected;
}, ";border:1px solid ", function (_ref7) {
  var theme = _ref7.theme;
  return theme.colors.primary2;
}, ";", (0, _tablekitUtils.variant)({
  prop: 'fieldSize',
  default: _tablekitInput.InputSize.Regular,
  variants: (_variants4 = {}, _defineProperty(_variants4, _tablekitInput.InputSize.Large, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Large, ";padding:0 18px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"))), _defineProperty(_variants4, _tablekitInput.InputSize.Regular, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Regular, ";padding:0 8px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"))), _defineProperty(_variants4, _tablekitInput.InputSize.Regular2, /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Regular2, ";padding:0 12px;" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"))), _defineProperty(_variants4, _tablekitInput.InputSize.Small, function (props) {
    return /*#__PURE__*/(0, _react.css)("height:", _tablekitTheme.FieldHeight.Small, ";padding:0 ", props.shape === _tablekitButton.ButtonShape.Circular ? '10px' : '6px', ";" + (process.env.NODE_ENV === "production" ? "" : ";label:HiddenInput;"));
  }), _variants4)
}), ";}&:checked:hover+div{box-shadow:0 0 0 1px ", function (_ref8) {
  var theme = _ref8.theme;
  return theme.colors.primary;
}, ";background:", function (_ref9) {
  var theme = _ref9.theme;
  return theme.colors.hoverPrimaryBackground;
}, ";}&:disabled+div,&:disabled:hover+div{border:2px solid transparent;background:", function (_ref10) {
  var theme = _ref10.theme;
  return theme.colors.disabledBackground;
}, ";color:", function (_ref11) {
  var theme = _ref11.theme;
  return theme.colors.textSubtle;
}, ";}&:focus+div:after{border-color:", function (_ref12) {
  var theme = _ref12.theme;
  return theme.colors.focusOutline;
}, ";}");
exports.HiddenInput = HiddenInput;
