// @tablecheck/tablekit-input/lib/esm/themes.js
var _INPUT_WIDTHS;

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import { PackageNamespace } from '@tablecheck/tablekit-package-namespace';
import { Size } from '@tablecheck/tablekit-theme';
var INPUT_WIDTHS = (_INPUT_WIDTHS = {}, _defineProperty(_INPUT_WIDTHS, Size.XSmall, '64px'), _defineProperty(_INPUT_WIDTHS, Size.Small, '128px'), _defineProperty(_INPUT_WIDTHS, Size.Regular, '180px'), _defineProperty(_INPUT_WIDTHS, Size.Regular2, '218px'), _defineProperty(_INPUT_WIDTHS, Size.Large, '256px'), _defineProperty(_INPUT_WIDTHS, Size.XLarge, '330px'), _defineProperty(_INPUT_WIDTHS, Size.XXLarge, '368px'), _INPUT_WIDTHS);
var inputThemeNamespace = PackageNamespace.Input;
var inputClassicTheme = {
  label: {
    textColor: function textColor(_ref) {
      var theme = _ref.theme;
      return theme.colors.text;
    }
  },
  error: {
    textColor: function textColor(_ref2) {
      var theme = _ref2.theme;
      return theme.colors.text;
    }
  },
  input: {
    default: {
      backgroundColor: function backgroundColor(_ref3) {
        var theme = _ref3.theme;
        return theme.colors.inputBackground;
      },
      borderColor: function borderColor(_ref4) {
        var theme = _ref4.theme;
        return theme.colors.border;
      },
      textColor: function textColor(_ref5) {
        var theme = _ref5.theme;
        return theme.colors.text;
      },
      placeholderColor: function placeholderColor(_ref6) {
        var theme = _ref6.theme;
        return theme.colors.textSubtle;
      },
      iconColor: function iconColor(_ref7) {
        var theme = _ref7.theme;
        return theme.colors.primary2;
      }
    },
    disabled: {
      borderColor: function borderColor(_ref8) {
        var theme = _ref8.theme;
        return theme.colors.border;
      },
      textColor: function textColor(_ref9) {
        var theme = _ref9.theme;
        return theme.colors.textSubtle;
      },
      iconColor: function iconColor(_ref10) {
        var theme = _ref10.theme;
        return theme.colors.textSubtle;
      }
    }
  }
};
var inputDarkTheme = inputClassicTheme;
export { INPUT_WIDTHS, inputClassicTheme, inputDarkTheme, inputThemeNamespace };
