// @tablecheck/tablekit-panel/lib/es5/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "Panel", {
  enumerable: true,
  get: function get() {
    return _Panel.Panel;
  }
});
Object.defineProperty(exports, "PanelPosition", {
  enumerable: true,
  get: function get() {
    return _types.PanelPosition;
  }
});
Object.defineProperty(exports, "ResizablePanel", {
  enumerable: true,
  get: function get() {
    return _ResizablePanel.ResizablePanel;
  }
});
Object.defineProperty(exports, "baseDefaultHeight", {
  enumerable: true,
  get: function get() {
    return _themes.baseDefaultHeight;
  }
});
Object.defineProperty(exports, "baseDefaultWidth", {
  enumerable: true,
  get: function get() {
    return _themes.baseDefaultWidth;
  }
});
Object.defineProperty(exports, "defaultMaxHeight", {
  enumerable: true,
  get: function get() {
    return _themes.defaultMaxHeight;
  }
});
Object.defineProperty(exports, "defaultMaxWidth", {
  enumerable: true,
  get: function get() {
    return _themes.defaultMaxWidth;
  }
});
Object.defineProperty(exports, "panelClassicTheme", {
  enumerable: true,
  get: function get() {
    return _themes.panelClassicTheme;
  }
});
Object.defineProperty(exports, "panelDarkTheme", {
  enumerable: true,
  get: function get() {
    return _themes.panelDarkTheme;
  }
});
Object.defineProperty(exports, "panelThemeNamespace", {
  enumerable: true,
  get: function get() {
    return _themes.panelThemeNamespace;
  }
});

var _Panel = require("./Panel.js");

var _ResizablePanel = require("./ResizablePanel.js");

var _themes = require("./themes.js");

var _types = require("./types.js");
