// @tablecheck/tablekit-panel/lib/es5/themes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.panelThemeNamespace = exports.panelDarkTheme = exports.panelClassicTheme = exports.defaultMaxWidth = exports.defaultMaxHeight = exports.baseDefaultWidth = exports.baseDefaultHeight = void 0;

var _tablekitPackageNamespace = require("@tablecheck/tablekit-package-namespace");

var baseDefaultWidth = 360;
exports.baseDefaultWidth = baseDefaultWidth;
var defaultMaxWidth = 1000;
exports.defaultMaxWidth = defaultMaxWidth;
var baseDefaultHeight = 360;
exports.baseDefaultHeight = baseDefaultHeight;
var defaultMaxHeight = '100vh';
exports.defaultMaxHeight = defaultMaxHeight;
var panelThemeNamespace = _tablekitPackageNamespace.PackageNamespace.Panel;
exports.panelThemeNamespace = panelThemeNamespace;
var panelClassicTheme = {
  backgroundColor: function backgroundColor(_ref) {
    var theme = _ref.theme;
    return theme.colors.surface;
  },
  borderColor: function borderColor(_ref2) {
    var theme = _ref2.theme;
    return theme.colors.border;
  },
  borderColorHover: function borderColorHover(_ref3) {
    var theme = _ref3.theme;
    return theme.colors.primary;
  }
};
exports.panelClassicTheme = panelClassicTheme;
var panelDarkTheme = panelClassicTheme;
exports.panelDarkTheme = panelDarkTheme;
