// @tablecheck/tablekit-theme/lib/es5/index.js
"use strict";

function _typeof(obj) { "@babel/helpers - typeof"; return _typeof = "function" == typeof Symbol && "symbol" == typeof Symbol.iterator ? function (obj) { return typeof obj; } : function (obj) { return obj && "function" == typeof Symbol && obj.constructor === Symbol && obj !== Symbol.prototype ? "symbol" : typeof obj; }, _typeof(obj); }

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "BORDER_RADIUS", {
  enumerable: true,
  get: function get() {
    return _Theme.BORDER_RADIUS;
  }
});
Object.defineProperty(exports, "CLASSIC_COLORS", {
  enumerable: true,
  get: function get() {
    return _palettes.CLASSIC_COLORS;
  }
});
exports.COLORS = void 0;
Object.defineProperty(exports, "DARK_COLORS", {
  enumerable: true,
  get: function get() {
    return _palettes.DARK_COLORS;
  }
});
Object.defineProperty(exports, "DEPTH", {
  enumerable: true,
  get: function get() {
    return _Theme.DEPTH;
  }
});
Object.defineProperty(exports, "FieldHeight", {
  enumerable: true,
  get: function get() {
    return _Theme.FieldHeight;
  }
});
Object.defineProperty(exports, "FontWeight", {
  enumerable: true,
  get: function get() {
    return _weights.FontWeight;
  }
});
Object.defineProperty(exports, "Level", {
  enumerable: true,
  get: function get() {
    return _types.Level;
  }
});
Object.defineProperty(exports, "Size", {
  enumerable: true,
  get: function get() {
    return _Theme.Size;
  }
});
Object.defineProperty(exports, "Spacing", {
  enumerable: true,
  get: function get() {
    return _Theme.Spacing;
  }
});
Object.defineProperty(exports, "TYPOGRAPHY", {
  enumerable: true,
  get: function get() {
    return _Theme.TYPOGRAPHY;
  }
});
Object.defineProperty(exports, "ThemeProvider", {
  enumerable: true,
  get: function get() {
    return _ThemeProvider.ThemeProvider;
  }
});
Object.defineProperty(exports, "ZIndex", {
  enumerable: true,
  get: function get() {
    return _Theme.ZIndex;
  }
});
Object.defineProperty(exports, "getCdnPath", {
  enumerable: true,
  get: function get() {
    return _buildFontFace.getCdnPath;
  }
});
Object.defineProperty(exports, "resetCss", {
  enumerable: true,
  get: function get() {
    return _resetCss.resetCss;
  }
});
Object.defineProperty(exports, "useColor", {
  enumerable: true,
  get: function get() {
    return _hooks.useColor;
  }
});

var colors = _interopRequireWildcard(require("./colors.js"));

exports.COLORS = colors;

var _palettes = require("./palettes.js");

var _hooks = require("./hooks.js");

var _Theme = require("./Theme.js");

var _types = require("./types.js");

var _weights = require("./fonts/weights.js");

var _buildFontFace = require("./fonts/buildFontFace.js");

var _ThemeProvider = require("./ThemeProvider.js");

var _resetCss = require("./resetCss.js");

function _getRequireWildcardCache(nodeInterop) { if (typeof WeakMap !== "function") return null; var cacheBabelInterop = new WeakMap(); var cacheNodeInterop = new WeakMap(); return (_getRequireWildcardCache = function _getRequireWildcardCache(nodeInterop) { return nodeInterop ? cacheNodeInterop : cacheBabelInterop; })(nodeInterop); }

function _interopRequireWildcard(obj, nodeInterop) { if (!nodeInterop && obj && obj.__esModule) { return obj; } if (obj === null || _typeof(obj) !== "object" && typeof obj !== "function") { return { default: obj }; } var cache = _getRequireWildcardCache(nodeInterop); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (key !== "default" && Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } newObj.default = obj; if (cache) { cache.set(obj, newObj); } return newObj; }
