// @tablecheck/tablekit-theme/lib/es5/palettes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.DARK_PRIMARY_TRANSLUCENT = exports.DARK_COLORS = exports.CLASSIC_PRIMARY_TRANSLUCENT = exports.CLASSIC_COLORS = void 0;

var _tablekitUtils = require("@tablecheck/tablekit-utils");

var _colors = require("./colors.js");

var CLASSIC_PRIMARY = _colors.PURPLE.L5;
var CLASSIC_PRIMARY_TRANSLUCENT = {
  L1: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.1),
  L2: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.2),
  L3: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.3),
  L4: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.4),
  L5: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.5),
  L6: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.6),
  L7: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.7),
  L8: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.8),
  L9: (0, _tablekitUtils.hexToRgba)(CLASSIC_PRIMARY, 0.9)
};
exports.CLASSIC_PRIMARY_TRANSLUCENT = CLASSIC_PRIMARY_TRANSLUCENT;
var CLASSIC_COLORS = {
  canvas: _colors.GRAY.L1,
  primary: CLASSIC_PRIMARY,
  primary2: _colors.PURPLE.L7,
  primaryHover: _colors.PURPLE.L6,
  primary2Hover: _colors.PURPLE.L8,
  primarySelected: CLASSIC_PRIMARY_TRANSLUCENT.L2,
  primary2Selected: CLASSIC_PRIMARY_TRANSLUCENT.L1,
  invertedText: 'white',
  secondary: _colors.BLUE.L5,
  secondary2: _colors.BLUE.L4,
  accent: _colors.YELLOW.L4,
  success: _colors.GREEN.L5,
  warning: _colors.YELLOW.L5,
  danger: _colors.RED.L5,
  text: _colors.GRAY.L8,
  textSubtle: _colors.GRAY_TRANSLUCENT.L6,
  textDisabled: _colors.GRAY_TRANSLUCENT.L3,
  surface: 'white',
  surfaceHigh: _colors.WHITE_TRANSLUCENT.L9,
  surfaceLow: _colors.GRAY_TRANSLUCENT.L3,
  border: _colors.GRAY_TRANSLUCENT.L3,
  inputBackground: _colors.WHITE_TRANSLUCENT.L4,
  iconFill: _colors.GRAY_TRANSLUCENT.L7,
  hoverNeutralBackground: _colors.GRAY_TRANSLUCENT.L2,
  hoverPrimaryBackground: CLASSIC_PRIMARY_TRANSLUCENT.L1,
  selectedBackground: CLASSIC_PRIMARY_TRANSLUCENT.L2,
  disabledBackground: _colors.GRAY_TRANSLUCENT.L3,
  focusOutline: _colors.BLUE.L4
};
exports.CLASSIC_COLORS = CLASSIC_COLORS;
var DARK_PRIMARY = _colors.PURPLE.L3;
var DARK_PRIMARY_TRANSLUCENT = {
  L1: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.1),
  L2: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.2),
  L3: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.3),
  L4: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.4),
  L5: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.5),
  L6: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.6),
  L7: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.7),
  L8: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.8),
  L9: (0, _tablekitUtils.hexToRgba)(DARK_PRIMARY, 0.9)
};
exports.DARK_PRIMARY_TRANSLUCENT = DARK_PRIMARY_TRANSLUCENT;
var DARK_COLORS = {
  canvas: _colors.GRAY.L9,
  primary: DARK_PRIMARY,
  primary2: _colors.PURPLE.L2,
  primaryHover: _colors.PURPLE.L4,
  primary2Hover: _colors.PURPLE.L5,
  primarySelected: DARK_PRIMARY_TRANSLUCENT.L2,
  primary2Selected: DARK_PRIMARY_TRANSLUCENT.L1,
  invertedText: _colors.GRAY.L8,
  secondary: _colors.BLUE.L5,
  secondary2: _colors.BLUE.L4,
  accent: _colors.YELLOW.L4,
  success: _colors.GREEN.L6,
  warning: _colors.YELLOW.L6,
  danger: _colors.RED.L6,
  text: _colors.WHITE_TRANSLUCENT.L9,
  textSubtle: _colors.WHITE_TRANSLUCENT.L6,
  textDisabled: _colors.GRAY_TRANSLUCENT.L4,
  surface: _colors.GRAY.L8,
  surfaceHigh: _colors.WHITE_TRANSLUCENT.L4,
  surfaceLow: _colors.GRAY_TRANSLUCENT.L4,
  border: _colors.WHITE_TRANSLUCENT.L4,
  inputBackground: _colors.WHITE_TRANSLUCENT.L1,
  iconFill: _colors.WHITE_TRANSLUCENT.L8,
  hoverNeutralBackground: _colors.WHITE_TRANSLUCENT.L3,
  hoverPrimaryBackground: DARK_PRIMARY_TRANSLUCENT.L2,
  selectedBackground: DARK_PRIMARY_TRANSLUCENT.L3,
  disabledBackground: _colors.WHITE_TRANSLUCENT.L4,
  focusOutline: _colors.BLUE.L4
};
exports.DARK_COLORS = DARK_COLORS;
