// @tablecheck/tablekit-theme/lib/es5/Theme.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.ZIndex = exports.TYPOGRAPHY = exports.Spacing = exports.Size = exports.FieldHeight = exports.DEPTH = exports.BORDER_RADIUS = void 0;

/**
 * FieldHeight are used to ensure buttons, inputs, selects and the like all
 * use consistent heights if put on a horizontal line
 */
var FieldHeight;
exports.FieldHeight = FieldHeight;

(function (FieldHeight) {
  FieldHeight["Regular"] = "40px";
  FieldHeight["Regular2"] = "48px";
  FieldHeight["Large"] = "60px";
  FieldHeight["Small"] = "32px";
})(FieldHeight || (exports.FieldHeight = FieldHeight = {}));
/**
 * This scale is used for spacing between elements, usually margin and padding
 */


var Spacing;
exports.Spacing = Spacing;

(function (Spacing) {
  Spacing["L1"] = "4px";
  Spacing["L2"] = "8px";
  Spacing["L3"] = "12px";
  Spacing["L4"] = "16px";
  Spacing["L5"] = "24px";
  Spacing["L6"] = "32px";
  Spacing["L7"] = "40px";
  Spacing["L8"] = "48px";
})(Spacing || (exports.Spacing = Spacing = {}));
/**
 * Standard sizes used for component sizing
 */


var Size;
exports.Size = Size;

(function (Size) {
  Size["XSmall"] = "xsmall";
  Size["Small"] = "small";
  Size["Regular"] = "regular";
  Size["Regular2"] = "regular2";
  Size["Large"] = "large";
  Size["XLarge"] = "xlarge";
  Size["XXLarge"] = "xxlarge";
})(Size || (exports.Size = Size = {}));
/**
 * z-index list based on NAME, this is easier to manage
 */


var ZIndex;
exports.ZIndex = ZIndex;

(function (ZIndex) {
  ZIndex[ZIndex["Card"] = 100] = "Card";
  ZIndex[ZIndex["Dialog"] = 200] = "Dialog";
  ZIndex[ZIndex["Sidenav"] = 300] = "Sidenav";
  ZIndex[ZIndex["Topnav"] = 310] = "Topnav";
  ZIndex[ZIndex["Layer"] = 400] = "Layer";
  ZIndex[ZIndex["Blanket"] = 500] = "Blanket";
  ZIndex[ZIndex["Modal"] = 510] = "Modal";
  ZIndex[ZIndex["Flag"] = 600] = "Flag";
  ZIndex[ZIndex["Spotlight"] = 700] = "Spotlight";
  ZIndex[ZIndex["Tooltip"] = 800] = "Tooltip";
})(ZIndex || (exports.ZIndex = ZIndex = {}));

var BORDER_RADIUS = 8;
exports.BORDER_RADIUS = BORDER_RADIUS;
var DEPTH = {
  RAISED: {
    boxShadow: '-3px 0px 4px -3px rgba(0, 0, 0, 0.1), 3px 0px 4px -3px rgba(0, 0, 0, 0.1), 0px 3px 4px -3px rgba(0, 0, 0, 0.1)'
  },
  OVERLAY: {
    boxShadow: '0px 2px 8px 0px rgba(0, 0, 0, 0.1)'
  },
  FIXED: {
    boxShadow: '0px 4px 12px 0px rgba(0, 0, 0, 0.1)'
  },
  FLOAT: {
    boxShadow: '0px 6px 16px 0px rgba(0, 0, 0, 0.1)'
  }
};
exports.DEPTH = DEPTH;
var SizeLevel;

(function (SizeLevel) {
  SizeLevel["L1"] = "L1";
  SizeLevel["L2"] = "L2";
  SizeLevel["L3"] = "L3";
  SizeLevel["L4"] = "L4";
  SizeLevel["L5"] = "L5";
  SizeLevel["L6"] = "L6";
  SizeLevel["L7"] = "L7";
  SizeLevel["L8"] = "L8";
  SizeLevel["L9"] = "L9";
  SizeLevel["L10"] = "L10";
})(SizeLevel || (SizeLevel = {}));
/**
 * @deprecated Use `MobileTypography` or `DesktopTypography` from '@tablecheck/tablekit-typography'
 */


var TYPOGRAPHY = {
  L1: {
    fontSize: '13px',
    lineHeight: '20px'
  },
  L2: {
    fontSize: '14px',
    lineHeight: '20px'
  },
  L3: {
    fontSize: '14px',
    lineHeight: '20px'
  },
  L4: {
    fontSize: '18px',
    lineHeight: '24px'
  },
  L5: {
    fontSize: '22px',
    lineHeight: '28px'
  },
  L6: {
    fontSize: '22px',
    lineHeight: '28px'
  },
  L7: {
    fontSize: '32px',
    lineHeight: '40px'
  },
  L8: {
    fontSize: '32px',
    lineHeight: '40px'
  },
  L9: {
    fontSize: '44px',
    lineHeight: '52px'
  },
  L10: {
    fontSize: '44px',
    lineHeight: '52px'
  }
};
exports.TYPOGRAPHY = TYPOGRAPHY;
