// @tablecheck/tablekit-logo/lib/es5/Logo.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Logo = void 0;

var _LogoSymbol = require("./LogoSymbol.js");

var _LogoWording = require("./LogoWording.js");

var _styled = require("./styled.js");

var _jsxRuntime = require("@emotion/react/jsx-runtime");

var Logo = function Logo(_ref) {
  var className = _ref.className,
      display = _ref.display,
      isDarkTheme = _ref.isDarkTheme,
      _ref$symbolSize = _ref.symbolSize,
      symbolSize = _ref$symbolSize === void 0 ? '24px' : _ref$symbolSize,
      _ref$wordingSize = _ref.wordingSize,
      wordingSize = _ref$wordingSize === void 0 ? '96px' : _ref$wordingSize;
  return (0, _jsxRuntime.jsxs)(_styled.LogoWrapper, {
    className: className,
    children: [(0, _jsxRuntime.jsx)(_LogoSymbol.LogoSymbol, {
      size: symbolSize,
      isDarkTheme: isDarkTheme
    }), (0, _jsxRuntime.jsx)(_LogoWording.LogoWording, {
      display: display,
      size: wordingSize,
      isDarkTheme: isDarkTheme
    })]
  });
};

exports.Logo = Logo;
