// @tablecheck/tablekit-modal-dialog/lib/esm/components/AnimatedContent.js
import { Spacing } from '@tablecheck/tablekit-theme';
import { useTransition } from 'react-spring';
import { ModalOverlay, ModalContent } from '../styled.js';
import { jsx, Fragment, jsxs } from '@emotion/react/jsx-runtime';

var AnimatedContent = function AnimatedContent(props) {
  var children = props.children,
      className = props.className,
      height = props.height,
      width = props.width,
      maxHeight = props.maxHeight,
      maxWidth = props.maxWidth,
      isChromeless = props.isChromeless,
      isOpen = props.isOpen,
      onEscapeKeyDown = props.onEscapeKeyDown,
      onOpenAutoFocus = props.onOpenAutoFocus,
      onCloseAutoFocus = props.onCloseAutoFocus,
      shouldPreventCloseOutside = props.shouldPreventCloseOutside,
      _onPointerDownOutside = props.onPointerDownOutside,
      testId = props['data-testid'];
  var transition = useTransition(isOpen, {
    from: {
      opacity: 0,
      top: Spacing.L6
    },
    enter: {
      opacity: 1,
      top: "".concat(parseInt(Spacing.L6, 10) / 2, "px")
    },
    leave: {
      opacity: 0,
      top: "".concat(parseInt(Spacing.L6, 10) * 2, "px")
    }
  });
  return jsx(Fragment, {
    children: transition(function (styles, item) {
      if (!item) {
        return null;
      }

      return jsxs(Fragment, {
        children: [jsx(ModalOverlay, {
          forceMount: true,
          style: {
            opacity: styles.opacity
          }
        }), jsx(ModalContent, {
          className: className,
          forceMount: true,
          height: height,
          width: width,
          maxWidth: maxWidth,
          maxHeight: maxHeight,
          style: {
            opacity: styles.opacity,
            top: styles.top
          },
          isChromeless: isChromeless,
          onEscapeKeyDown: onEscapeKeyDown,
          onCloseAutoFocus: onCloseAutoFocus,
          onOpenAutoFocus: onOpenAutoFocus,
          onPointerDownOutside: function onPointerDownOutside(e) {
            if (_onPointerDownOutside) {
              _onPointerDownOutside(e);
            }

            if (shouldPreventCloseOutside) {
              e.preventDefault();
            }
          },
          "data-testid": testId,
          children: children
        })]
      });
    })
  });
};

export { AnimatedContent };
