// @tablecheck/tablekit-theme/lib/esm/palettes.js
import { hexToRgba } from '@tablecheck/tablekit-utils';
import { PURPLE, GRAY, BLUE, YELLOW, GREEN, RED, GRAY_TRANSLUCENT, WHITE_TRANSLUCENT } from './colors.js';
var CLASSIC_PRIMARY = PURPLE.L5;
var CLASSIC_PRIMARY_TRANSLUCENT = {
  L1: hexToRgba(CLASSIC_PRIMARY, 0.1),
  L2: hexToRgba(CLASSIC_PRIMARY, 0.2),
  L3: hexToRgba(CLASSIC_PRIMARY, 0.3),
  L4: hexToRgba(CLASSIC_PRIMARY, 0.4),
  L5: hexToRgba(CLASSIC_PRIMARY, 0.5),
  L6: hexToRgba(CLASSIC_PRIMARY, 0.6),
  L7: hexToRgba(CLASSIC_PRIMARY, 0.7),
  L8: hexToRgba(CLASSIC_PRIMARY, 0.8),
  L9: hexToRgba(CLASSIC_PRIMARY, 0.9)
};
var CLASSIC_COLORS = {
  canvas: GRAY.L1,
  primary: CLASSIC_PRIMARY,
  primary2: PURPLE.L7,
  primaryHover: PURPLE.L6,
  primary2Hover: PURPLE.L8,
  primarySelected: CLASSIC_PRIMARY_TRANSLUCENT.L2,
  primary2Selected: CLASSIC_PRIMARY_TRANSLUCENT.L1,
  invertedText: 'white',
  secondary: BLUE.L5,
  secondary2: BLUE.L4,
  accent: YELLOW.L4,
  success: GREEN.L5,
  warning: YELLOW.L5,
  danger: RED.L5,
  text: GRAY.L8,
  textSubtle: GRAY_TRANSLUCENT.L6,
  textDisabled: GRAY_TRANSLUCENT.L3,
  surface: 'white',
  surfaceHigh: WHITE_TRANSLUCENT.L9,
  surfaceLow: GRAY_TRANSLUCENT.L3,
  border: GRAY_TRANSLUCENT.L3,
  inputBackground: WHITE_TRANSLUCENT.L4,
  iconFill: GRAY_TRANSLUCENT.L7,
  hoverNeutralBackground: GRAY_TRANSLUCENT.L2,
  hoverPrimaryBackground: CLASSIC_PRIMARY_TRANSLUCENT.L1,
  selectedBackground: CLASSIC_PRIMARY_TRANSLUCENT.L2,
  disabledBackground: GRAY_TRANSLUCENT.L3,
  focusOutline: BLUE.L4
};
var DARK_PRIMARY = PURPLE.L3;
var DARK_PRIMARY_TRANSLUCENT = {
  L1: hexToRgba(DARK_PRIMARY, 0.1),
  L2: hexToRgba(DARK_PRIMARY, 0.2),
  L3: hexToRgba(DARK_PRIMARY, 0.3),
  L4: hexToRgba(DARK_PRIMARY, 0.4),
  L5: hexToRgba(DARK_PRIMARY, 0.5),
  L6: hexToRgba(DARK_PRIMARY, 0.6),
  L7: hexToRgba(DARK_PRIMARY, 0.7),
  L8: hexToRgba(DARK_PRIMARY, 0.8),
  L9: hexToRgba(DARK_PRIMARY, 0.9)
};
var DARK_COLORS = {
  canvas: GRAY.L9,
  primary: DARK_PRIMARY,
  primary2: PURPLE.L2,
  primaryHover: PURPLE.L4,
  primary2Hover: PURPLE.L5,
  primarySelected: DARK_PRIMARY_TRANSLUCENT.L2,
  primary2Selected: DARK_PRIMARY_TRANSLUCENT.L1,
  invertedText: GRAY.L8,
  secondary: BLUE.L5,
  secondary2: BLUE.L4,
  accent: YELLOW.L4,
  success: GREEN.L6,
  warning: YELLOW.L6,
  danger: RED.L6,
  text: WHITE_TRANSLUCENT.L9,
  textSubtle: WHITE_TRANSLUCENT.L6,
  textDisabled: GRAY_TRANSLUCENT.L4,
  surface: GRAY.L8,
  surfaceHigh: WHITE_TRANSLUCENT.L4,
  surfaceLow: GRAY_TRANSLUCENT.L4,
  border: WHITE_TRANSLUCENT.L4,
  inputBackground: WHITE_TRANSLUCENT.L1,
  iconFill: WHITE_TRANSLUCENT.L8,
  hoverNeutralBackground: WHITE_TRANSLUCENT.L3,
  hoverPrimaryBackground: DARK_PRIMARY_TRANSLUCENT.L2,
  selectedBackground: DARK_PRIMARY_TRANSLUCENT.L3,
  disabledBackground: WHITE_TRANSLUCENT.L4,
  focusOutline: BLUE.L4
};
export { CLASSIC_COLORS, CLASSIC_PRIMARY_TRANSLUCENT, DARK_COLORS, DARK_PRIMARY_TRANSLUCENT };
