// @tablecheck/tablekit-topnav/lib/es5/themes.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.topnavThemeNamespace = exports.topnavDarkTheme = exports.topnavClassicTheme = void 0;

var _tablekitPackageNamespace = require("@tablecheck/tablekit-package-namespace");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var topnavThemeNamespace = _tablekitPackageNamespace.PackageNamespace.Topnav;
exports.topnavThemeNamespace = topnavThemeNamespace;
var topnavClassicTheme = {
  backgroundColor: function backgroundColor(_ref) {
    var theme = _ref.theme;
    return theme.colors.surface;
  },
  textColor: function textColor(_ref2) {
    var theme = _ref2.theme;
    return theme.colors.text;
  },
  borderColor: _tablekitTheme.COLORS.GRAY.L3
};
exports.topnavClassicTheme = topnavClassicTheme;
var topnavDarkTheme = {
  backgroundColor: function backgroundColor(_ref3) {
    var theme = _ref3.theme;
    return theme.colors.surface;
  },
  textColor: function textColor(_ref4) {
    var theme = _ref4.theme;
    return theme.colors.text;
  },
  borderColor: function borderColor(_ref5) {
    var theme = _ref5.theme;
    return theme.colors.border;
  }
};
exports.topnavDarkTheme = topnavDarkTheme;
