// @tablecheck/tablekit-select/lib/es5/components/index.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
Object.defineProperty(exports, "ClearIndicator", {
  enumerable: true,
  get: function get() {
    return _indicators.ClearIndicator;
  }
});
Object.defineProperty(exports, "DropdownIndicator", {
  enumerable: true,
  get: function get() {
    return _indicators.DropdownIndicator;
  }
});
Object.defineProperty(exports, "ErrorIndicator", {
  enumerable: true,
  get: function get() {
    return _indicators.ErrorIndicator;
  }
});
exports.IndicatorSeparator = void 0;
Object.defineProperty(exports, "LoadingIndicator", {
  enumerable: true,
  get: function get() {
    return _indicators.LoadingIndicator;
  }
});
Object.defineProperty(exports, "SuccessIndicator", {
  enumerable: true,
  get: function get() {
    return _indicators.SuccessIndicator;
  }
});
Object.defineProperty(exports, "WarningIndicator", {
  enumerable: true,
  get: function get() {
    return _indicators.WarningIndicator;
  }
});

var _indicators = require("./indicators.js");

var IndicatorSeparator = function IndicatorSeparator() {
  return null;
};

exports.IndicatorSeparator = IndicatorSeparator;
