// @tablecheck/tablekit-textarea/lib/esm/Textarea.js
var _excluded = ["className", "type", "shouldFitContainer", "id", "appearance", "isRequired", "label", "inputCss", "inputClassName", "message", "isMessageHidden", "isSpellCheckEnabled", "maxLength", "autoComplete", "name", "placeholder", "form", "onBlur", "onChange", "onFocus", "onKeyDown", "onKeyPress", "onKeyUp", "value", "size", "enableResize", "minimumRows", "shouldAutoExpand", "width"];

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); enumerableOnly && (symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; })), keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = null != arguments[i] ? arguments[i] : {}; i % 2 ? ownKeys(Object(source), !0).forEach(function (key) { _defineProperty(target, key, source[key]); }) : Object.getOwnPropertyDescriptors ? Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)) : ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

import { Icon, getIcon, DangerIcon, WarningIcon, SuccessIcon } from '@tablecheck/tablekit-icon';
import { InputSize, Appearance, Label, InputContainer, InputPrependElement, InputAppendElement, InputMessage } from '@tablecheck/tablekit-input';
import { Spinner } from '@tablecheck/tablekit-spinner';
import { Size } from '@tablecheck/tablekit-theme';
import { forwardRef, useRef, cloneElement } from 'react';
import { TextareaInputWrapper, TextareaInput } from './styled.js';
import { jsx, jsxs } from '@emotion/react/jsx-runtime';

function setRef(ref, element) {
  if (!element || !ref) return;
  if (typeof ref === 'function') ref(element);else ref.current = element;
}

var Textarea = /*#__PURE__*/forwardRef(function (props, _ref) {
  var className = props.className,
      type = props.type,
      shouldFitContainer = props.shouldFitContainer,
      id = props.id,
      appearance = props.appearance,
      isRequired = props.isRequired,
      label = props.label,
      inputCss = props.inputCss,
      inputClassName = props.inputClassName,
      message = props.message,
      isMessageHidden = props.isMessageHidden,
      isSpellCheckEnabled = props.isSpellCheckEnabled,
      maxLength = props.maxLength,
      autoComplete = props.autoComplete,
      name = props.name,
      placeholder = props.placeholder,
      form = props.form,
      onBlur = props.onBlur,
      onChange = props.onChange,
      onFocus = props.onFocus,
      onKeyDown = props.onKeyDown,
      onKeyPress = props.onKeyPress,
      onKeyUp = props.onKeyUp,
      value = props.value,
      _props$size = props.size,
      size = _props$size === void 0 ? InputSize.Regular : _props$size,
      shouldEnableResize = props.enableResize,
      minimumRows = props.minimumRows,
      shouldAutoExpand = props.shouldAutoExpand,
      _props$width = props.width,
      width = _props$width === void 0 ? Size.Regular : _props$width,
      restProps = _objectWithoutProperties(props, _excluded);

  var iconBefore = props.iconBefore,
      iconAfter = props.iconAfter;
  var isDisabled = appearance === Appearance.Disabled;
  var innerRef = useRef();

  var autoResize = function autoResize() {
    if (!innerRef.current) return;
    setTimeout(function () {
      if (!innerRef.current) return;
      innerRef.current.style.height = 'auto';
      innerRef.current.style.height = "".concat(innerRef.current.scrollHeight, "px");
    });
  };

  var keyDownHandler = shouldAutoExpand ? function (event) {
    if (onKeyDown) onKeyDown(event);
    autoResize();
  } : onKeyDown;

  if (type === 'search') {
    iconBefore = jsx(Icon, {
      icon: getIcon('search')
    });
  }

  var labelElement = label !== undefined ? jsx(Label, {
    htmlFor: id,
    isRequired: isRequired,
    label: label
  }) : null;

  if (appearance === Appearance.Loading) {
    iconAfter = jsx(Spinner, {});
  } else if (appearance === Appearance.Invalid) {
    iconAfter = jsx(DangerIcon, {});
  } else if (appearance === Appearance.Warning) {
    iconAfter = jsx(WarningIcon, {});
  } else if (appearance === Appearance.Success) {
    iconAfter = jsx(SuccessIcon, {});
  } else if (iconAfter) {
    iconAfter = /*#__PURE__*/cloneElement(iconAfter, {
      size: Size.Regular
    });
  }

  return jsxs(InputContainer, {
    className: className,
    shouldFitContainer: shouldFitContainer,
    width: width,
    children: [labelElement, jsxs(TextareaInputWrapper, {
      children: [iconBefore && jsx(InputPrependElement, {
        elementSize: size,
        isDisabled: isDisabled,
        children: /*#__PURE__*/cloneElement(iconBefore, {
          size: Size.Regular
        })
      }), jsx(TextareaInput, _objectSpread(_objectSpread({}, restProps), {}, {
        autoComplete: autoComplete,
        className: inputClassName,
        css: inputCss,
        disabled: isDisabled,
        form: form,
        id: id,
        maxLength: maxLength,
        name: name,
        onBlur: onBlur,
        onChange: onChange,
        onFocus: onFocus,
        onKeyDown: keyDownHandler,
        onKeyPress: onKeyPress,
        onKeyUp: onKeyUp,
        placeholder: placeholder,
        readOnly: appearance === Appearance.ReadOnly,
        required: isRequired,
        spellCheck: isSpellCheckEnabled,
        size: size,
        type: type,
        value: value,
        minimumRows: minimumRows,
        enableResize: shouldEnableResize || isDisabled,
        ref: function ref(r) {
          setRef(_ref, r);
          setRef(innerRef, r);
          if (shouldAutoExpand) autoResize();
        }
      })), iconAfter && jsx(InputAppendElement, {
        elementSize: size,
        isDisabled: isDisabled,
        children: /*#__PURE__*/cloneElement(iconAfter, {
          size: Size.Regular
        })
      })]
    }), !isMessageHidden && jsx(InputMessage, {
      children: message
    })]
  });
});
export { Textarea };
