// @tablecheck/tablekit-modal-dialog/lib/esm/styled.js
var _excluded = ["hasCloseIcon"];

var _WIDTHS;

function _objectWithoutProperties(source, excluded) { if (source == null) return {}; var target = _objectWithoutPropertiesLoose(source, excluded); var key, i; if (Object.getOwnPropertySymbols) { var sourceSymbolKeys = Object.getOwnPropertySymbols(source); for (i = 0; i < sourceSymbolKeys.length; i++) { key = sourceSymbolKeys[i]; if (excluded.indexOf(key) >= 0) continue; if (!Object.prototype.propertyIsEnumerable.call(source, key)) continue; target[key] = source[key]; } } return target; }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _defineProperty(obj, key, value) { if (key in obj) { Object.defineProperty(obj, key, { value: value, enumerable: true, configurable: true, writable: true }); } else { obj[key] = value; } return obj; }

import _styled from '@emotion/styled/base';
import { css } from '@emotion/react';
import { Root, Overlay, Content } from '@radix-ui/react-dialog';
import { Size, COLORS, ZIndex, Spacing, BORDER_RADIUS, DEPTH } from '@tablecheck/tablekit-theme';
import { Typography } from '@tablecheck/tablekit-typography';
import { getThemeValue, mediaQuery, padding } from '@tablecheck/tablekit-utils';
import { animated } from 'react-spring';
import { modalThemeNamespace, modalClassicTheme } from './themes.js';

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

var ModalRoot = Root;
var ALLOWED_WIDTH = [Size.Large, Size.XLarge, Size.Regular, Size.Small];
var WIDTHS = (_WIDTHS = {}, _defineProperty(_WIDTHS, Size.Small, 400), _defineProperty(_WIDTHS, Size.Regular, 600), _defineProperty(_WIDTHS, Size.Large, 800), _defineProperty(_WIDTHS, Size.XLarge, 968), _WIDTHS);

var isAllowedModalWidth = function isAllowedModalWidth(width) {
  return ALLOWED_WIDTH.indexOf(width) !== -1;
};

var ModalOverlay = animated( /*#__PURE__*/_styled(Overlay, process.env.NODE_ENV === "production" ? {
  target: "e1a1xx716"
} : {
  target: "e1a1xx716",
  label: "ModalOverlay"
})("background-color:", COLORS.GRAY_TRANSLUCENT.L6, ";bottom:0;left:0;position:fixed;right:0;top:0;pointer-events:auto;z-index:", ZIndex.Modal, ";"));

var _ref4 = process.env.NODE_ENV === "production" ? {
  name: "d3v9zr",
  styles: "overflow:hidden"
} : {
  name: "lxra99-ModalContent",
  styles: "overflow:hidden;label:ModalContent;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};

var ModalContent = animated( /*#__PURE__*/_styled(Content, process.env.NODE_ENV === "production" ? {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['shouldPreventOverflow', 'height', 'maxHeight', 'width', 'maxWidth', 'isChromeless', 'className'].indexOf(prop) === -1;
  },
  target: "e1a1xx715"
} : {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['shouldPreventOverflow', 'height', 'maxHeight', 'width', 'maxWidth', 'isChromeless', 'className'].indexOf(prop) === -1;
  },
  target: "e1a1xx715",
  label: "ModalContent"
})("background-color:", getThemeValue("".concat(modalThemeNamespace, ".body.backgroundColor"), modalClassicTheme.body.backgroundColor), ";color:", function (_ref5) {
  var theme = _ref5.theme;
  return theme.colors.text;
}, ";outline:0;display:flex;flex-direction:column;position:fixed;top:calc(", Spacing.L6, " / 2);left:50%;transform:translateX(-50%);z-index:", ZIndex.Modal, ";", mediaQuery('maxWidth', function () {
  var currentSize = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "calc(100% - ".concat(Spacing.L6, ")");
  return typeof currentSize === 'string' ? "max-width: ".concat(currentSize, ";") : "max-width: ".concat(currentSize, "px;");
}), " ", mediaQuery('maxHeight', function () {
  var currentSize = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : "calc(100% - ".concat(Spacing.L5, ")");
  return typeof currentSize === 'string' ? "max-height: ".concat(currentSize, ";") : "max-height: ".concat(currentSize, "px;");
}), " height:", function (_ref6) {
  var height = _ref6.height;
  return typeof height === 'number' ? "".concat(height, "px") : height || 'auto';
}, ";width:", function (_ref7) {
  var width = _ref7.width;

  if (!width) {
    return 'auto';
  }

  if (typeof width === 'number') {
    return "".concat(width, "px");
  }

  if (isAllowedModalWidth(width)) {
    return "".concat(WIDTHS[width], "px");
  }

  return width;
}, ";", function (_ref8) {
  var isChromeless = _ref8.isChromeless;
  return !isChromeless ? /*#__PURE__*/css("border-radius:", BORDER_RADIUS, "px;", DEPTH.FIXED, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:ModalContent;")) : '';
}, " ", function (_ref9) {
  var shouldPreventOverflow = _ref9.shouldPreventOverflow;
  return shouldPreventOverflow ? _ref4 : '';
}, ";"));

var _ref3 = process.env.NODE_ENV === "production" ? {
  name: "1xuiarr",
  styles: "overflow-y:auto;overflow-x:hidden;max-height:100%"
} : {
  name: "4xyxuz-ModalBody",
  styles: "overflow-y:auto;overflow-x:hidden;max-height:100%;label:ModalBody;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};

var ModalBody = /*#__PURE__*/_styled('div', process.env.NODE_ENV === "production" ? {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['hasGutters', 'isScrollable'].indexOf(prop) === -1;
  },
  target: "e1a1xx714"
} : {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['hasGutters', 'isScrollable'].indexOf(prop) === -1;
  },
  target: "e1a1xx714",
  label: "ModalBody"
})("flex:1 1 auto;padding:", function (_ref10) {
  var hasGutters = _ref10.hasGutters;
  return hasGutters ? "0 ".concat(Spacing.L5) : 0;
}, ";margin:", function (_ref11) {
  var hasGutters = _ref11.hasGutters;
  return hasGutters ? "".concat(Spacing.L2, " 0") : 0;
}, ";", function (_ref12) {
  var isScrollable = _ref12.isScrollable;
  return isScrollable ? _ref3 : '';
}, ";");

var _ref = process.env.NODE_ENV === "production" ? {
  name: "l8l8b8",
  styles: "white-space:nowrap;overflow:hidden;text-overflow:ellipsis"
} : {
  name: "mi2dzi-ModalTitle",
  styles: "white-space:nowrap;overflow:hidden;text-overflow:ellipsis;label:ModalTitle;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};

var _ref2 = process.env.NODE_ENV === "production" ? {
  name: "1um6peu",
  styles: "word-wrap:break-word;width:100%"
} : {
  name: "1p73qs6-ModalTitle",
  styles: "word-wrap:break-word;width:100%;label:ModalTitle;",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
};

var ModalTitle = /*#__PURE__*/_styled('h4', process.env.NODE_ENV === "production" ? {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'isMultiline';
  },
  target: "e1a1xx713"
} : {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'isMultiline';
  },
  target: "e1a1xx713",
  label: "ModalTitle"
})(Typography.Heading2, ";color:", function (_ref13) {
  var theme = _ref13.theme;
  return theme.colors.text;
}, ";font-weight:400;line-height:37px;letter-spacing:-0.008em;flex:1;", function (_ref14) {
  var isMultiline = _ref14.isMultiline;
  return isMultiline ? _ref2 : _ref;
}, ";");

var BaseWrapper = _styled("div", process.env.NODE_ENV === "production" ? {
  target: "e1a1xx712"
} : {
  target: "e1a1xx712",
  label: "BaseWrapper"
})("align-items:flex-start;color:", function (_ref15) {
  var theme = _ref15.theme;
  return theme.colors.text;
}, ";display:flex;flex-direction:", function (_ref16) {
  var isRtl = _ref16.theme.isRtl;
  return isRtl ? 'row-reverse' : 'row';
}, ";flex:0 0 auto;min-height:56px;padding:", Spacing.L3, " ", Spacing.L5, ";width:100%;");

var getKeyline = function getKeyline(position) {
  return function (_ref17) {
    var hasKeyline = _ref17.hasKeyline,
        theme = _ref17.theme;

    if (hasKeyline) {
      var color = getThemeValue("".concat(modalThemeNamespace, ".keylineColor"), modalClassicTheme.keylineColor)({
        theme: theme
      });
      var height = getThemeValue("".concat(modalThemeNamespace, ".keylineHeight"), modalClassicTheme.keylineHeight)({
        theme: theme
      });
      return /*#__PURE__*/css("z-index:1;box-shadow:0 ", position === 'top' ? '-' : '', height, " 0 0 ", color, ";" + (process.env.NODE_ENV === "production" ? "" : ";label:getKeyline;"));
    }

    return '';
  };
};

var HeaderWrapper = /*#__PURE__*/_styled(BaseWrapper, process.env.NODE_ENV === "production" ? {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['hasCloseIcon', 'hasKeyline'].indexOf(prop) === -1;
  },
  target: "e1a1xx711"
} : {
  shouldForwardProp: function shouldForwardProp(prop) {
    return ['hasCloseIcon', 'hasKeyline'].indexOf(prop) === -1;
  },
  target: "e1a1xx711",
  label: "HeaderWrapper"
})(function (_ref18) {
  var hasCloseIcon = _ref18.hasCloseIcon,
      props = _objectWithoutProperties(_ref18, _excluded);

  return hasCloseIcon ? /*#__PURE__*/css(padding({
    top: Spacing.L3,
    bottom: Spacing.L3,
    left: Spacing.L5,
    right: Spacing.L4
  })(props), ";" + (process.env.NODE_ENV === "production" ? "" : ";label:HeaderWrapper;")) : '';
}, ";justify-content:space-between;", getKeyline('bottom'), ";");

var FooterWrapper = /*#__PURE__*/_styled(BaseWrapper, process.env.NODE_ENV === "production" ? {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'hasKeyline';
  },
  target: "e1a1xx710"
} : {
  shouldForwardProp: function shouldForwardProp(prop) {
    return prop !== 'hasKeyline';
  },
  target: "e1a1xx710",
  label: "FooterWrapper"
})("justify-content:flex-end;", getKeyline('top'), ";");

export { FooterWrapper, HeaderWrapper, ModalBody, ModalContent, ModalOverlay, ModalRoot, ModalTitle };
