// @tablecheck/tablekit-modal-dialog/lib/esm/themes.js
import { PackageNamespace } from '@tablecheck/tablekit-package-namespace';
var modalThemeNamespace = PackageNamespace.ModalDialog;
var modalClassicTheme = {
  body: {
    textColor: function textColor(_ref) {
      var theme = _ref.theme;
      return theme.colors.text;
    },
    backgroundColor: function backgroundColor(_ref2) {
      var theme = _ref2.theme;
      return theme.colors.surface;
    }
  },
  keylineColor: function keylineColor(_ref3) {
    var theme = _ref3.theme;
    return theme.colors.border;
  },
  keylineHeight: '2px'
};
var modalDarkTheme = {
  body: {
    textColor: function textColor(_ref4) {
      var theme = _ref4.theme;
      return theme.colors.text;
    },
    backgroundColor: function backgroundColor(_ref5) {
      var theme = _ref5.theme;
      return theme.colors.surface;
    }
  },
  keylineColor: function keylineColor(_ref6) {
    var theme = _ref6.theme;
    return theme.colors.border;
  },
  keylineHeight: '2px'
};
export { modalClassicTheme, modalDarkTheme, modalThemeNamespace };
