// @tablecheck/tablekit-textarea/lib/es5/styled.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.TextareaInputWrapper = exports.TextareaInput = void 0;

var _base = _interopRequireDefault(require("@emotion/styled/base"));

var _tablekitInput = require("@tablecheck/tablekit-input");

var _tablekitTheme = require("@tablecheck/tablekit-theme");

var _tablekitTypography = require("@tablecheck/tablekit-typography");

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { default: obj }; }

function _EMOTION_STRINGIFIED_CSS_ERROR__() {
  return "You have tried to stringify object returned from `css` function. It isn't supposed to be used directly (e.g. as value of the `className` prop), but rather handed to emotion so it can handle it (e.g. as value of `css` prop).";
}

function calculateMinimumHeight(props) {
  var _props$minimumRows = props.minimumRows,
      minimumRows = _props$minimumRows === void 0 ? 1 : _props$minimumRows;
  var padding = parseFloat(_tablekitTheme.Spacing.L3) * 2;
  var typographySettings = _tablekitTypography.Typography.Body2;
  var lineHeight = parseFloat(typographySettings.lineHeight);
  return "".concat(padding + lineHeight * minimumRows, "px");
}

var TextareaInput = /*#__PURE__*/(0, _base.default)(_tablekitInput.StyledInput.withComponent('textarea', process.env.NODE_ENV === "production" ? {
  target: "e12v3g1i2"
} : {
  target: "e12v3g1i2",
  label: "TextareaInput"
}), process.env.NODE_ENV === "production" ? {
  target: "e12v3g1i1"
} : {
  target: "e12v3g1i1",
  label: "TextareaInput"
})("box-sizing:border-box;height:auto;position:relative;display:flex;flex:1 1 100%;padding-top:", _tablekitTheme.Spacing.L3, ";padding-bottom:", _tablekitTheme.Spacing.L3, ";min-height:", calculateMinimumHeight, ";", _tablekitTypography.Typography.Body2, ";", function (_ref) {
  var enableResize = _ref.enableResize;
  return enableResize ? '' : 'resize: none;';
}, ";&::-ms-clear{display:none;}");
exports.TextareaInput = TextareaInput;
var TextareaInputWrapper = (0, _base.default)("div", process.env.NODE_ENV === "production" ? {
  target: "e12v3g1i0"
} : {
  target: "e12v3g1i0",
  label: "TextareaInputWrapper"
})(process.env.NODE_ENV === "production" ? {
  name: "1ww8r31",
  styles: "position:relative;width:100%;display:flex;flex:1 1 auto"
} : {
  name: "1ww8r31",
  styles: "position:relative;width:100%;display:flex;flex:1 1 auto",
  toString: _EMOTION_STRINGIFIED_CSS_ERROR__
});
exports.TextareaInputWrapper = TextareaInputWrapper;
